<!DOCTYPE html>
<html lang="en"><head>
<script src="mpg6_files/libs/clipboard/clipboard.min.js"></script>
<script src="mpg6_files/libs/quarto-html/tabby.min.js"></script>
<script src="mpg6_files/libs/quarto-html/popper.min.js"></script>
<script src="mpg6_files/libs/quarto-html/tippy.umd.min.js"></script>
<link href="mpg6_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="mpg6_files/libs/quarto-html/light-border.css" rel="stylesheet">
<link href="mpg6_files/libs/quarto-html/quarto-html.min.css" rel="stylesheet" data-mode="light">
<link href="mpg6_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles"><meta charset="utf-8">
  <meta name="generator" content="quarto-1.5.57">

  <meta name="author" content="Sarah Urbut, MD PhD">
  <meta name="dcterms.date" content="2025-04-17">
  <title>A Practical Primer on Bayesian Statistics</title>
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">
  <link rel="stylesheet" href="mpg6_files/libs/revealjs/dist/reset.css">
  <link rel="stylesheet" href="mpg6_files/libs/revealjs/dist/reveal.css">
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    div.columns{display: flex; gap: min(4vw, 1.5em);}
    div.column{flex: auto; overflow-x: auto;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    ul.task-list li input[type="checkbox"] {
      width: 0.8em;
      margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
      vertical-align: middle;
    }
    /* CSS for syntax highlighting */
    pre > code.sourceCode { white-space: pre; position: relative; }
    pre > code.sourceCode > span { line-height: 1.25; }
    pre > code.sourceCode > span:empty { height: 1.2em; }
    .sourceCode { overflow: visible; }
    code.sourceCode > span { color: inherit; text-decoration: inherit; }
    div.sourceCode { margin: 1em 0; }
    pre.sourceCode { margin: 0; }
    @media screen {
    div.sourceCode { overflow: auto; }
    }
    @media print {
    pre > code.sourceCode { white-space: pre-wrap; }
    pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
    }
    pre.numberSource code
      { counter-reset: source-line 0; }
    pre.numberSource code > span
      { position: relative; left: -4em; counter-increment: source-line; }
    pre.numberSource code > span > a:first-child::before
      { content: counter(source-line);
        position: relative; left: -1em; text-align: right; vertical-align: baseline;
        border: none; display: inline-block;
        -webkit-touch-callout: none; -webkit-user-select: none;
        -khtml-user-select: none; -moz-user-select: none;
        -ms-user-select: none; user-select: none;
        padding: 0 4px; width: 4em;
      }
    pre.numberSource { margin-left: 3em;  padding-left: 4px; }
    div.sourceCode
      { color: #24292e;  }
    @media screen {
    pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
    }
    code span { color: #24292e; } /* Normal */
    code span.al { color: #ff5555; font-weight: bold; } /* Alert */
    code span.an { color: #6a737d; } /* Annotation */
    code span.at { color: #d73a49; } /* Attribute */
    code span.bn { color: #005cc5; } /* BaseN */
    code span.bu { color: #d73a49; } /* BuiltIn */
    code span.cf { color: #d73a49; } /* ControlFlow */
    code span.ch { color: #032f62; } /* Char */
    code span.cn { color: #005cc5; } /* Constant */
    code span.co { color: #6a737d; } /* Comment */
    code span.cv { color: #6a737d; } /* CommentVar */
    code span.do { color: #6a737d; } /* Documentation */
    code span.dt { color: #d73a49; } /* DataType */
    code span.dv { color: #005cc5; } /* DecVal */
    code span.er { color: #ff5555; text-decoration: underline; } /* Error */
    code span.ex { color: #d73a49; font-weight: bold; } /* Extension */
    code span.fl { color: #005cc5; } /* Float */
    code span.fu { color: #6f42c1; } /* Function */
    code span.im { color: #032f62; } /* Import */
    code span.in { color: #6a737d; } /* Information */
    code span.kw { color: #d73a49; } /* Keyword */
    code span.op { color: #24292e; } /* Operator */
    code span.ot { color: #6f42c1; } /* Other */
    code span.pp { color: #d73a49; } /* Preprocessor */
    code span.re { color: #6a737d; } /* RegionMarker */
    code span.sc { color: #005cc5; } /* SpecialChar */
    code span.ss { color: #032f62; } /* SpecialString */
    code span.st { color: #032f62; } /* String */
    code span.va { color: #e36209; } /* Variable */
    code span.vs { color: #032f62; } /* VerbatimString */
    code span.wa { color: #ff5555; } /* Warning */
  </style>
  <link rel="stylesheet" href="mpg6_files/libs/revealjs/dist/theme/quarto.css">
  <link href="mpg6_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.css" rel="stylesheet">
  <link href="mpg6_files/libs/revealjs/plugin/reveal-menu/menu.css" rel="stylesheet">
  <link href="mpg6_files/libs/revealjs/plugin/reveal-menu/quarto-menu.css" rel="stylesheet">
  <link href="mpg6_files/libs/revealjs/plugin/quarto-support/footer.css" rel="stylesheet">
  <style type="text/css">

  .callout {
    margin-top: 1em;
    margin-bottom: 1em;  
    border-radius: .25rem;
  }

  .callout.callout-style-simple { 
    padding: 0em 0.5em;
    border-left: solid #acacac .3rem;
    border-right: solid 1px silver;
    border-top: solid 1px silver;
    border-bottom: solid 1px silver;
    display: flex;
  }

  .callout.callout-style-default {
    border-left: solid #acacac .3rem;
    border-right: solid 1px silver;
    border-top: solid 1px silver;
    border-bottom: solid 1px silver;
  }

  .callout .callout-body-container {
    flex-grow: 1;
  }

  .callout.callout-style-simple .callout-body {
    font-size: 1rem;
    font-weight: 400;
  }

  .callout.callout-style-default .callout-body {
    font-size: 0.9rem;
    font-weight: 400;
  }

  .callout.callout-titled.callout-style-simple .callout-body {
    margin-top: 0.2em;
  }

  .callout:not(.callout-titled) .callout-body {
      display: flex;
  }

  .callout:not(.no-icon).callout-titled.callout-style-simple .callout-content {
    padding-left: 1.6em;
  }

  .callout.callout-titled .callout-header {
    padding-top: 0.2em;
    margin-bottom: -0.2em;
  }

  .callout.callout-titled .callout-title  p {
    margin-top: 0.5em;
    margin-bottom: 0.5em;
  }
    
  .callout.callout-titled.callout-style-simple .callout-content  p {
    margin-top: 0;
  }

  .callout.callout-titled.callout-style-default .callout-content  p {
    margin-top: 0.7em;
  }

  .callout.callout-style-simple div.callout-title {
    border-bottom: none;
    font-size: .9rem;
    font-weight: 600;
    opacity: 75%;
  }

  .callout.callout-style-default  div.callout-title {
    border-bottom: none;
    font-weight: 600;
    opacity: 85%;
    font-size: 0.9rem;
    padding-left: 0.5em;
    padding-right: 0.5em;
  }

  .callout.callout-style-default div.callout-content {
    padding-left: 0.5em;
    padding-right: 0.5em;
  }

  .callout.callout-style-simple .callout-icon::before {
    height: 1rem;
    width: 1rem;
    display: inline-block;
    content: "";
    background-repeat: no-repeat;
    background-size: 1rem 1rem;
  }

  .callout.callout-style-default .callout-icon::before {
    height: 0.9rem;
    width: 0.9rem;
    display: inline-block;
    content: "";
    background-repeat: no-repeat;
    background-size: 0.9rem 0.9rem;
  }

  .callout-title {
    display: flex
  }
    
  .callout-icon::before {
    margin-top: 1rem;
    padding-right: .5rem;
  }

  .callout.no-icon::before {
    display: none !important;
  }

  .callout.callout-titled .callout-body > .callout-content > :last-child {
    padding-bottom: 0.5rem;
    margin-bottom: 0;
  }

  .callout.callout-titled .callout-icon::before {
    margin-top: .5rem;
    padding-right: .5rem;
  }

  .callout:not(.callout-titled) .callout-icon::before {
    margin-top: 1rem;
    padding-right: .5rem;
  }

  /* Callout Types */

  div.callout-note {
    border-left-color: #4582ec !important;
  }

  div.callout-note .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAEU0lEQVRYCcVXTWhcVRQ+586kSUMMxkyaElstCto2SIhitS5Ek8xUKV2poatCcVHtUlFQk8mbaaziwpWgglJwVaquitBOfhQXFlqlzSJpFSpIYyXNjBNiTCck7x2/8/LeNDOZxDuEkgOXe++553zfefee+/OYLOXFk3+1LLrRdiO81yNqZ6K9cG0P3MeFaMIQjXssE8Z1JzLO9ls20MBZX7oG8w9GxB0goaPrW5aNMp1yOZIa7Wv6o2ykpLtmAPs/vrG14Z+6d4jpbSKuhdcSyq9wGMPXjonwmESXrriLzFGOdDBLB8Y6MNYBu0dRokSygMA/mrun8MGFN3behm6VVAwg4WR3i6FvYK1T7MHo9BK7ydH+1uurECoouk5MPRyVSBrBHMYwVobG2aOXM07sWrn5qgB60rc6mcwIDJtQrnrEr44kmy+UO9r0u9O5/YbkS9juQckLed3DyW2XV/qWBBB3ptvI8EUY3I9p/67OW+g967TNr3Sotn3IuVlfMLVnsBwH4fsnebJvyGm5GeIUA3jljERmrv49SizPYuq+z7c2H/jlGC+Ghhupn/hcapqmcudB9jwJ/3jvnvu6vu5lVzF1fXyZuZZ7U8nRmVzytvT+H3kilYvH09mLWrQdwFSsFEsxFVs5fK7A0g8gMZjbif4ACpKbjv7gNGaD8bUrlk8x+KRflttr22JEMRUbTUwwDQScyzPgedQHZT0xnx7ujw2jfVfExwYHwOsDTjLdJ2ebmeQIlJ7neo41s/DrsL3kl+W2lWvAga0tR3zueGr6GL78M3ifH0rGXrBC2aAR8uYcIA5gwV8zIE8onoh8u0Fca/ciF7j1uOzEnqcIm59sEXoGc0+z6+H45V1CvAvHcD7THztu669cnp+L0okAeIc6zjbM/24LgGM1gZk7jnRu1aQWoU9sfUOuhrmtaPIO3YY1KLLWZaEO5TKUbMY5zx8W9UJ6elpLwKXbsaZ4EFl7B4bMtDv0iRipKoDQT2sNQI9b1utXFdYisi+wzZ/ri/1m7QfDgEuvgUUEIJPq3DhX/5DWNqIXDOweC2wvIR90Oq3lDpdMIgD2r0dXvGdsEW5H6x6HLRJYU7C69VefO1x8Gde1ZFSJLfWS1jbCnhtOPxmpfv2LXOA2Xk2tvnwKKPFuZ/oRmwBwqRQDcKNeVQkYcOjtWVBuM/JuYw5b6isojIkYxyYAFn5K7ZBF10fea52y8QltAg6jnMqNHFBmGkQ1j+U43HMi2xMar1Nv0zGsf1s8nUsmUtPOOrbFIR8bHFDMB5zL13Gmr/kGlCkUzedTzzmzsaJXhYawnA3UmARpiYj5ooJZiUoxFRtK3X6pgNPv+IZVPcnwbOl6f+aBaO1CNvPW9n9LmCp01nuSaTRF2YxHqZ8DYQT6WsXT+RD6eUztwYLZ8rM+rcPxamv1VQzFUkzFXvkiVrySGQgJNvXHJAxiU3/NwiC03rSf05VBaPtu/Z7/B8Yn/w7eguloAAAAAElFTkSuQmCC');
  }

  div.callout-note.callout-style-default .callout-title {
    background-color: #dae6fb
  }

  div.callout-important {
    border-left-color: #d9534f !important;
  }

  div.callout-important .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAEKklEQVRYCcVXTWhcVRS+575MJym48A+hSRFr00ySRQhURRfd2HYjk2SSTokuBCkU2o0LoSKKraKIBTcuFCoidGFD08nkBzdREbpQ1EDNIv8qSGMFUboImMSZd4/f9zJv8ibJMC8xJQfO3HPPPef7zrvvvnvviIkpC9nsw0UttFunbUhpFzFtarSd6WJkStVMw5xyVqYTvkwfzuf/5FgtkVoB0729j1rjXwThS7Vio+Mo6DNnvLfahoZ+i/o32lULuJ3NNiz7q6+pyAUkJaFF6JwaM2lUJlV0MlnQn5aTRbEu0SEqHUa0A4AdiGuB1kFXRfVyg5d87+Dg4DL6m2TLAub60ilj7A1Ec4odSAc8X95sHh7+ZRPCFo6Fnp7HfU/fBng/hi10CjCnWnJjsxvDNxWw0NfV6Rv5GgP3I3jGWXumdTD/3cbEOP2ZbOZp69yniG3FQ9z1jD7bnBu9Fc2tKGC2q+uAJOQHBDRiZX1x36o7fWBs7J9ownbtO+n0/qWkvW7UPIfc37WgT6ZGR++EOJyeQDSb9UB+DZ1G6DdLDzyS+b/kBCYGsYgJbSQHuThGKRcw5xdeQf8YdNHsc6ePXrlSYMBuSIAFTGAtQo+VuALo4BX83N190NWZWbynBjhOHsmNfFWLeL6v+ynsA58zDvvAC8j5PkbOcXCMg2PZFk3q8MjI7WAG/Dp9AwP7jdGBOOQkAvlFUB+irtm16I1Zw9YBcpGTGXYmk3kQIC/Cds55l+iMI3jqhjAuaoe+am2Jw5GT3Nbz3CkE12NavmzN5+erJW7046n/CH1RO/RVa8lBLozXk9uqykkGAyRXLWlLv5jyp4RFsG5vGVzpDLnIjTWgnRy2Rr+tDKvRc7Y8AyZq10jj8DqXdnIRNtFZb+t/ZRtXcDiVnzpqx8mPcDWxgARUqx0W1QB9MeUZiNrV4qP+Ehc+BpNgATsTX8ozYKL2NtFYAHc84fG7ndxUPr+AR/iQSns7uSUufAymwDOb2+NjK27lEFocm/EE2WpyIy/Hi66MWuMKJn8RvxIcj87IM5Vh9663ziW36kR0HNenXuxmfaD8JC7tfKbrhFr7LiZCrMjrzTeGx+PmkosrkNzW94ObzwocJ7A1HokLolY+AvkTiD/q1H0cN48c5EL8Crkttsa/AXQVDmutfyku0E7jShx49XqV3MFK8IryDhYVbj7Sj2P2eBxwcXoe8T8idsKKPRcnZw1b+slFTubwUwhktrfnAt7J++jwQtLZcm3sr9LQrjRzz6cfMv9aLvgmnAGvpoaGLxM4mAEaLV7iAzQ3oU0IvD5x9ix3yF2RAAuYAOO2f7PEFWCXZ4C9Pb2UsgDeVnFSpbFK7/IWu7TPTvBqzbGdCHOJQSxiEjt6IyZmxQyEJHv6xyQsYk//moVFsN2zP6fRImjfq7/n/wFDguUQFNEwugAAAABJRU5ErkJggg==');
  }

  div.callout-important.callout-style-default .callout-title {
    background-color: #f7dddc
  }

  div.callout-warning {
    border-left-color: #f0ad4e !important;
  }

  div.callout-warning .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAETklEQVRYCeVWW2gcVRg+58yaTUnizqbipZeX4uWhBEniBaoUX1Ioze52t7sRq6APio9V9MEaoWlVsFasRq0gltaAPuxms8lu0gcviE/FFOstVbSIxgcv6SU7EZqmdc7v9+9mJtNks51NTUH84ed889/PP+cmxP+d5FIbMJmNbpREu4WUkiTtCicKny0l1pIKmBzovF2S+hIJHX8iEu3hZJ5lNZGqyRrGSIQpq15AzF28jgpeY6yk6GVdrfFqdrD6Iw+QlB8g0YS2g7dyQmXM/IDhBhT0UCiRf59lfqmmDvzRt6kByV/m4JjtzuaujMUM2c5Z2d6JdKrRb3K2q6mA+oYVz8JnDdKPmmNthzkAk/lN63sYPgevrguc72aZX/L9C6x09GYyxBgCX4NlvyGUHOKELlm5rXeR1kchuChJt4SSwyddZRXgvwMGvYo4QSlk3/zkHD8UHxwVJA6zjZZqP8v8kK8OWLnIZtLyCAJagYC4rTGW/9Pqj92N/c+LUaAj27movwbi19tk/whRCIE7Q9vyI6yvRpftAKVTdUjOW40X3h5OXsKCdmFcx0xlLJoSuQngnrJe7Kcjm4OMq9FlC7CMmScQANuNvjfP3PjGXDBaUQmbp296S5L4DrpbrHN1T87ZVEZVCzg1FF0Ft+dKrlLukI+/c9ENo+TvlTDbYFvuKPtQ9+l052rXrgKoWkDAFnvh0wTOmYn8R5f4k/jN/fZiCM1tQx9jQQ4ANhqG4hiL0qIFTGViG9DKB7GYzgubnpofgYRwO+DFjh0Zin2m4b/97EDkXkc+f6xYAPX0KK2I/7fUQuwzuwo/L3AkcjugPNixC8cHf0FyPjWlItmLxWw4Ou9YsQCr5fijMGoD/zpdRy95HRysyXA74MWOnscpO4j2y3HAVisw85hX5+AFBRSHt4ShfLFkIMXTqyKFc46xdzQM6XbAi702a7sy04J0+feReMFKp5q9esYLCqAZYw/k14E/xcLLsFElaornTuJB0svMuJINy8xkIYuL+xPAlWRceH6+HX7THJ0djLUom46zREu7tTkxwmf/FdOZ/sh6Q8qvEAiHpm4PJ4a/doJe0gH1t+aHRgCzOvBvJedEK5OFE5jpm4AGP2a8Dxe3gGJ/pAutug9Gp6he92CsSsWBaEcxGx0FHytmIpuqGkOpldqNYQK8cSoXvd+xLxXADw0kf6UkJNFtdo5MOgaLjiQOQHcn+A6h5NuL2s0qsC2LOM75PcF3yr5STuBSAcGG+meA14K/CI21HcS4LBT6tv0QAh8Dr5l93AhZzG5ZJ4VxAqdZUEl9z7WJ4aN+svMvwHHL21UKTd1mqvChH7/Za5xzXBBKrUcB0TQ+Ulgkfbi/H/YT5EptrGzsEK7tR1B7ln9BBwckYfMiuSqklSznIuoIIOM42MQO+QnduCoFCI0bpkzjCjddHPN/F+2Yu+sd9bKNpVwHhbS3LluK/0zgfwD0xYI5dXuzlQAAAABJRU5ErkJggg==');
  }

  div.callout-warning.callout-style-default .callout-title {
    background-color: #fcefdc
  }

  div.callout-tip {
    border-left-color: #02b875 !important;
  }

  div.callout-tip .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAADr0lEQVRYCe1XTWgTQRj9ZjZV8a9SPIkKgj8I1bMHsUWrqYLVg4Ue6v9BwZOxSYsIerFao7UiUryIqJcqgtpimhbBXoSCVxUFe9CTiogUrUp2Pt+3aUI2u5vdNh4dmMzOzHvvezuz8xNFM0mjnbXaNu1MvFWRXkXEyE6aYOYJpdW4IXuA4r0fo8qqSMDBU0v1HJUgVieAXxzCsdE/YJTdFcVIZQNMyhruOMJKXYFoLfIfIvVIMWdsrd+Rpd86ZmyzzjJmLStqRn0v8lzkb4rVIXvnpScOJuAn2ACC65FkPzEdEy4TPWRLJ2h7z4cArXzzaOdKlbOvKKX25Wl00jSnrwVxAg3o4dRxhO13RBSdNvH0xSARv3adTXbBdTf64IWO2vH0LT+cv4GR1DJt+DUItaQogeBX/chhbTBxEiZ6gftlDNXTrvT7co4ub5A6gp9HIcHvzTa46OS5fBeP87Qm0fQkr4FsYgVQ7Qg+ZayaDg9jhg1GkWj8RG6lkeSacrrHgDaxdoBiZPg+NXV/KifMuB6//JmYH4CntVEHy/keA6x4h4CU5oFy8GzrBS18cLJMXcljAKB6INjWsRcuZBWVaS3GDrqB7rdapVIeA+isQ57Eev9eCqzqOa81CY05VLd6SamW2wA2H3SiTbnbSxmzfp7WtKZkqy4mdyAlGx7ennghYf8voqp9cLSgKdqNfa6RdRsAAkPwRuJZNbpByn+RrJi1RXTwdi8RQF6ymDwGMAtZ6TVE+4uoKh+MYkcLsT0Hk8eAienbiGdjJHZTpmNjlbFJNKDVAp2fJlYju6IreQxQ08UJDNYdoLSl6AadO+fFuCQqVMB1NJwPm69T04Wv5WhfcWyfXQB+wXRs1pt+nCknRa0LVzSA/2B+a9+zQJadb7IyyV24YAxKp2Jqs3emZTuNnKxsah+uabKbMk7CbTgJx/zIgQYErIeTKRQ9yD9wxVof5YolPHqaWo7TD6tJlh7jQnK5z2n3+fGdggIOx2kaa2YI9QWarc5Ce1ipNWMKeSG4DysFF52KBmTNMmn5HqCFkwy34rDg05gDwgH3bBi+sgFhN/e8QvRn8kbamCOhgrZ9GJhFDgfcMHzFb6BAtjKpFhzTjwv1KCVuxHvCbsSiEz4CANnj84cwHdFXAbAOJ4LTSAawGWFn5tDhLMYz6nWeU2wJfIhmIJBefcd/A5FWQWGgrWzyORZ3Q6HuV+Jf0Bj+BTX69fm1zWgK7By1YTXchFDORywnfQ7GpzOo6S+qECrsx2ifVQAAAABJRU5ErkJggg==');
  }

  div.callout-tip.callout-style-default .callout-title {
    background-color: #ccf1e3
  }

  div.callout-caution {
    border-left-color: #fd7e14 !important;
  }

  div.callout-caution .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAACV0lEQVRYCdVWzWoUQRCuqp2ICBLJXgITZL1EfQDBW/bkzUMUD7klD+ATSHBEfAIfQO+iXsWDxJsHL96EHAwhgzlkg8nBg25XWb0zIb0zs9muYYWkoKeru+vn664fBqElyZNuyh167NXJ8Ut8McjbmEraKHkd7uAnAFku+VWdb3reSmRV8PKSLfZ0Gjn3a6Xlcq9YGb6tADjn+lUfTXtVmaZ1KwBIvFI11rRXlWlatwIAAv2asaa9mlB9wwygiDX26qaw1yYPzFXg2N1GgG0FMF8Oj+VIx7E/03lHx8UhvYyNZLN7BwSPgekXXLribw7w5/c8EF+DBK5idvDVYtEEwMeYefjjLAdEyQ3M9nfOkgnPTEkYU+sxMq0BxNR6jExrAI31H1rzvLEfRIdgcv1XEdj6QTQAS2wtstEALLG1yEZ3QhH6oDX7ExBSFEkFINXH98NTrme5IOaaA7kIfiu2L8A3qhH9zRbukdCqdsA98TdElyeMe5BI8Rs2xHRIsoTSSVFfCFCWGPn9XHb4cdobRIWABNf0add9jakDjQJpJ1bTXOJXnnRXHRf+dNL1ZV1MBRCXhMbaHqGI1JkKIL7+i8uffuP6wVQAzO7+qVEbF6NbS0LJureYcWXUUhH66nLR5rYmva+2tjRFtojkM2aD76HEGAD3tPtKM309FJg5j/K682ywcWJ3PASCcycH/22u+Bh7Aa0ehM2Fu4z0SAE81HF9RkB21c5bEn4Dzw+/qNOyXr3DCTQDMBOdhi4nAgiFDGCinIa2owCEChUwD8qzd03PG+qdW/4fDzjUMcE1ZpIAAAAASUVORK5CYII=');
  }

  div.callout-caution.callout-style-default .callout-title {
    background-color: #ffe5d0
  }

  </style>
  <style type="text/css">
    .reveal div.sourceCode {
      margin: 0;
      overflow: auto;
    }
    .reveal div.hanging-indent {
      margin-left: 1em;
      text-indent: -1em;
    }
    .reveal .slide:not(.center) {
      height: 100%;
    }
    .reveal .slide.scrollable {
      overflow-y: auto;
    }
    .reveal .footnotes {
      height: 100%;
      overflow-y: auto;
    }
    .reveal .slide .absolute {
      position: absolute;
      display: block;
    }
    .reveal .footnotes ol {
      counter-reset: ol;
      list-style-type: none; 
      margin-left: 0;
    }
    .reveal .footnotes ol li:before {
      counter-increment: ol;
      content: counter(ol) ". "; 
    }
    .reveal .footnotes ol li > p:first-child {
      display: inline-block;
    }
    .reveal .slide ul,
    .reveal .slide ol {
      margin-bottom: 0.5em;
    }
    .reveal .slide ul li,
    .reveal .slide ol li {
      margin-top: 0.4em;
      margin-bottom: 0.2em;
    }
    .reveal .slide ul[role="tablist"] li {
      margin-bottom: 0;
    }
    .reveal .slide ul li > *:first-child,
    .reveal .slide ol li > *:first-child {
      margin-block-start: 0;
    }
    .reveal .slide ul li > *:last-child,
    .reveal .slide ol li > *:last-child {
      margin-block-end: 0;
    }
    .reveal .slide .columns:nth-child(3) {
      margin-block-start: 0.8em;
    }
    .reveal blockquote {
      box-shadow: none;
    }
    .reveal .tippy-content>* {
      margin-top: 0.2em;
      margin-bottom: 0.7em;
    }
    .reveal .tippy-content>*:last-child {
      margin-bottom: 0.2em;
    }
    .reveal .slide > img.stretch.quarto-figure-center,
    .reveal .slide > img.r-stretch.quarto-figure-center {
      display: block;
      margin-left: auto;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-left,
    .reveal .slide > img.r-stretch.quarto-figure-left  {
      display: block;
      margin-left: 0;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-right,
    .reveal .slide > img.r-stretch.quarto-figure-right  {
      display: block;
      margin-left: auto;
      margin-right: 0; 
    }
  </style>
</head>
<body class="quarto-light">
  <div class="reveal">
    <div class="slides">

<section id="title-slide" class="quarto-title-block center">
  <h1 class="title">A Practical Primer on Bayesian Statistics</h1>

<div class="quarto-title-authors">
<div class="quarto-title-author">
<div class="quarto-title-author-name">
Sarah Urbut, MD PhD 
</div>
</div>
</div>

  <p class="date">2025-04-17</p>
</section>
<section id="overview" class="slide level2">
<h2>Overview</h2>
<p>In this seminar, we’ll (attempt to) cover key Bayesian concepts critical for modern genomics:</p>
<ol type="1">
<li><strong>P-values vs.&nbsp;Posterior Probabilities</strong>: Why Bayesian thinking helps avoid misinterpretations<br>
</li>
<li><strong>Conjugate Models</strong>: Elegant solutions for population genetic inference<br>
</li>
<li><strong>Mixture Models</strong>: Powerful tools for complex genomic data<br>
</li>
<li><strong>Bayesian Clinical &amp; Adaptive Designs</strong>: Learning and adapting as data accumulates</li>
</ol>
</section>
<section>
<section id="the-p-value-paradox" class="title-slide slide level1 center">
<h1>1. The P-value Paradox</h1>

</section>
<section id="lindleys-paradox" class="slide level2">
<h2>Lindley’s Paradox</h2>
<blockquote>
<p>“A result that is statistically significant may not be scientifically significant” - Dennis Lindley (1957)</p>
</blockquote>
<div class="callout callout-note callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<p><strong>Note</strong></p>
</div>
<div class="callout-content">
<p>Lindley’s 1957 paper demonstrated how p-values and Bayes factors can lead to contradictory conclusions <span class="citation" data-cites="lindley1957">[@lindley1957]</span></p>
</div>
</div>
</div>
</section>
<section id="the-evidence-paradox" class="slide level2">
<h2>The Evidence Paradox</h2>
<p>Sometimes a result can be unlikely under the null hypothesis but even more unlikely under the alternative!</p>
</section>
<section id="example" class="slide level2">
<h2>Example</h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href=""></a><span class="co"># Create example of evidence paradox</span></span>
<span id="cb1-2"><a href=""></a>z <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="sc">-</span><span class="dv">5</span>, <span class="dv">5</span>, <span class="at">length.out =</span> <span class="dv">1000</span>)</span>
<span id="cb1-3"><a href=""></a>null_density <span class="ot">&lt;-</span> <span class="fu">dnorm</span>(z, <span class="dv">0</span>, <span class="dv">1</span>)</span>
<span id="cb1-4"><a href=""></a>alt_density <span class="ot">&lt;-</span> <span class="fu">dnorm</span>(z, <span class="dv">5</span>, <span class="dv">1</span>)  <span class="co"># Alternative with larger variance</span></span>
<span id="cb1-5"><a href=""></a></span>
<span id="cb1-6"><a href=""></a><span class="co"># Create example point</span></span>
<span id="cb1-7"><a href=""></a>z_obs <span class="ot">&lt;-</span> <span class="fl">2.5</span>  <span class="co"># A "significant" observation</span></span>
<span id="cb1-8"><a href=""></a>null_prob <span class="ot">&lt;-</span> <span class="fu">dnorm</span>(z_obs, <span class="dv">0</span>, <span class="dv">1</span>)</span>
<span id="cb1-9"><a href=""></a>alt_prob <span class="ot">&lt;-</span> <span class="fu">dnorm</span>(z_obs, <span class="dv">6</span>, <span class="dv">1</span>)</span>
<span id="cb1-10"><a href=""></a></span>
<span id="cb1-11"><a href=""></a><span class="co"># Create data frame for plotting</span></span>
<span id="cb1-12"><a href=""></a>paradox_data <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb1-13"><a href=""></a>    <span class="at">z =</span> z,</span>
<span id="cb1-14"><a href=""></a>    <span class="at">Null =</span> null_density,</span>
<span id="cb1-15"><a href=""></a>    <span class="at">Alternative =</span> alt_density</span>
<span id="cb1-16"><a href=""></a>)</span>
<span id="cb1-17"><a href=""></a></span>
<span id="cb1-18"><a href=""></a><span class="co"># Plot</span></span>
<span id="cb1-19"><a href=""></a><span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb1-20"><a href=""></a>    <span class="fu">geom_line</span>(<span class="at">data =</span> paradox_data, <span class="fu">aes</span>(<span class="at">x =</span> z, <span class="at">y =</span> Null, <span class="at">color =</span> <span class="st">"Null"</span>), <span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb1-21"><a href=""></a>    <span class="fu">geom_line</span>(<span class="at">data =</span> paradox_data, <span class="fu">aes</span>(<span class="at">x =</span> z, <span class="at">y =</span> Alternative, <span class="at">color =</span> <span class="st">"Alternative"</span>), <span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb1-22"><a href=""></a>    <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> z_obs, <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb1-23"><a href=""></a>    <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x =</span> z_obs, <span class="at">y =</span> null_prob), <span class="at">color =</span> <span class="st">"red"</span>, <span class="at">size =</span> <span class="dv">5</span>) <span class="sc">+</span></span>
<span id="cb1-24"><a href=""></a>    <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x =</span> z_obs, <span class="at">y =</span> alt_prob), <span class="at">color =</span> <span class="st">"blue"</span>, <span class="at">size =</span> <span class="dv">5</span>) <span class="sc">+</span></span>
<span id="cb1-25"><a href=""></a>    <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"Null"</span> <span class="ot">=</span> <span class="st">"red"</span>, <span class="st">"Alternative"</span> <span class="ot">=</span> <span class="st">"blue"</span>)) <span class="sc">+</span></span>
<span id="cb1-26"><a href=""></a>    <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> z_obs <span class="sc">-</span> <span class="fl">1.5</span>, <span class="at">y =</span> null_prob<span class="fl">+0.1</span>, </span>
<span id="cb1-27"><a href=""></a>             <span class="at">label =</span> <span class="fu">sprintf</span>(<span class="st">"p(z|H_0) = %.3f"</span>, null_prob), <span class="at">size =</span> <span class="dv">4</span>) <span class="sc">+</span></span>
<span id="cb1-28"><a href=""></a>    <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> z_obs <span class="sc">+</span><span class="fl">1.5</span>, <span class="at">y =</span> alt_prob<span class="fl">+0.1</span>, </span>
<span id="cb1-29"><a href=""></a>             <span class="at">label =</span> <span class="fu">sprintf</span>(<span class="st">"p(z|H_1) = %.3f"</span>, alt_prob), <span class="at">size =</span><span class="dv">4</span> ) <span class="sc">+</span></span>
<span id="cb1-30"><a href=""></a>    <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"The Evidence Paradox"</span>,</span>
<span id="cb1-31"><a href=""></a>         <span class="at">subtitle =</span> <span class="st">"A result can be unlikely under H0 but even more unlikely under H1"</span>,</span>
<span id="cb1-32"><a href=""></a>         <span class="at">x =</span> <span class="st">"Z-score"</span>,</span>
<span id="cb1-33"><a href=""></a>         <span class="at">y =</span> <span class="st">"Density"</span>) <span class="sc">+</span></span>
<span id="cb1-34"><a href=""></a>    <span class="fu">theme_minimal</span>(<span class="at">base_size =</span> <span class="dv">20</span>) <span class="sc">+</span></span>
<span id="cb1-35"><a href=""></a>    <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"bottom"</span>,</span>
<span id="cb1-36"><a href=""></a>          <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">16</span>),</span>
<span id="cb1-37"><a href=""></a>          <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">18</span>),</span>
<span id="cb1-38"><a href=""></a>          <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">18</span>),</span>
<span id="cb1-39"><a href=""></a>          <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">18</span>),</span>
<span id="cb1-40"><a href=""></a>          <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">22</span>),</span>
<span id="cb1-41"><a href=""></a>          <span class="at">plot.subtitle =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>

</div>
<img data-src="mpg6_files/figure-revealjs/unnamed-chunk-1-1.png" width="960" class="r-stretch"><aside class="notes">
<p>This paradox shows why we need to consider both the null and alternative hypotheses: - A small p-value only tells us the result is unlikely under <span class="math inline">\(H_0\)</span> - But for real evidence, we need the result to be more likely under <span class="math inline">\(H_1\)</span> than <span class="math inline">\(H_0\)</span> - This is why Bayes factors (likelihood ratios) are more informative than p-values</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="the-p-value-fallacy" class="slide level2">
<h2>The P-value Fallacy</h2>
<p>What we want: P(Hypothesis|Data)<br>
What we get: P(Data|Null Hypothesis)</p>
<p>Question for the class: What is a P value?</p>
<p>Prepare to be amazed: Frequentists are all Bayesians! (under some qualifications)</p>
</section>
<section id="probabilistic-interpretation-of-estimates" class="slide level2">
<h2>Probabilistic Interpretation of Estimates</h2>
<p>In the Bayesian framework:</p>
<ul>
<li><strong>Parameters are random variables</strong> with distributions, not fixed values</li>
<li><strong>Uncertainty is represented directly</strong> through probability distributions</li>
<li><strong>All evidence is integrated coherently</strong> within a probability framework</li>
<li><strong>Natural quantification of uncertainty</strong> <em>without</em> hypothetical repeated sampling</li>
<li><strong>Interpretation is direct and intuitive</strong> for researchers and clinicians</li>
</ul>
</section>
<section id="on-the-surface-the-same-but-whos-running" class="slide level2">
<h2>On the surface the same but … who’s running?</h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href=""></a><span class="co"># Create example data for frequentist vs Bayesian comparison</span></span>
<span id="cb2-2"><a href=""></a>x <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="fl">0.2</span>, <span class="fl">0.8</span>, <span class="at">length =</span> <span class="dv">1000</span>)</span>
<span id="cb2-3"><a href=""></a></span>
<span id="cb2-4"><a href=""></a><span class="co"># Bayesian posterior (beta distribution)</span></span>
<span id="cb2-5"><a href=""></a>post_y <span class="ot">&lt;-</span> <span class="fu">dbeta</span>(x, <span class="dv">70</span>, <span class="dv">30</span>)</span>
<span id="cb2-6"><a href=""></a></span>
<span id="cb2-7"><a href=""></a><span class="co"># Create data frame for plotting</span></span>
<span id="cb2-8"><a href=""></a>plot_data <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb2-9"><a href=""></a>  <span class="at">x =</span> x,</span>
<span id="cb2-10"><a href=""></a>  <span class="at">y =</span> post_y</span>
<span id="cb2-11"><a href=""></a>)</span>
<span id="cb2-12"><a href=""></a></span>
<span id="cb2-13"><a href=""></a><span class="co"># Point estimate</span></span>
<span id="cb2-14"><a href=""></a>point_est <span class="ot">&lt;-</span> <span class="fl">0.7</span></span>
<span id="cb2-15"><a href=""></a>ci_lower <span class="ot">&lt;-</span> <span class="fl">0.61</span></span>
<span id="cb2-16"><a href=""></a>ci_upper <span class="ot">&lt;-</span> <span class="fl">0.79</span></span>
<span id="cb2-17"><a href=""></a>cred_lower <span class="ot">&lt;-</span> <span class="fu">qbeta</span>(<span class="fl">0.025</span>, <span class="dv">70</span>, <span class="dv">30</span>)</span>
<span id="cb2-18"><a href=""></a>cred_upper <span class="ot">&lt;-</span> <span class="fu">qbeta</span>(<span class="fl">0.975</span>, <span class="dv">70</span>, <span class="dv">30</span>)</span>
<span id="cb2-19"><a href=""></a></span>
<span id="cb2-20"><a href=""></a><span class="co"># Get maximum y value for annotation positioning</span></span>
<span id="cb2-21"><a href=""></a>max_y_val <span class="ot">&lt;-</span> <span class="fu">max</span>(plot_data<span class="sc">$</span>y)</span>
<span id="cb2-22"><a href=""></a></span>
<span id="cb2-23"><a href=""></a><span class="co"># Plot</span></span>
<span id="cb2-24"><a href=""></a><span class="fu">ggplot</span>(plot_data, <span class="fu">aes</span>(<span class="at">x =</span> x, <span class="at">y =</span> y)) <span class="sc">+</span></span>
<span id="cb2-25"><a href=""></a>  <span class="fu">geom_line</span>(<span class="at">size =</span> <span class="fl">1.2</span>, <span class="at">color =</span> <span class="st">"darkblue"</span>) <span class="sc">+</span></span>
<span id="cb2-26"><a href=""></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> point_est, <span class="at">color =</span> <span class="st">"red"</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb2-27"><a href=""></a>  <span class="fu">geom_segment</span>(<span class="fu">aes</span>(<span class="at">x =</span> ci_lower, <span class="at">xend =</span> ci_upper, <span class="at">y =</span> max_y_val<span class="sc">/</span><span class="dv">10</span>, <span class="at">yend =</span> max_y_val<span class="sc">/</span><span class="dv">10</span>), </span>
<span id="cb2-28"><a href=""></a>               <span class="at">color =</span> <span class="st">"darkred"</span>, <span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb2-29"><a href=""></a>  <span class="fu">geom_segment</span>(<span class="fu">aes</span>(<span class="at">x =</span> cred_lower, <span class="at">xend =</span> cred_upper, <span class="at">y =</span> max_y_val<span class="sc">/</span><span class="dv">5</span>, <span class="at">yend =</span> max_y_val<span class="sc">/</span><span class="dv">5</span>), </span>
<span id="cb2-30"><a href=""></a>               <span class="at">color =</span> <span class="st">"darkblue"</span>, <span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb2-31"><a href=""></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> point_est <span class="sc">+</span> <span class="fl">0.05</span>, <span class="at">y =</span> max_y_val<span class="sc">/</span><span class="fl">1.5</span>, </span>
<span id="cb2-32"><a href=""></a>           <span class="at">label =</span> <span class="st">"Point Estimate"</span>, <span class="at">color =</span> <span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb2-33"><a href=""></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> ci_lower <span class="sc">-</span> <span class="fl">0.05</span>, <span class="at">y =</span> max_y_val<span class="sc">/</span><span class="dv">10</span>, </span>
<span id="cb2-34"><a href=""></a>           <span class="at">label =</span> <span class="st">"95% CI"</span>, <span class="at">color =</span> <span class="st">"darkred"</span>) <span class="sc">+</span></span>
<span id="cb2-35"><a href=""></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> cred_upper <span class="sc">+</span> <span class="fl">0.05</span>, <span class="at">y =</span> max_y_val<span class="sc">/</span><span class="dv">5</span>, </span>
<span id="cb2-36"><a href=""></a>           <span class="at">label =</span> <span class="st">"95% Credible Interval"</span>, <span class="at">color =</span> <span class="st">"darkblue"</span>) <span class="sc">+</span></span>
<span id="cb2-37"><a href=""></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Bayesian Posterior Distribution"</span>,</span>
<span id="cb2-38"><a href=""></a>       <span class="at">subtitle =</span> <span class="st">"Directly interpretable probability statements about parameters"</span>,</span>
<span id="cb2-39"><a href=""></a>       <span class="at">x =</span> <span class="st">"Parameter Value"</span>, </span>
<span id="cb2-40"><a href=""></a>       <span class="at">y =</span> <span class="st">"Posterior Density"</span>) <span class="sc">+</span></span>
<span id="cb2-41"><a href=""></a>  <span class="fu">theme_minimal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>

</div>
<img data-src="mpg6_files/figure-revealjs/unnamed-chunk-2-1.png" width="960" class="r-stretch"><aside class="notes">
<table class="caption-top">
<colgroup>
<col style="width: 47%">
<col style="width: 52%">
</colgroup>
<thead>
<tr class="header">
<th>Bayesian Credible Interval</th>
<th>Frequentist Confidence Interval</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>“95% probability the parameter is between a and b”</td>
<td>“If we repeated the experiment many times, 95% of intervals constructed would contain the true parameter”</td>
</tr>
<tr class="even">
<td>Directly interpretable as probability statement about the parameter</td>
<td>Cannot be interpreted as probability statement about the parameter</td>
</tr>
<tr class="odd">
<td>Incorporates prior information</td>
<td>No mechanism to incorporate prior information (why would you run?)</td>
</tr>
<tr class="even">
<td>Can be asymmetric, reflecting asymmetric uncertainty</td>
<td>Typically symmetric by construction</td>
</tr>
<tr class="odd">
<td>Conditioning on the observed data</td>
<td>Based on hypothetical repeated (theoretical) sampling</td>
</tr>
</tbody>
</table>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="bayes-theorem---the-core-idea" class="slide level2">
<h2>Bayes’ Theorem - The Core Idea</h2>
<p><span class="math display">\[P(H|D) = \frac{P(D|H) \times P(H)}{P(D)}\]</span></p>
<p>Where:</p>
<ul>
<li><span class="math inline">\(P(H|D)\)</span> is the <strong>posterior probability</strong> - what we want to know</li>
<li><span class="math inline">\(P(D|H)\)</span> is the <strong>likelihood</strong> - how probable the data is under our hypothesis</li>
<li><span class="math inline">\(P(H)\)</span> is the <strong>prior probability</strong> - what we knew before</li>
<li><span class="math inline">\(P(D)\)</span> is the <strong>evidence</strong> - a normalizing constant</li>
</ul>
<p>Simply: <strong>Posterior ∝ Likelihood × Prior</strong></p>
</section>
<section id="bayesian-updating-visual-intuition" class="slide level2">
<h2>Bayesian Updating: Visual Intuition</h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href=""></a><span class="co"># Create data for three distributions</span></span>
<span id="cb3-2"><a href=""></a>x <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="fl">0.001</span>, <span class="fl">0.999</span>, <span class="at">length =</span> <span class="dv">1000</span>)</span>
<span id="cb3-3"><a href=""></a>prior <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">x =</span> x, <span class="at">y =</span> <span class="fu">dbeta</span>(x, <span class="dv">2</span>, <span class="dv">3</span>), <span class="at">Distribution =</span> <span class="st">"Prior: Beta(2,3)"</span>)</span>
<span id="cb3-4"><a href=""></a>likelihood <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">x =</span> x, <span class="at">y =</span> <span class="fu">dbeta</span>(x, <span class="dv">7</span>, <span class="dv">3</span>), <span class="at">Distribution =</span> <span class="st">"Likelihood (Data)"</span>)</span>
<span id="cb3-5"><a href=""></a>posterior <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">x =</span> x, <span class="at">y =</span> <span class="fu">dbeta</span>(x, <span class="dv">9</span>, <span class="dv">6</span>), <span class="at">Distribution =</span> <span class="st">"Posterior: Beta(9,6)"</span>)</span>
<span id="cb3-6"><a href=""></a></span>
<span id="cb3-7"><a href=""></a><span class="co"># Combine data</span></span>
<span id="cb3-8"><a href=""></a>all_distributions <span class="ot">&lt;-</span> <span class="fu">rbind</span>(prior, likelihood, posterior)</span>
<span id="cb3-9"><a href=""></a></span>
<span id="cb3-10"><a href=""></a><span class="co"># Plot all three distributions</span></span>
<span id="cb3-11"><a href=""></a><span class="fu">ggplot</span>(all_distributions, <span class="fu">aes</span>(<span class="at">x =</span> x, <span class="at">y =</span> y, <span class="at">color =</span> Distribution)) <span class="sc">+</span></span>
<span id="cb3-12"><a href=""></a>  <span class="fu">geom_line</span>(<span class="at">size =</span> <span class="fl">1.2</span>) <span class="sc">+</span></span>
<span id="cb3-13"><a href=""></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"darkgreen"</span>, <span class="st">"purple"</span>, <span class="st">"red"</span>)) <span class="sc">+</span></span>
<span id="cb3-14"><a href=""></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Bayesian Updating of Allele Frequency Estimate"</span>,</span>
<span id="cb3-15"><a href=""></a>       <span class="at">subtitle =</span> <span class="st">"Combining prior knowledge with new data"</span>,</span>
<span id="cb3-16"><a href=""></a>       <span class="at">x =</span> <span class="st">"Allele Frequency"</span>, </span>
<span id="cb3-17"><a href=""></a>       <span class="at">y =</span> <span class="st">"Density"</span>) <span class="sc">+</span></span>
<span id="cb3-18"><a href=""></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"bottom"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>

</div>
<img data-src="mpg6_files/figure-revealjs/bayesian-updating-1.png" width="960" class="r-stretch"></section>
<section id="p-values-vs.-bayes-factors-definitions" class="slide level2">
<h2>P-values vs.&nbsp;Bayes Factors: Definitions</h2>
<p><strong>P-value</strong>:<br>
- <span class="math inline">\(P(data|H_0)\)</span> - probability of data given null hypothesis<br>
- Measures compatibility of data with null hypothesis<br>
- Does not directly measure evidence for alternative</p>
<p><strong>Bayes Factor</strong>:<br>
- <span class="math inline">\(BF_{10} = \frac{P(data|H_1)}{P(data|H_0)}\)</span> - ratio of likelihoods</p>
<p>- <span class="math inline">\(BF_{01} = \frac{P(data|H_0)}{P(data|H_1)}\)</span> - ratio of likelihoods<br>
- Directly compares evidence for alternative vs.&nbsp;null OR null vs alternative<br>
- Tells you how much to update your beliefs</p>
</section>
<section id="the-p-value-fallacy-1" class="slide level2">
<h2>The P-value Fallacy</h2>
<p><strong>Scenario</strong>: Testing a SNP for disease association</p>
<p><strong>Traditional approach</strong>:<br>
- Obtain p = 0.001<br>
- Declare “significant association”<br>
- Publish result</p>
<p><strong>The fallacy</strong>:<br>
- p = 0.001 means “1 in 1000 chance of seeing this data if no association exists”<br>
- NOT “999 in 1000 chance the association is real”</p>
</section>
<section id="visualize" class="slide level2">
<h2>Visualize</h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href=""></a><span class="co"># Parameters</span></span>
<span id="cb4-2"><a href=""></a>prior_prob <span class="ot">&lt;-</span> <span class="dv">1</span><span class="sc">/</span><span class="dv">1000</span>  <span class="co"># Prior probability of true association</span></span>
<span id="cb4-3"><a href=""></a>p_value <span class="ot">&lt;-</span> <span class="fl">0.001</span>      <span class="co"># Observed p-value</span></span>
<span id="cb4-4"><a href=""></a></span>
<span id="cb4-5"><a href=""></a><span class="co"># Convert p-value to z-score</span></span>
<span id="cb4-6"><a href=""></a>z <span class="ot">&lt;-</span> <span class="fu">qnorm</span>(p_value<span class="sc">/</span><span class="dv">2</span>, <span class="at">lower.tail =</span> <span class="cn">FALSE</span>)</span>
<span id="cb4-7"><a href=""></a></span>
<span id="cb4-8"><a href=""></a><span class="co"># Calculate Bayes factor (approximate)</span></span>
<span id="cb4-9"><a href=""></a>bf <span class="ot">&lt;-</span> <span class="fu">exp</span>(z<span class="sc">^</span><span class="dv">2</span><span class="sc">/</span><span class="dv">2</span>)</span>
<span id="cb4-10"><a href=""></a></span>
<span id="cb4-11"><a href=""></a><span class="co"># Calculate posterior probability</span></span>
<span id="cb4-12"><a href=""></a>posterior <span class="ot">&lt;-</span> (prior_prob <span class="sc">*</span> bf) <span class="sc">/</span> (prior_prob <span class="sc">*</span> bf <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">-</span> prior_prob))</span>
<span id="cb4-13"><a href=""></a></span>
<span id="cb4-14"><a href=""></a><span class="co"># Create data for visualization</span></span>
<span id="cb4-15"><a href=""></a>df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb4-16"><a href=""></a>  <span class="at">Stage =</span> <span class="fu">factor</span>(<span class="fu">c</span>(<span class="st">"Prior"</span>, <span class="st">"p-value"</span>, <span class="st">"Posterior"</span>), </span>
<span id="cb4-17"><a href=""></a>                <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"Prior"</span>, <span class="st">"p-value"</span>, <span class="st">"Posterior"</span>)),</span>
<span id="cb4-18"><a href=""></a>  <span class="at">Probability =</span> <span class="fu">c</span>(prior_prob, <span class="dv">1</span><span class="sc">-</span>p_value, posterior),</span>
<span id="cb4-19"><a href=""></a>  <span class="at">Label =</span> <span class="fu">c</span>(<span class="fu">paste0</span>(<span class="fu">round</span>(prior_prob<span class="sc">*</span><span class="dv">100</span>, <span class="dv">2</span>), <span class="st">"%"</span>), </span>
<span id="cb4-20"><a href=""></a>           <span class="fu">paste0</span>(<span class="fu">round</span>((<span class="dv">1</span><span class="sc">-</span>p_value)<span class="sc">*</span><span class="dv">100</span>, <span class="dv">1</span>), <span class="st">"%"</span>), </span>
<span id="cb4-21"><a href=""></a>           <span class="fu">paste0</span>(<span class="fu">round</span>(posterior<span class="sc">*</span><span class="dv">100</span>, <span class="dv">1</span>), <span class="st">"%"</span>))</span>
<span id="cb4-22"><a href=""></a>)</span>
<span id="cb4-23"><a href=""></a></span>
<span id="cb4-24"><a href=""></a><span class="co"># Create plot</span></span>
<span id="cb4-25"><a href=""></a><span class="fu">ggplot</span>(df, <span class="fu">aes</span>(<span class="at">x =</span> Stage, <span class="at">y =</span> Probability)) <span class="sc">+</span></span>
<span id="cb4-26"><a href=""></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">fill =</span> <span class="fu">c</span>(<span class="st">"lightblue"</span>, <span class="st">"orange"</span>, <span class="st">"darkgreen"</span>), <span class="at">width =</span> <span class="fl">0.6</span>) <span class="sc">+</span></span>
<span id="cb4-27"><a href=""></a>  <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">label =</span> Label), <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.5</span>, <span class="at">size =</span> <span class="dv">5</span>) <span class="sc">+</span></span>
<span id="cb4-28"><a href=""></a>  <span class="fu">ylim</span>(<span class="dv">0</span>, <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb4-29"><a href=""></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"The p-value Fallacy"</span>,</span>
<span id="cb4-30"><a href=""></a>       <span class="at">subtitle =</span> <span class="st">"p = 0.001 doesn't mean 99.9% chance of true association (^app)"</span>,</span>
<span id="cb4-31"><a href=""></a>       <span class="at">x =</span> <span class="st">""</span>, <span class="at">y =</span> <span class="st">"Probability"</span>) <span class="sc">+</span></span>
<span id="cb4-32"><a href=""></a>  <span class="fu">theme_minimal</span>(<span class="at">base_size =</span> <span class="dv">14</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>

</div>
<img data-src="mpg6_files/figure-revealjs/unnamed-chunk-3-1.png" width="960" class="r-stretch"><p><strong>Key insight</strong>: With a realistic prior of 1/1000, a “significant” p-value of 0.001 only gives ~18% posterior probability of true association! (app interlude) <a href="https://surbut.shinyapps.io/shinypval/">Shiny P!</a></p>
</section>
<section id="the-mathematical-connection-for-b01" class="slide level2">
<h2>The Mathematical Connection … for B(01)</h2>
<p>Under certain conditions, p-values can be converted to minimum Bayes factors, see <a href="https://hannig.cloudapps.unc.edu/STOR654/handouts/SellkeBayarriBerger2001.pdf">Sellke, Bayarri, and Berger</a>.</p>
<p><span class="math display">\[BF_{min} ≈ -e \times p \times \log(p)\]</span></p>
<p>Meaning even the most favorable interpretation of a p-value provides less evidence than typically assumed:</p>
<table class="caption-top">
<thead>
<tr class="header">
<th>p-value</th>
<th>Minimum Bayes Factor</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>0.05</td>
<td>0.37</td>
</tr>
<tr class="even">
<td>0.01</td>
<td>0.084</td>
</tr>
<tr class="odd">
<td>0.001</td>
<td>0.0083</td>
</tr>
</tbody>
</table>
<div class="callout callout-none no-icon callout-style-simple">
<div class="callout-body">
<div class="callout-content">
<p>What is the Minimum Bayes Factor? This formula (derived by Sellke, Bayarri, and Berger) represents the <strong>smallest possible Bayes factor</strong> that could correspond to a given p-value, regardless of the specific alternative hypothesis being tested (i.e p(D|H0)/p(D/H1))</p>
</div>
</div>
</div>
<aside class="notes">
<p>Why “Minimum”?</p>
<p>It’s called “minimum” because:</p>
<ul>
<li>It assumes the most favorable conditions for the alternative hypothesis</li>
<li>It represents the strongest possible evidence against the null that could be derived from a p-value</li>
<li>It’s the smallest value the Bayes factor could take (stronger evidence against null = smaller Bayes factor)</li>
<li></li>
</ul>
<p>The MBF represents the ratio of likelihoods:<br>
<span class="math inline">\(\text{MBF} = \frac{P(\text{data}|H_0)}{P(\text{data}|H_1)}\)</span></p>
<p>For example, with p = 0.05: MBF = 0.37 This means the data are at most 1/0.37 ≈ 2.7 times more likely under the alternative than the null<br>
Even with the most optimistic assumptions, the evidence against the null is modest</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
<!-- P-values systematically overstate the evidence against the null hypothesis. -->
<!-- When converted to the Bayes factor scale, even a seemingly impressive -->
<!-- p-value often translates to much more moderate evidence against the null -->
<!-- hypothesis than most researchers would expect. -->
<!-- ## P-values vs. Bayes Factors in Genomics -->
<!-- ```{r} -->
<!-- # Create data -->
<!-- p_values <- 10^seq(-8, -2, 0.5) -->
<!-- log10_p <- -log10(p_values) -->
<!-- min_bf <- -exp(1) * p_values * log(p_values) -->
<!-- log10_min_bf <- -log10(min_bf) -->
<!-- # Create plot data -->
<!-- plot_data <- data.frame( -->
<!--   log10_p = log10_p, -->
<!--   log10_min_bf = log10_min_bf -->
<!-- ) -->
<!-- # Plot -->
<!-- ggplot(plot_data, aes(x = log10_p, y = log10_min_bf)) + -->
<!--   geom_line(size = 1.5, color = "blue") + -->
<!--   geom_abline(slope = 1, intercept = 0, linetype = "dashed", color = "red") + -->
<!--   geom_vline(xintercept = -log10(5e-8), linetype = "dotted", color = "darkgreen") + -->
<!--   annotate("text", x = 8, y = 4, label = "GWAS significance\nthreshold", color = "darkgreen") + -->
<!--   labs(title = "Strength of Evidence: P-values vs. Bayes Factors", -->
<!--        x = "-log10(p-value)", -->
<!--        y = "-log10(Minimum Bayes Factor)") + -->
<!--   theme_minimal(base_size = 14) -->
<!-- ``` -->
<!-- The GWAS significance threshold of p \< 5×10⁻⁸ corresponds to much -->
<!-- stronger evidence than p = 0.05. -->
<!-- ::: callout-note -->
<!-- Berger and Sellke (1987) showed that p-values systematically overstate -->
<!-- evidence against the null [@berger1987] -->
<!-- ::: -->
</section>
<section id="interpreting-bayes-factors-..-back-to-bf-10" class="slide level2">
<h2>Interpreting Bayes Factors .. back to BF 10</h2>
<p>Bayes factors have a natural interpretation:</p>
<table class="caption-top">
<thead>
<tr class="header">
<th>Bayes Factor (<span class="math inline">\(BF_{10}\)</span>)</th>
<th>Evidence for H1</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>1 - 3</td>
<td>Barely worth mentioning</td>
</tr>
<tr class="even">
<td>3 - 10</td>
<td>Substantial</td>
</tr>
<tr class="odd">
<td>10 - 30</td>
<td>Strong</td>
</tr>
<tr class="even">
<td>30 - 100</td>
<td>Very strong</td>
</tr>
<tr class="odd">
<td>&gt; 100</td>
<td>Extreme</td>
</tr>
</tbody>
</table>
<p>A <span class="math inline">\(BF_{10} = 10\)</span> means the data are 10 times more likely under H1 than H0.</p>
</section>
<section id="from-bayes-factor-to-posterior-probability" class="slide level2">
<h2>From Bayes Factor to Posterior Probability</h2>
<p>Bayes’ theorem connects all the pieces:</p>
<p><span class="math display">\[P(H_1|data) = \frac{P(data|H_1)P(H_1)}{P(data|H_1)P(H_1) + P(data|H_0)P(H_0)}\]</span></p>
<p>This can be rewritten using the Bayes factor BF10:</p>
<p><span class="math display">\[P(H_1|data) = \frac{BF_{10} \times P(H_1)}{BF_{10} \times P(H_1) + P(H_0)}\]</span></p>
</section>
<section id="posterior-odds-formulation" class="slide level2">
<h2>Posterior Odds Formulation</h2>
<p>A simplified version:</p>
<p><span class="math display">\[\text{Posterior Odds} = \text{Bayes Factor} \times \text{Prior Odds}\]</span></p>
<p>Or:</p>
<p><span class="math display">\[\frac{P(H_1|data)}{P(H_0|data)} = BF_{10} \times \frac{P(H_1)}{P(H_0)}\]</span></p>
<p>This clearly shows how Bayes factors calibrate our prior beliefs (and we can use this on the wards!).</p>
<aside class="notes">
<p>Benefits of Bayes Factors for Genomics</p>
<ol type="1">
<li><strong>Calibrated evidence</strong>: Direct measure of evidence strength<br>
</li>
<li><strong>Multiple testing</strong>: Naturally incorporates prior odds<br>
</li>
<li><strong>Replication</strong>: Coherent framework for combining evidence across studies<br>
</li>
<li><strong>Diverse hypotheses</strong>: Can compare non-nested models<br>
</li>
<li><strong>Positive evidence</strong>: Can support null hypothesis, not just reject it<br>
</li>
<li><strong>Study design</strong>: Allows stopping when evidence is sufficient</li>
</ol>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="key-takeaways" class="slide level2">
<h2>Key Takeaways</h2>
<ol type="1">
<li>P-values answer a different question than most scientists ask<br>
</li>
<li>Bayes factors directly compare competing hypotheses<br>
</li>
<li>Even “significant” p-values provide weaker evidence than typically assumed<br>
</li>
<li>Bayes factors have a natural interpretation as evidence strength<br>
</li>
<li>Converting to posterior probabilities requires considering prior odds —</li>
</ol>
</section>
<section id="the-multiple-testing-challenge" class="slide level2">
<h2>2. The Multiple Testing Challenge</h2>
<p>Modern genomics routinely tests <strong>MANY</strong> hypotheses:</p>
<ul>
<li>20,000+ genes in differential expression: <a href="https://www.pnas.org/doi/10.1073/pnas.1530509100">Storey et al</a><br>
</li>
<li>Millions of variants in GWAS: <a href="https://academic.oup.com/biostatistics/article-abstract/18/2/275/2557030?redirectedFrom=fulltext">Stephens et al</a><br>
</li>
<li>Compounded potential interactions across multiple phenotypes: <a href="https://www.nature.com/articles/s41588-018-0268-8">Urbut al</a></li>
</ul>
<p><strong>The consequence</strong>: Many “significant” findings are actually false positives: but which ones?</p>
</section>
<section id="the-traditional-approach" class="slide level2">
<h2>The Traditional Approach</h2>
<p>When testing m hypotheses at significance level α:</p>
<ul>
<li>Expected number of false positives: m × α<br>
</li>
<li>With m = 1,000,000 and <span class="math inline">\(\alpha\)</span> = 0.05: <strong>50,000 false positives!</strong></li>
</ul>
<p><strong>Frequentist solutions</strong>:<br>
- Bonferroni correction: α/m<br>
- False Discovery Rate (FDR) control (Benjamini-Hochberg)<br>
- Family-wise error rate (FWER) control (<strong>Drawing time</strong>)?)</p>
</section>
<section id="visualizing-multiple-hypothesis-testing" class="slide level2">
<h2>Visualizing Multiple Hypothesis Testing</h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href=""></a><span class="co"># Set parameters</span></span>
<span id="cb5-2"><a href=""></a><span class="fu">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb5-3"><a href=""></a>m <span class="ot">&lt;-</span> <span class="dv">10000</span>  <span class="co"># Total number of tests</span></span>
<span id="cb5-4"><a href=""></a>pi0 <span class="ot">&lt;-</span> <span class="fl">0.7</span>  <span class="co"># True proportion of null hypotheses</span></span>
<span id="cb5-5"><a href=""></a>alpha <span class="ot">&lt;-</span> <span class="fl">0.05</span>  <span class="co"># Significance threshold</span></span>
<span id="cb5-6"><a href=""></a></span>
<span id="cb5-7"><a href=""></a><span class="co"># Generate p-values</span></span>
<span id="cb5-8"><a href=""></a><span class="co"># From null hypotheses</span></span>
<span id="cb5-9"><a href=""></a>n_null <span class="ot">&lt;-</span> <span class="fu">round</span>(m <span class="sc">*</span> pi0)</span>
<span id="cb5-10"><a href=""></a>p_null <span class="ot">&lt;-</span> <span class="fu">runif</span>(n_null)</span>
<span id="cb5-11"><a href=""></a></span>
<span id="cb5-12"><a href=""></a><span class="co"># From alternative hypotheses</span></span>
<span id="cb5-13"><a href=""></a>n_alt <span class="ot">&lt;-</span> m <span class="sc">-</span> n_null</span>
<span id="cb5-14"><a href=""></a>p_alt <span class="ot">&lt;-</span> <span class="fu">rbeta</span>(n_alt, <span class="fl">0.3</span>, <span class="dv">8</span>)  <span class="co"># This creates p-values that tend to be small</span></span>
<span id="cb5-15"><a href=""></a></span>
<span id="cb5-16"><a href=""></a><span class="co"># Combine p-values</span></span>
<span id="cb5-17"><a href=""></a>p_values <span class="ot">&lt;-</span> <span class="fu">c</span>(p_null, p_alt)</span>
<span id="cb5-18"><a href=""></a></span>
<span id="cb5-19"><a href=""></a><span class="co"># Calculate key quantities</span></span>
<span id="cb5-20"><a href=""></a>D_p <span class="ot">&lt;-</span> <span class="fu">sum</span>(p_values <span class="sc">&lt;=</span> alpha)  <span class="co"># Number of significant tests</span></span>
<span id="cb5-21"><a href=""></a>FP_p <span class="ot">&lt;-</span> pi0 <span class="sc">*</span> m <span class="sc">*</span> alpha       <span class="co"># Expected number of false positives</span></span>
<span id="cb5-22"><a href=""></a>FDR_p <span class="ot">&lt;-</span> FP_p <span class="sc">/</span> D_p           <span class="co"># False discovery rate</span></span>
<span id="cb5-23"><a href=""></a></span>
<span id="cb5-24"><a href=""></a><span class="co"># Create data frame for plotting</span></span>
<span id="cb5-25"><a href=""></a>df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb5-26"><a href=""></a>  <span class="at">p_value =</span> p_values,</span>
<span id="cb5-27"><a href=""></a>  <span class="at">significant =</span> p_values <span class="sc">&lt;=</span> alpha</span>
<span id="cb5-28"><a href=""></a>)</span>
<span id="cb5-29"><a href=""></a></span>
<span id="cb5-30"><a href=""></a><span class="co"># Create plot</span></span>
<span id="cb5-31"><a href=""></a><span class="fu">ggplot</span>(df, <span class="fu">aes</span>(<span class="at">x =</span> p_value, <span class="at">fill =</span> significant)) <span class="sc">+</span></span>
<span id="cb5-32"><a href=""></a>  <span class="fu">geom_histogram</span>(<span class="at">bins =</span> <span class="dv">50</span>, <span class="at">boundary =</span> <span class="dv">0</span>) <span class="sc">+</span></span>
<span id="cb5-33"><a href=""></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> pi0 <span class="sc">*</span> m<span class="sc">/</span><span class="dv">50</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">color =</span> <span class="st">"red"</span>) <span class="sc">+</span>  <span class="co"># Line at pi0</span></span>
<span id="cb5-34"><a href=""></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="fl">0.5</span>, <span class="at">y =</span> pi0 <span class="sc">*</span> m<span class="sc">/</span><span class="dv">40</span>, </span>
<span id="cb5-35"><a href=""></a>           <span class="at">label =</span> <span class="fu">paste</span>(<span class="st">"π₀ ="</span>, pi0), <span class="at">color =</span> <span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb5-36"><a href=""></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="fl">0.7</span>, <span class="at">y =</span> <span class="fu">max</span>(<span class="fu">hist</span>(df<span class="sc">$</span>p_value, <span class="at">breaks =</span> <span class="dv">50</span>, <span class="at">plot =</span> <span class="cn">FALSE</span>)<span class="sc">$</span>counts) <span class="sc">*</span> <span class="fl">0.8</span>,</span>
<span id="cb5-37"><a href=""></a>           <span class="at">label =</span> <span class="fu">paste</span>(<span class="st">"D("</span>, alpha, <span class="st">") ="</span>, D_p)) <span class="sc">+</span></span>
<span id="cb5-38"><a href=""></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="fl">0.7</span>, <span class="at">y =</span> <span class="fu">max</span>(<span class="fu">hist</span>(df<span class="sc">$</span>p_value, <span class="at">breaks =</span> <span class="dv">50</span>, <span class="at">plot =</span> <span class="cn">FALSE</span>)<span class="sc">$</span>counts) <span class="sc">*</span> <span class="fl">0.7</span>,</span>
<span id="cb5-39"><a href=""></a>           <span class="at">label =</span> <span class="fu">paste</span>(<span class="st">"FP("</span>, alpha, <span class="st">") ="</span>, <span class="fu">round</span>(FP_p))) <span class="sc">+</span></span>
<span id="cb5-40"><a href=""></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="fl">0.7</span>, <span class="at">y =</span> <span class="fu">max</span>(<span class="fu">hist</span>(df<span class="sc">$</span>p_value, <span class="at">breaks =</span> <span class="dv">50</span>, <span class="at">plot =</span> <span class="cn">FALSE</span>)<span class="sc">$</span>counts) <span class="sc">*</span> <span class="fl">0.6</span>,</span>
<span id="cb5-41"><a href=""></a>           <span class="at">label =</span> <span class="fu">paste</span>(<span class="st">"FDR("</span>, alpha, <span class="st">") ="</span>, <span class="fu">round</span>(FDR_p, <span class="dv">3</span>))) <span class="sc">+</span></span>
<span id="cb5-42"><a href=""></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"grey70"</span>, <span class="st">"blue"</span>)) <span class="sc">+</span></span>
<span id="cb5-43"><a href=""></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Distribution of P-values"</span>,</span>
<span id="cb5-44"><a href=""></a>       <span class="at">subtitle =</span> <span class="fu">paste0</span>(<span class="st">"m = "</span>, m, <span class="st">" tests, π0 = "</span>, pi0, <span class="st">" null proportion"</span>),</span>
<span id="cb5-45"><a href=""></a>       <span class="at">x =</span> <span class="st">"p-value"</span>,</span>
<span id="cb5-46"><a href=""></a>       <span class="at">y =</span> <span class="st">"Count"</span>,</span>
<span id="cb5-47"><a href=""></a>       <span class="at">fill =</span> <span class="st">"Significant"</span>) <span class="sc">+</span></span>
<span id="cb5-48"><a href=""></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb5-49"><a href=""></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"bottom"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>

</div>
<img data-src="mpg6_files/figure-revealjs/unnamed-chunk-4-1.png" width="960" class="r-stretch"><p>The challenge: Separating true signals (blue) from noise (gray) when true effects are rare. How do you know <span class="math inline">\(pi0\)</span>?vThe Bayesian solutions …</p>
</section>
<section id="local-false-discovery-rate-lfdr" class="slide level2">
<h2>Local False Discovery Rate (LFDR)</h2>
<p>Instead of controlling the overall FDR, we can calculate the probability that each individual test is a false discovery:</p>
<p><span class="math display">\[\text{LFDR}(z) = P(H_0|z) = \frac{(1-\pi_1)f_0(z)}{f(z)}\]</span></p>
<p>Where:</p>
<p>- <span class="math inline">\(\pi_1\)</span> = proportion of true effects</p>
<p>- <span class="math inline">\(f_0(z)\)</span> = null distribution</p>
<p>- <span class="math inline">\(f(z)\)</span> = observed distribution</p>
<aside class="notes">
<p>Why LFDR is Better</p>
<ol type="1">
<li><strong>Test-specific</strong>: Each test gets its own probability of being false</li>
<li><strong>Direct interpretation</strong>: “This test has 5% chance of being false”</li>
<li><strong>Natural handling of correlation</strong>: No need for independence assumptions</li>
<li><strong>Flexible</strong>: Can incorporate prior information about effect sizes</li>
</ol>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="the-mathematical-framework" class="slide level2">
<h2>The Mathematical Framework</h2>
<p>In multiple testing, we have two distributions:</p>
<ol type="1">
<li><p><strong>Null Distribution</strong> (<span class="math inline">\(f_0\)</span>): <span class="math display">\[f_0(z) = \mathcal{N}(0, se^2)\]</span> This is the standard normal distribution for tests where <span class="math inline">\(H_0\)</span> is true.</p></li>
<li><p><strong>Alternative Distribution</strong> (<span class="math inline">\(f_1\)</span>): <span class="math display">\[f_1(z) = \mathcal{N}(0, se^2 + \sigma^2)\]</span> This is a normal distribution with increased <em>true effect</em> variance for tests where <span class="math inline">\(H_1\)</span> is true.</p></li>
</ol>
</section>
<section id="the-mathematical-framework-1" class="slide level2">
<h2>The Mathematical Framework</h2>
<ol start="3" type="1">
<li><strong>Overall Distribution</strong> (<span class="math inline">\(f\)</span>): <span class="math display">\[f(z) = (1-\pi_1)f_0(z) + \pi_1f_1(z)\]</span> A mixture of null and alternative distributions (estimate posterior probability of non-null components) <strong>Homework (or in class exercise</strong>!</li>
</ol>
<p><span class="math display">\[\begin{aligned}
\hat{\beta}&amp;\sim N(\beta,se^2)\\
\beta &amp;\sim N(0,\sigma^2)\\
p(\beta) &amp;\sim \sum_k \pi_k N(0,\sigma_{k}^2)
\end{aligned}\]</span></p>
</section>
<section id="real-data-mixture-of-noisy-and-real-effects" class="slide level2">
<h2>Real data: Mixture of noisy and real effects</h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href=""></a><span class="co"># Create visualization of effect size mixtures</span></span>
<span id="cb6-2"><a href=""></a><span class="fu">set.seed</span>(<span class="dv">789</span>)</span>
<span id="cb6-3"><a href=""></a></span>
<span id="cb6-4"><a href=""></a><span class="co"># Parameters</span></span>
<span id="cb6-5"><a href=""></a>n_variants <span class="ot">&lt;-</span> <span class="dv">500</span></span>
<span id="cb6-6"><a href=""></a>pi0 <span class="ot">&lt;-</span> <span class="fl">0.95</span>  <span class="co"># Proportion of null effects</span></span>
<span id="cb6-7"><a href=""></a></span>
<span id="cb6-8"><a href=""></a><span class="co"># Generate true effects</span></span>
<span id="cb6-9"><a href=""></a>is_null <span class="ot">&lt;-</span> <span class="fu">rbinom</span>(n_variants, <span class="dv">1</span>, pi0)</span>
<span id="cb6-10"><a href=""></a>true_effects <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="dv">0</span>, n_variants)</span>
<span id="cb6-11"><a href=""></a>true_effects[is_null <span class="sc">==</span> <span class="dv">0</span>] <span class="ot">&lt;-</span> <span class="fu">rnorm</span>(<span class="fu">sum</span>(is_null <span class="sc">==</span> <span class="dv">0</span>), <span class="dv">0</span>, <span class="fl">0.5</span>)</span>
<span id="cb6-12"><a href=""></a>true_effects[is_null <span class="sc">==</span> <span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="fu">rnorm</span>(<span class="fu">sum</span>(is_null <span class="sc">==</span> <span class="dv">1</span>), <span class="dv">0</span>, <span class="dv">2</span>)</span>
<span id="cb6-13"><a href=""></a><span class="co"># Add noise to create observed effects</span></span>
<span id="cb6-14"><a href=""></a>observed_effects <span class="ot">&lt;-</span> true_effects <span class="sc">+</span> <span class="fu">rnorm</span>(n_variants, <span class="dv">0</span>, <span class="fl">0.2</span>)</span>
<span id="cb6-15"><a href=""></a></span>
<span id="cb6-16"><a href=""></a><span class="co"># Create data frame for plotting</span></span>
<span id="cb6-17"><a href=""></a>effect_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb6-18"><a href=""></a>  <span class="at">Variant =</span> <span class="dv">1</span><span class="sc">:</span>n_variants,</span>
<span id="cb6-19"><a href=""></a>  <span class="at">TrueEffect =</span> true_effects,</span>
<span id="cb6-20"><a href=""></a>  <span class="at">ObservedEffect =</span> observed_effects,</span>
<span id="cb6-21"><a href=""></a>  <span class="at">IsNull =</span> <span class="fu">factor</span>(is_null)</span>
<span id="cb6-22"><a href=""></a>)</span>
<span id="cb6-23"><a href=""></a></span>
<span id="cb6-24"><a href=""></a><span class="co"># Calculate LFDR for each z-score</span></span>
<span id="cb6-25"><a href=""></a>calc_lfdr <span class="ot">&lt;-</span> <span class="cf">function</span>(z, <span class="at">pi1=</span><span class="fl">0.10</span>, <span class="at">sigma=</span><span class="dv">4</span>) {  <span class="co"># Match the parameters used to generate data</span></span>
<span id="cb6-26"><a href=""></a>  f0 <span class="ot">&lt;-</span> <span class="fu">dnorm</span>(z, <span class="dv">0</span>, <span class="dv">1</span>)</span>
<span id="cb6-27"><a href=""></a>  f1 <span class="ot">&lt;-</span> <span class="fu">dnorm</span>(z, <span class="dv">0</span>, <span class="fu">sqrt</span>(<span class="dv">1</span> <span class="sc">+</span> sigma<span class="sc">^</span><span class="dv">2</span>))</span>
<span id="cb6-28"><a href=""></a>  f <span class="ot">&lt;-</span> (<span class="dv">1</span><span class="sc">-</span>pi1)<span class="sc">*</span>f0 <span class="sc">+</span> pi1<span class="sc">*</span>f1</span>
<span id="cb6-29"><a href=""></a>  (<span class="dv">1</span><span class="sc">-</span>pi1)<span class="sc">*</span>f0<span class="sc">/</span>f</span>
<span id="cb6-30"><a href=""></a>}</span>
<span id="cb6-31"><a href=""></a></span>
<span id="cb6-32"><a href=""></a></span>
<span id="cb6-33"><a href=""></a><span class="co"># Plot</span></span>
<span id="cb6-34"><a href=""></a><span class="fu">ggplot</span>(effect_df, <span class="fu">aes</span>(<span class="at">x =</span> ObservedEffect, <span class="at">fill =</span> IsNull)) <span class="sc">+</span></span>
<span id="cb6-35"><a href=""></a>  <span class="fu">geom_density</span>(<span class="at">alpha=</span><span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb6-36"><a href=""></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"gray70"</span>, <span class="st">"red"</span>), </span>
<span id="cb6-37"><a href=""></a>                   <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"Null"</span>, <span class="st">"True"</span>),</span>
<span id="cb6-38"><a href=""></a>                   <span class="at">name =</span> <span class="st">"Variant Type"</span>) <span class="sc">+</span></span>
<span id="cb6-39"><a href=""></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Mixture of Effect Sizes in GWAS"</span>,</span>
<span id="cb6-40"><a href=""></a>       <span class="at">subtitle =</span> <span class="st">"Most variants have no effect (null)"</span>,</span>
<span id="cb6-41"><a href=""></a>       <span class="at">x =</span> <span class="st">"Observed Effect Size"</span>, </span>
<span id="cb6-42"><a href=""></a>       <span class="at">y =</span> <span class="st">"Count"</span>) <span class="sc">+</span></span>
<span id="cb6-43"><a href=""></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb6-44"><a href=""></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"bottom"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>

</div>
<img data-src="mpg6_files/figure-revealjs/unnamed-chunk-5-1.png" width="960" class="r-stretch"></section>
<section id="empirical-bayes-approach" class="slide level2">
<h2>Empirical Bayes Approach</h2>
<p>Key contributions:</p>
<ol type="1">
<li><strong>Estimate from data</strong>: Learn <span class="math inline">\(\pi_1\)</span> and effect size distribution from the data</li>
<li><strong>Adaptive shrinkage</strong>: Stronger shrinkage for uncertain estimates</li>
<li><strong>Correlation structure</strong>: Account for sharing of noise (mashr)</li>
</ol>
<div class="callout callout-none no-icon callout-style-simple">
<div class="callout-body">
<div class="callout-content">
<ol type="1">
<li><strong>ashr</strong>: Adaptive shrinkage for effect sizes</li>
<li><strong>mashr</strong>: Multivariate adaptive shrinkage</li>
<li><strong>flashier</strong>: Sparse factor analysis (more on aladynoulli to come)</li>
</ol>
</div>
</div>
</div>
<aside class="notes">
<p>In the empirical Bayes approach: 1. Effects are shrunk adaptively based on both their size and the overall distribution of effects 2. Small observed effects are shrunk more heavily toward zero because they’re more likely to be noise 3. Large observed effects undergo less shrinkage because they’re more likely to be real 4. The amount of shrinkage is learned from the data itself through: Estimation of π₁ (proportion of non-null effects) Estimation of σ (scale of true effects) 5. The local false discovery rate (LFDR) determines shrinkage amount for each effect</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="empirical-bayes-approach-1" class="slide level2">
<h2>Empirical Bayes Approach</h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href=""></a><span class="co"># Visualization of adaptive shrinkage</span></span>
<span id="cb7-2"><a href=""></a><span class="fu">set.seed</span>(<span class="dv">456</span>)</span>
<span id="cb7-3"><a href=""></a>n_tests <span class="ot">&lt;-</span> <span class="dv">1000</span></span>
<span id="cb7-4"><a href=""></a>pi1 <span class="ot">&lt;-</span> <span class="fl">0.10</span></span>
<span id="cb7-5"><a href=""></a>sigma <span class="ot">&lt;-</span> <span class="dv">2</span></span>
<span id="cb7-6"><a href=""></a></span>
<span id="cb7-7"><a href=""></a><span class="co"># Generate data</span></span>
<span id="cb7-8"><a href=""></a>true_effects <span class="ot">&lt;-</span> <span class="fu">rnorm</span>(n_tests, <span class="dv">0</span>, sigma)</span>
<span id="cb7-9"><a href=""></a>is_null <span class="ot">&lt;-</span> <span class="fu">rbinom</span>(n_tests, <span class="dv">1</span>, <span class="dv">1</span><span class="sc">-</span>pi1)</span>
<span id="cb7-10"><a href=""></a>true_effects[is_null <span class="sc">==</span> <span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb7-11"><a href=""></a></span>
<span id="cb7-12"><a href=""></a><span class="co"># Add noise</span></span>
<span id="cb7-13"><a href=""></a>observed_effects <span class="ot">&lt;-</span> true_effects <span class="sc">+</span> <span class="fu">rnorm</span>(n_tests, <span class="dv">0</span>, <span class="dv">1</span>)</span>
<span id="cb7-14"><a href=""></a></span>
<span id="cb7-15"><a href=""></a><span class="co"># Calculate posterior means (simplified)</span></span>
<span id="cb7-16"><a href=""></a>posterior_means <span class="ot">&lt;-</span> observed_effects <span class="sc">*</span> (<span class="dv">1</span> <span class="sc">-</span> <span class="fu">calc_lfdr</span>(observed_effects, pi1, sigma))</span>
<span id="cb7-17"><a href=""></a></span>
<span id="cb7-18"><a href=""></a><span class="co"># Create data frame</span></span>
<span id="cb7-19"><a href=""></a>shrinkage_data <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb7-20"><a href=""></a>  <span class="at">Observed =</span> observed_effects,</span>
<span id="cb7-21"><a href=""></a>  <span class="at">True =</span> true_effects,</span>
<span id="cb7-22"><a href=""></a>  <span class="at">Shrunk =</span> posterior_means,</span>
<span id="cb7-23"><a href=""></a>  <span class="at">is_null=</span>is_null</span>
<span id="cb7-24"><a href=""></a>)</span>
<span id="cb7-25"><a href=""></a></span>
<span id="cb7-26"><a href=""></a><span class="co"># Plot</span></span>
<span id="cb7-27"><a href=""></a><span class="fu">ggplot</span>(shrinkage_data, <span class="fu">aes</span>(<span class="at">x =</span> Observed, <span class="at">y =</span> Shrunk,<span class="at">col=</span><span class="fu">as.factor</span>(is_null))) <span class="sc">+</span></span>
<span id="cb7-28"><a href=""></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb7-29"><a href=""></a>  <span class="fu">geom_abline</span>(<span class="at">slope =</span> <span class="dv">1</span>, <span class="at">intercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb7-30"><a href=""></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Adaptive Shrinkage"</span>,</span>
<span id="cb7-31"><a href=""></a>       <span class="at">subtitle =</span> <span class="st">"Stronger shrinkage for uncertain estimates"</span>,</span>
<span id="cb7-32"><a href=""></a>       <span class="at">x =</span> <span class="st">"Observed Effect"</span>, </span>
<span id="cb7-33"><a href=""></a>       <span class="at">y =</span> <span class="st">"Shrunk Effect"</span>) <span class="sc">+</span></span>
<span id="cb7-34"><a href=""></a>  <span class="fu">theme_minimal</span>(<span class="at">base_size =</span> <span class="dv">16</span>)  <span class="co"># Increased base size</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>

</div>
<img data-src="mpg6_files/figure-revealjs/unnamed-chunk-7-1.png" width="960" class="r-stretch"></section>
<section id="practical-implementation" class="slide level2">
<h2>Practical Implementation</h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href=""></a><span class="co"># Create visualization of the distributions</span></span>
<span id="cb8-2"><a href=""></a><span class="fu">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb8-3"><a href=""></a>z <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="sc">-</span><span class="dv">5</span>, <span class="dv">5</span>, <span class="at">length.out =</span> <span class="dv">1000</span>)</span>
<span id="cb8-4"><a href=""></a>pi1 <span class="ot">&lt;-</span> <span class="fl">0.05</span></span>
<span id="cb8-5"><a href=""></a></span>
<span id="cb8-6"><a href=""></a><span class="co"># Compare different prior variances</span></span>
<span id="cb8-7"><a href=""></a>sigma1 <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb8-8"><a href=""></a>sigma2 <span class="ot">&lt;-</span> <span class="dv">2</span></span>
<span id="cb8-9"><a href=""></a>sigma3 <span class="ot">&lt;-</span> <span class="dv">4</span></span>
<span id="cb8-10"><a href=""></a></span>
<span id="cb8-11"><a href=""></a><span class="co"># Calculate densities for different prior variances</span></span>
<span id="cb8-12"><a href=""></a>f0 <span class="ot">&lt;-</span> <span class="fu">dnorm</span>(z, <span class="dv">0</span>, <span class="dv">1</span>)  <span class="co"># Null distribution</span></span>
<span id="cb8-13"><a href=""></a>f1_small <span class="ot">&lt;-</span> <span class="fu">dnorm</span>(z, <span class="dv">0</span>, <span class="fu">sqrt</span>(<span class="dv">1</span> <span class="sc">+</span> sigma1<span class="sc">^</span><span class="dv">2</span>))  <span class="co"># Alternative with small variance</span></span>
<span id="cb8-14"><a href=""></a>f1_med <span class="ot">&lt;-</span> <span class="fu">dnorm</span>(z, <span class="dv">0</span>, <span class="fu">sqrt</span>(<span class="dv">1</span> <span class="sc">+</span> sigma2<span class="sc">^</span><span class="dv">2</span>))    <span class="co"># Alternative with medium variance</span></span>
<span id="cb8-15"><a href=""></a>f1_large <span class="ot">&lt;-</span> <span class="fu">dnorm</span>(z, <span class="dv">0</span>, <span class="fu">sqrt</span>(<span class="dv">1</span> <span class="sc">+</span> sigma3<span class="sc">^</span><span class="dv">2</span>))  <span class="co"># Alternative with large variance</span></span>
<span id="cb8-16"><a href=""></a></span>
<span id="cb8-17"><a href=""></a><span class="co"># Create data frame</span></span>
<span id="cb8-18"><a href=""></a>dist_data <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb8-19"><a href=""></a>  <span class="at">z =</span> <span class="fu">rep</span>(z, <span class="dv">4</span>),</span>
<span id="cb8-20"><a href=""></a>  <span class="at">density =</span> <span class="fu">c</span>(f0, f1_small, f1_med, f1_large),</span>
<span id="cb8-21"><a href=""></a>  <span class="at">Distribution =</span> <span class="fu">factor</span>(<span class="fu">rep</span>(<span class="fu">c</span>(<span class="st">"Null"</span>, <span class="st">"Alt (σ²=1)"</span>, <span class="st">"Alt (σ²=4)"</span>, <span class="st">"Alt (σ²=16)"</span>), </span>
<span id="cb8-22"><a href=""></a>                          <span class="at">each =</span> <span class="fu">length</span>(z)))</span>
<span id="cb8-23"><a href=""></a>)</span>
<span id="cb8-24"><a href=""></a></span>
<span id="cb8-25"><a href=""></a><span class="co"># Plot with larger size</span></span>
<span id="cb8-26"><a href=""></a><span class="fu">ggplot</span>(dist_data, <span class="fu">aes</span>(<span class="at">x =</span> z, <span class="at">y =</span> density, <span class="at">color =</span> Distribution)) <span class="sc">+</span></span>
<span id="cb8-27"><a href=""></a>  <span class="fu">geom_line</span>(<span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb8-28"><a href=""></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"Null"</span> <span class="ot">=</span> <span class="st">"red"</span>, </span>
<span id="cb8-29"><a href=""></a>                               <span class="st">"Alt (σ²=1)"</span> <span class="ot">=</span> <span class="st">"blue"</span>, </span>
<span id="cb8-30"><a href=""></a>                               <span class="st">"Alt (σ²=4)"</span> <span class="ot">=</span> <span class="st">"darkgreen"</span>,</span>
<span id="cb8-31"><a href=""></a>                               <span class="st">"Alt (σ²=16)"</span> <span class="ot">=</span> <span class="st">"purple"</span>)) <span class="sc">+</span></span>
<span id="cb8-32"><a href=""></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Effect of Prior Variance on Alternative Distribution"</span>,</span>
<span id="cb8-33"><a href=""></a>       <span class="at">subtitle =</span> <span class="st">"Larger prior variance (σ²) leads to less shrinkage"</span>,</span>
<span id="cb8-34"><a href=""></a>       <span class="at">x =</span> <span class="st">"Z-score"</span>, </span>
<span id="cb8-35"><a href=""></a>       <span class="at">y =</span> <span class="st">"Density"</span>) <span class="sc">+</span></span>
<span id="cb8-36"><a href=""></a>  <span class="fu">theme_minimal</span>(<span class="at">base_size =</span> <span class="dv">20</span>) <span class="sc">+</span>  <span class="co"># Increased base font size</span></span>
<span id="cb8-37"><a href=""></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"bottom"</span>,</span>
<span id="cb8-38"><a href=""></a>        <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">16</span>),</span>
<span id="cb8-39"><a href=""></a>        <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">18</span>),</span>
<span id="cb8-40"><a href=""></a>        <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">16</span>),</span>
<span id="cb8-41"><a href=""></a>        <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">18</span>),</span>
<span id="cb8-42"><a href=""></a>        <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">22</span>),</span>
<span id="cb8-43"><a href=""></a>        <span class="at">plot.subtitle =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">18</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure>
<p><img data-src="mpg6_files/figure-revealjs/unnamed-chunk-8-1.png" width="1440"></p>
</figure>
</div>
</div>
</div>
<aside class="notes">
<p>Effect of Prior Variance on Shrinkage</p>
<p>The prior variance (<span class="math inline">\(\sigma^2\)</span>) directly affects how much we shrink our estimates:</p>
<ol type="1">
<li><strong>Small prior variance</strong> (<span class="math inline">\(\sigma^2\)</span> small):
<ul>
<li>Alternative distribution is narrow</li>
<li>Strong shrinkage toward zero</li>
<li>Conservative estimates</li>
<li>Higher threshold for calling something significant</li>
</ul></li>
<li><strong>Large prior variance</strong> (<span class="math inline">\(\sigma^2\)</span> large):
<ul>
<li><p>Alternative distribution is wide</p></li>
<li><p>Less shrinkage toward zero</p></li>
<li><p>More willing to accept large effect sizes</p></li>
<li><p>Lower threshold for calling something significant</p></li>
</ul></li>
</ol>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href=""></a><span class="co"># Demonstrate shrinkage with different prior variances</span></span>
<span id="cb9-2"><a href=""></a><span class="fu">set.seed</span>(<span class="dv">456</span>)</span>
<span id="cb9-3"><a href=""></a>n_tests <span class="ot">&lt;-</span> <span class="dv">1000</span></span>
<span id="cb9-4"><a href=""></a>true_effects <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="dv">0</span>, n_tests <span class="sc">*</span> <span class="fl">0.95</span>), <span class="fu">rnorm</span>(n_tests <span class="sc">*</span> <span class="fl">0.05</span>, <span class="dv">0</span>, <span class="dv">2</span>))</span>
<span id="cb9-5"><a href=""></a>observed_effects <span class="ot">&lt;-</span> true_effects <span class="sc">+</span> <span class="fu">rnorm</span>(n_tests, <span class="dv">0</span>, <span class="dv">1</span>)</span>
<span id="cb9-6"><a href=""></a></span>
<span id="cb9-7"><a href=""></a><span class="co"># Calculate posterior means with different prior variances</span></span>
<span id="cb9-8"><a href=""></a>calc_shrinkage <span class="ot">&lt;-</span> <span class="cf">function</span>(z, pi1, sigma) {</span>
<span id="cb9-9"><a href=""></a>  f0 <span class="ot">&lt;-</span> <span class="fu">dnorm</span>(z, <span class="dv">0</span>, <span class="dv">1</span>)</span>
<span id="cb9-10"><a href=""></a>  f1 <span class="ot">&lt;-</span> <span class="fu">dnorm</span>(z, <span class="dv">0</span>, <span class="fu">sqrt</span>(<span class="dv">1</span> <span class="sc">+</span> sigma<span class="sc">^</span><span class="dv">2</span>))</span>
<span id="cb9-11"><a href=""></a>  f <span class="ot">&lt;-</span> (<span class="dv">1</span><span class="sc">-</span>pi1)<span class="sc">*</span>f0 <span class="sc">+</span> pi1<span class="sc">*</span>f1</span>
<span id="cb9-12"><a href=""></a>  lfdr <span class="ot">&lt;-</span> (<span class="dv">1</span><span class="sc">-</span>pi1)<span class="sc">*</span>f0<span class="sc">/</span>f</span>
<span id="cb9-13"><a href=""></a>  z <span class="sc">*</span> (<span class="dv">1</span> <span class="sc">-</span> lfdr)  <span class="co"># Simplified shrinkage estimate</span></span>
<span id="cb9-14"><a href=""></a>}</span>
<span id="cb9-15"><a href=""></a></span>
<span id="cb9-16"><a href=""></a>shrunk_small <span class="ot">&lt;-</span> <span class="fu">calc_shrinkage</span>(observed_effects, <span class="fl">0.05</span>, <span class="dv">1</span>)</span>
<span id="cb9-17"><a href=""></a>shrunk_med <span class="ot">&lt;-</span> <span class="fu">calc_shrinkage</span>(observed_effects, <span class="fl">0.05</span>, <span class="dv">2</span>)</span>
<span id="cb9-18"><a href=""></a>shrunk_large <span class="ot">&lt;-</span> <span class="fu">calc_shrinkage</span>(observed_effects, <span class="fl">0.05</span>, <span class="dv">4</span>)</span>
<span id="cb9-19"><a href=""></a></span>
<span id="cb9-20"><a href=""></a><span class="co"># Create data frame for plotting</span></span>
<span id="cb9-21"><a href=""></a>shrinkage_data <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb9-22"><a href=""></a>  <span class="at">Observed =</span> <span class="fu">rep</span>(observed_effects, <span class="dv">3</span>),</span>
<span id="cb9-23"><a href=""></a>  <span class="at">Shrunk =</span> <span class="fu">c</span>(shrunk_small, shrunk_med, shrunk_large),</span>
<span id="cb9-24"><a href=""></a>  <span class="at">PriorVar =</span> <span class="fu">factor</span>(<span class="fu">rep</span>(<span class="fu">c</span>(<span class="st">"σ²=1"</span>, <span class="st">"σ²=4"</span>, <span class="st">"σ²=16"</span>), <span class="at">each =</span> n_tests))</span>
<span id="cb9-25"><a href=""></a>)</span>
<span id="cb9-26"><a href=""></a></span>
<span id="cb9-27"><a href=""></a><span class="co"># Plot with larger size</span></span>
<span id="cb9-28"><a href=""></a><span class="fu">ggplot</span>(shrinkage_data, <span class="fu">aes</span>(<span class="at">x =</span> Observed, <span class="at">y =</span> Shrunk, <span class="at">color =</span> PriorVar)) <span class="sc">+</span></span>
<span id="cb9-29"><a href=""></a>  <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>, <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb9-30"><a href=""></a>  <span class="fu">geom_abline</span>(<span class="at">slope =</span> <span class="dv">1</span>, <span class="at">intercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb9-31"><a href=""></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"σ²=1"</span> <span class="ot">=</span> <span class="st">"blue"</span>, </span>
<span id="cb9-32"><a href=""></a>                               <span class="st">"σ²=4"</span> <span class="ot">=</span> <span class="st">"darkgreen"</span>,</span>
<span id="cb9-33"><a href=""></a>                               <span class="st">"σ²=16"</span> <span class="ot">=</span> <span class="st">"purple"</span>)) <span class="sc">+</span></span>
<span id="cb9-34"><a href=""></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Effect of Prior Variance on Shrinkage"</span>,</span>
<span id="cb9-35"><a href=""></a>       <span class="at">subtitle =</span> <span class="st">"Larger prior variance leads to less shrinkage toward zero"</span>,</span>
<span id="cb9-36"><a href=""></a>       <span class="at">x =</span> <span class="st">"Observed Effect"</span>, </span>
<span id="cb9-37"><a href=""></a>       <span class="at">y =</span> <span class="st">"Shrunk Effect"</span>,</span>
<span id="cb9-38"><a href=""></a>       <span class="at">color =</span> <span class="st">"Prior Variance"</span>) <span class="sc">+</span></span>
<span id="cb9-39"><a href=""></a>  <span class="fu">theme_minimal</span>(<span class="at">base_size =</span> <span class="dv">20</span>) <span class="sc">+</span></span>
<span id="cb9-40"><a href=""></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"bottom"</span>,</span>
<span id="cb9-41"><a href=""></a>        <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">16</span>),</span>
<span id="cb9-42"><a href=""></a>        <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">18</span>),</span>
<span id="cb9-43"><a href=""></a>        <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">16</span>),</span>
<span id="cb9-44"><a href=""></a>        <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">18</span>),</span>
<span id="cb9-45"><a href=""></a>        <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">22</span>),</span>
<span id="cb9-46"><a href=""></a>        <span class="at">plot.subtitle =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">18</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure>
<p><img data-src="mpg6_files/figure-revealjs/unnamed-chunk-9-1.png" width="1440"></p>
</figure>
</div>
</div>
</div>
<aside class="notes">
<p>This relationship between prior variance and shrinkage is crucial in genomics:</p>
<ul>
<li>For rare variants or small studies: Use smaller <span class="math inline">\(\sigma^2\)</span> to be conservative</li>
<li>For common variants or large studies: Can use larger <span class="math inline">\(\sigma^2\)</span></li>
<li>For follow-up studies: Can use previous effect size estimates to inform <span class="math inline">\(\sigma^2\)</span></li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="multivariate-adaptive-shrinkage-mash" class="slide level2">
<h2>Multivariate Adaptive Shrinkage (mash)</h2>
<p>When we have effects across multiple groups (e.g., tissues, conditions), we can borrow strength:</p>
<p><span class="math display">\[\begin{align*}
\text{Effects across groups: } &amp; \mathbf{B}_j \sim N(\mathbf{0}, \mathbf{U}_k) \\
\text{Mixture model: } &amp; p(\mathbf{B}_j) = \sum_{k=1}^K \pi_k N(\mathbf{0}, \mathbf{U}_k)
\end{align*}\]</span></p>
<p>Where <span class="math inline">\(\mathbf{U}_k\)</span> captures different patterns of sharing: - Equal effects across groups<br>
- Group-specific effects<br>
- Correlated effects<br>
- Structured patterns</p>
</section>
<section id="understanding-covariance-matrices-in-multivariate-normal-distribution" class="slide level2">
<h2>Understanding Covariance Matrices in Multivariate Normal Distribution</h2>
<p>Let’s consider a bivariate normal distribution with covariance matrix U_k:</p>
<p><span class="math display">\[\mathbf{U}\_k =
\begin{pmatrix}
\sigma^2_1 &amp; \rho\sigma_1\sigma_2 \\
\rho\sigma_1\sigma_2 &amp; \sigma^2_2
\end{pmatrix}\]</span></p>
<ul>
<li><span class="math inline">\(\sigma^2_1\)</span> and <span class="math inline">\(\sigma^2_2\)</span>: trait variances<br>
</li>
<li><span class="math inline">\(\rho\)</span> is the correlation coefficient<br>
</li>
<li><span class="math inline">\(\rho\sigma_1\sigma_2\)</span> represents the covariance.</li>
</ul>
</section>
<section id="more-math" class="slide level2">
<h2>More math</h2>
<p>The multivariate normal density is then:</p>
<p><span class="math display">\[p(\mathbf{x}|\mathbf{U}\_k) = \frac{1}{2\pi|\mathbf{U}_k|^{1/2}}
\exp\left(-\frac{1}{2}\mathbf{x}^T\mathbf{U}_k^{-1}\mathbf{x}\right)\]</span></p>
<p>Different covariance matrices (<span class="math inline">\(\mathbf{U}_k\)</span>) lead to different shapes:</p>
<ol type="1">
<li>When (<span class="math inline">\(\rho\)</span> = 0):
<ul>
<li>Effects are independent</li>
<li>approaches univariate shrinkage but ..</li>
</ul></li>
<li>When (<span class="math inline">\(\rho\)</span> <span class="math inline">\(\neq\)</span> 0):
<ul>
<li>Effects are correlated, ‘spread’ of cloud <span class="math inline">\(\rho\)</span></li>
<li>Slope determined by relative <em>variance</em></li>
</ul></li>
</ol>
</section>
<section id="how-would-you-get-a-11-line" class="slide level2">
<h2>How would you get a 1:1 line?</h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href=""></a><span class="fu">library</span>(MASS)</span>
<span id="cb10-2"><a href=""></a>sig1 <span class="ot">=</span><span class="dv">2</span>;sig2<span class="ot">=</span><span class="dv">3</span></span>
<span id="cb10-3"><a href=""></a>cov1<span class="ot">=</span><span class="fu">matrix</span>(<span class="fu">c</span>(sig1<span class="sc">^</span><span class="dv">2</span>,sig1<span class="sc">*</span>sig2,sig2<span class="sc">*</span>sig1,sig2<span class="sc">^</span><span class="dv">2</span>),<span class="at">byrow =</span> <span class="cn">TRUE</span>,<span class="at">nrow=</span><span class="dv">2</span>)</span>
<span id="cb10-4"><a href=""></a>m<span class="ot">=</span><span class="fu">mvrnorm</span>(<span class="dv">10000</span>,<span class="at">mu =</span> <span class="fu">rep</span>(<span class="dv">0</span>,<span class="dv">2</span>),<span class="at">Sigma =</span> cov1)</span>
<span id="cb10-5"><a href=""></a><span class="fu">plot</span>(m[,<span class="dv">1</span>],m[,<span class="dv">2</span>],<span class="at">xlab=</span><span class="st">"Effect 1"</span>,<span class="at">ylab=</span><span class="st">"Effect 2"</span>)</span>
<span id="cb10-6"><a href=""></a><span class="fu">abline</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>

</div>
<img data-src="mpg6_files/figure-revealjs/unnamed-chunk-10-1.png" width="960" class="r-stretch"></section>
<section id="bringing-it-home" class="slide level2">
<h2>Bringing it home!</h2>

<img data-src="mpg6_files/figure-revealjs/unnamed-chunk-11-1.png" width="960" class="r-stretch"></section>
<section id="summary" class="slide level2">
<h2>Summary</h2>
<ul>
<li><p><strong>Observed Effects are noisy</strong></p></li>
<li><p><strong>Only considering only information in one subgroup ignores the <em>hints</em> we get from abound</strong></p></li>
<li><p><strong>Sharing is caring!</strong></p></li>
</ul>
</section></section>
<section>
<section id="conjugate-priors-why-theyre-beautiful" class="title-slide slide level1 center">
<h1>3. Conjugate Priors: Why They’re Beautiful</h1>
<p><strong>Definition</strong>: A prior is conjugate when the posterior has the same distribution family as the prior.</p>
<p><strong><em>Three Examples</em></strong> <strong>for Genomics: Beta, Dirichlet, Normal</strong></p>
<p>Let’s see this mathematically:</p>
<!-- ## Conjugate Priors: Why They're Beautiful -->
<!-- For example, in the Normal-Normal case: \[ \begin{align} -->
<!-- \text{Prior: } & \theta \sim N(\mu_0, \sigma^2_0) \\ -->
<!-- \text{Likelihood: } & X|\theta \sim N(\theta, \sigma^2) \\ -->
<!-- \text{Posterior: } & \theta|X \sim N\left(\frac{\sigma^2_0 X + \sigma^2\mu_0}{\sigma^2_0 + \sigma^2}, \frac{\sigma^2_0\sigma^2}{\sigma^2_0 + \sigma^2}\right) -->
<!-- \end{align} -->
<!-- The beauty is that: 1. Prior starts as Normal 2. Data comes from Normal -->
<!-- 3. Posterior ends up Normal 4. Just with updated parameters! -->
<!-- This makes computation tractable and interpretation intuitive - we're -->
<!-- just updating our beliefs while staying in the same family of -->
<!-- distributions. -->
</section>
<section id="beta-binomial-conjugacy-adding-counts-intuition-perfect-for-allele-frequencies" class="slide level2">
<h2>Beta-Binomial Conjugacy: Adding Counts Intuition, Perfect for Allele Frequencies</h2>
<p>The <span class="math inline">\(\beta(\alpha, \beta)\)</span> distribution updates by simply adding successes to (<span class="math inline">\(\alpha\)</span>) and failures to (<span class="math inline">\(\beta\)</span>)!</p>
<p><span class="math display">\[\begin{align}
\text{Prior: } &amp; \theta \sim \text{Beta}(\alpha_0, \beta_0) \\
\text{Data: } &amp; \text{Observe: } s \text{ successes, } f \text{ failures} \\
\text{Posterior: } &amp; \theta \sim \text{Beta}(\alpha_0 + s, \beta_0 + f)
\end{align}\]</span></p>
</section>
<section id="beta-binomial-perfect-for-allele-frequencies" class="slide level2">
<h2>Beta-Binomial: Perfect for Allele Frequencies</h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href=""></a><span class="co"># Create a visualization of Beta-Binomial for allele frequencies</span></span>
<span id="cb11-2"><a href=""></a><span class="fu">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb11-3"><a href=""></a></span>
<span id="cb11-4"><a href=""></a><span class="co"># Parameters</span></span>
<span id="cb11-5"><a href=""></a>true_af <span class="ot">&lt;-</span> <span class="fl">0.3</span></span>
<span id="cb11-6"><a href=""></a>n_samples <span class="ot">&lt;-</span> <span class="dv">20</span></span>
<span id="cb11-7"><a href=""></a>n_reads_per_sample <span class="ot">&lt;-</span> <span class="dv">30</span></span>
<span id="cb11-8"><a href=""></a></span>
<span id="cb11-9"><a href=""></a><span class="co"># Generate data</span></span>
<span id="cb11-10"><a href=""></a>genotypes <span class="ot">&lt;-</span> <span class="fu">rbinom</span>(n_samples, <span class="dv">2</span>, true_af)</span>
<span id="cb11-11"><a href=""></a>read_counts <span class="ot">&lt;-</span> <span class="fu">rbinom</span>(n_samples, n_reads_per_sample, genotypes<span class="sc">/</span><span class="dv">2</span>)</span>
<span id="cb11-12"><a href=""></a>total_alt <span class="ot">&lt;-</span> <span class="fu">sum</span>(read_counts)</span>
<span id="cb11-13"><a href=""></a>total_reads <span class="ot">&lt;-</span> n_samples <span class="sc">*</span> n_reads_per_sample</span>
<span id="cb11-14"><a href=""></a></span>
<span id="cb11-15"><a href=""></a><span class="co"># Create data for plotting</span></span>
<span id="cb11-16"><a href=""></a>x <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="at">length.out =</span> <span class="dv">200</span>)</span>
<span id="cb11-17"><a href=""></a></span>
<span id="cb11-18"><a href=""></a><span class="co"># Different priors</span></span>
<span id="cb11-19"><a href=""></a>priors <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb11-20"><a href=""></a>  <span class="st">"Flat"</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">1</span>),</span>
<span id="cb11-21"><a href=""></a>  <span class="st">"Informative"</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="dv">10</span>, <span class="dv">20</span>),</span>
<span id="cb11-22"><a href=""></a>  <span class="st">"Incorrect"</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="dv">20</span>, <span class="dv">5</span>)</span>
<span id="cb11-23"><a href=""></a>)</span>
<span id="cb11-24"><a href=""></a></span>
<span id="cb11-25"><a href=""></a><span class="co"># Create data frame for plotting</span></span>
<span id="cb11-26"><a href=""></a>plot_data <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb11-27"><a href=""></a></span>
<span id="cb11-28"><a href=""></a><span class="cf">for</span> (prior_name <span class="cf">in</span> <span class="fu">names</span>(priors)) {</span>
<span id="cb11-29"><a href=""></a>  prior_params <span class="ot">&lt;-</span> priors[[prior_name]]</span>
<span id="cb11-30"><a href=""></a>  alpha_prior <span class="ot">&lt;-</span> prior_params[<span class="dv">1</span>]</span>
<span id="cb11-31"><a href=""></a>  beta_prior <span class="ot">&lt;-</span> prior_params[<span class="dv">2</span>]</span>
<span id="cb11-32"><a href=""></a>  </span>
<span id="cb11-33"><a href=""></a>  <span class="co"># Prior</span></span>
<span id="cb11-34"><a href=""></a>  prior_y <span class="ot">&lt;-</span> <span class="fu">dbeta</span>(x, alpha_prior, beta_prior)</span>
<span id="cb11-35"><a href=""></a>  </span>
<span id="cb11-36"><a href=""></a>  <span class="co"># Posterior</span></span>
<span id="cb11-37"><a href=""></a>  alpha_post <span class="ot">&lt;-</span> alpha_prior <span class="sc">+</span> total_alt</span>
<span id="cb11-38"><a href=""></a>  beta_post <span class="ot">&lt;-</span> beta_prior <span class="sc">+</span> total_reads <span class="sc">-</span> total_alt</span>
<span id="cb11-39"><a href=""></a>  posterior_y <span class="ot">&lt;-</span> <span class="fu">dbeta</span>(x, alpha_post, beta_post)</span>
<span id="cb11-40"><a href=""></a>  </span>
<span id="cb11-41"><a href=""></a>  <span class="co"># Add to data frame</span></span>
<span id="cb11-42"><a href=""></a>  plot_data <span class="ot">&lt;-</span> <span class="fu">rbind</span>(plot_data, </span>
<span id="cb11-43"><a href=""></a>                    <span class="fu">data.frame</span>(</span>
<span id="cb11-44"><a href=""></a>                      <span class="at">x =</span> x,</span>
<span id="cb11-45"><a href=""></a>                      <span class="at">y =</span> prior_y,</span>
<span id="cb11-46"><a href=""></a>                      <span class="at">Distribution =</span> <span class="st">"Prior"</span>,</span>
<span id="cb11-47"><a href=""></a>                      <span class="at">Prior =</span> prior_name</span>
<span id="cb11-48"><a href=""></a>                    ),</span>
<span id="cb11-49"><a href=""></a>                    <span class="fu">data.frame</span>(</span>
<span id="cb11-50"><a href=""></a>                      <span class="at">x =</span> x,</span>
<span id="cb11-51"><a href=""></a>                      <span class="at">y =</span> posterior_y,</span>
<span id="cb11-52"><a href=""></a>                      <span class="at">Distribution =</span> <span class="st">"Posterior"</span>,</span>
<span id="cb11-53"><a href=""></a>                      <span class="at">Prior =</span> prior_name</span>
<span id="cb11-54"><a href=""></a>                    ))</span>
<span id="cb11-55"><a href=""></a>}</span>
<span id="cb11-56"><a href=""></a></span>
<span id="cb11-57"><a href=""></a><span class="co"># Add true value line</span></span>
<span id="cb11-58"><a href=""></a>plot_data<span class="sc">$</span>Distribution <span class="ot">&lt;-</span> <span class="fu">factor</span>(plot_data<span class="sc">$</span>Distribution, </span>
<span id="cb11-59"><a href=""></a>                               <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"Prior"</span>, <span class="st">"Posterior"</span>))</span>
<span id="cb11-60"><a href=""></a></span>
<span id="cb11-61"><a href=""></a><span class="co"># Create plot</span></span>
<span id="cb11-62"><a href=""></a>  <span class="fu">ggplot</span>(plot_data, <span class="fu">aes</span>(<span class="at">x =</span> x, <span class="at">y =</span> y, <span class="at">linetype =</span> Distribution, <span class="at">color =</span> <span class="fu">as.factor</span>(Prior))) <span class="sc">+</span></span>
<span id="cb11-63"><a href=""></a>  <span class="fu">geom_line</span>(<span class="at">size =</span> <span class="fl">1.2</span>) <span class="sc">+</span></span>
<span id="cb11-64"><a href=""></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> true_af, <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">color =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb11-65"><a href=""></a>  <span class="co">#scale_color_manual(values = c("Prior" = "blue", "Posterior" = "red")) +</span></span>
<span id="cb11-66"><a href=""></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Beta-Binomial for Allele Frequency Estimation"</span>,</span>
<span id="cb11-67"><a href=""></a>       <span class="at">subtitle =</span> <span class="fu">paste0</span>(<span class="st">"True AF = "</span>, true_af, <span class="st">", Observed = "</span>, <span class="fu">round</span>(total_alt<span class="sc">/</span>total_reads, <span class="dv">3</span>)),</span>
<span id="cb11-68"><a href=""></a>       <span class="at">x =</span> <span class="st">"Allele Frequency"</span>, </span>
<span id="cb11-69"><a href=""></a>       <span class="at">y =</span> <span class="st">"Density"</span>) <span class="sc">+</span></span>
<span id="cb11-70"><a href=""></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"bottom"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>

</div>
<img data-src="mpg6_files/figure-revealjs/unnamed-chunk-12-1.png" width="960" class="r-stretch"><aside class="notes">
<p>even under an ‘incoorrect prior’, you can’t help but come to the truth with adeuqate data</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="simple-example" class="slide level2">
<h2>Simple Example:</h2>
<p>Why TB was a geneticist: Imagine tracking minor allele frequency:</p>
<ul>
<li><p><strong>Start</strong>: Beta(2, 8) - prior belief allele is rare</p></li>
<li><p><strong>Observe</strong>: AABB AABB ABBB (5 A’s, 7 B’s)</p></li>
<li><p><strong>Updates</strong>: After 1st group: Beta(4, 10)</p></li>
<li><p>After 2nd group: Beta(6, 12)</p></li>
<li><p>After 3rd group: Beta(7, 15)</p></li>
</ul>
</section>
<section id="why-this-is-beautiful" class="slide level2">
<h2>Why this is beautiful:</h2>
<ol type="1">
<li>Each A (success) adds 1 to <span class="math inline">\(\alpha\)</span></li>
<li>Each B (failure) adds 1 to <span class="math inline">\(\beta\)</span></li>
<li>(<span class="math inline">\(\alpha\)</span>) = prior successes + observed successes</li>
<li>(<span class="math inline">\(\beta\)</span>) = prior failures + observed failures</li>
<li>Posterior mean = (<span class="math inline">\(\frac{\alpha}{\alpha + \beta}\)</span>)</li>
</ol>
<aside class="notes">
<p>sum of a and b is like precision, higher concentration at mean mean is a/a+b</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="continuous-updating" class="slide level2">
<h2>Continuous updating</h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href=""></a><span class="co"># Create a function to generate conjugate updating visualization</span></span>
<span id="cb12-2"><a href=""></a>plot_conjugate_updating <span class="ot">&lt;-</span> <span class="cf">function</span>() {</span>
<span id="cb12-3"><a href=""></a>  <span class="co"># Initial prior</span></span>
<span id="cb12-4"><a href=""></a>  alpha0 <span class="ot">&lt;-</span> <span class="dv">2</span></span>
<span id="cb12-5"><a href=""></a>  beta0 <span class="ot">&lt;-</span> <span class="dv">8</span></span>
<span id="cb12-6"><a href=""></a>  </span>
<span id="cb12-7"><a href=""></a>  <span class="co"># Data observations (success/failure)</span></span>
<span id="cb12-8"><a href=""></a>  data <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">0</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">0</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">0</span>)</span>
<span id="cb12-9"><a href=""></a>  </span>
<span id="cb12-10"><a href=""></a>  <span class="co"># Create sequence for plotting</span></span>
<span id="cb12-11"><a href=""></a>  x <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="at">length.out =</span> <span class="dv">200</span>)</span>
<span id="cb12-12"><a href=""></a>  </span>
<span id="cb12-13"><a href=""></a>  <span class="co"># Initialize data frame for plotting</span></span>
<span id="cb12-14"><a href=""></a>  plot_data <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb12-15"><a href=""></a>  </span>
<span id="cb12-16"><a href=""></a>  <span class="co"># Initial prior</span></span>
<span id="cb12-17"><a href=""></a>  prior_y <span class="ot">&lt;-</span> <span class="fu">dbeta</span>(x, alpha0, beta0)</span>
<span id="cb12-18"><a href=""></a>  plot_data <span class="ot">&lt;-</span> <span class="fu">rbind</span>(plot_data, <span class="fu">data.frame</span>(</span>
<span id="cb12-19"><a href=""></a>    <span class="at">x =</span> x, <span class="at">y =</span> prior_y, </span>
<span id="cb12-20"><a href=""></a>    <span class="at">step =</span> <span class="dv">0</span>, </span>
<span id="cb12-21"><a href=""></a>    <span class="at">label =</span> <span class="fu">paste0</span>(<span class="st">"Prior: Beta("</span>, alpha0, <span class="st">","</span>, beta0, <span class="st">")"</span>)</span>
<span id="cb12-22"><a href=""></a>  ))</span>
<span id="cb12-23"><a href=""></a>  </span>
<span id="cb12-24"><a href=""></a>  <span class="co"># Update step by step</span></span>
<span id="cb12-25"><a href=""></a>  alpha <span class="ot">&lt;-</span> alpha0</span>
<span id="cb12-26"><a href=""></a>  beta <span class="ot">&lt;-</span> beta0</span>
<span id="cb12-27"><a href=""></a>  </span>
<span id="cb12-28"><a href=""></a>  <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(data)) {</span>
<span id="cb12-29"><a href=""></a>    <span class="co"># Update parameters</span></span>
<span id="cb12-30"><a href=""></a>    <span class="cf">if</span> (data[i] <span class="sc">==</span> <span class="dv">1</span>) {</span>
<span id="cb12-31"><a href=""></a>      alpha <span class="ot">&lt;-</span> alpha <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb12-32"><a href=""></a>    } <span class="cf">else</span> {</span>
<span id="cb12-33"><a href=""></a>      beta <span class="ot">&lt;-</span> beta <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb12-34"><a href=""></a>    }</span>
<span id="cb12-35"><a href=""></a>    </span>
<span id="cb12-36"><a href=""></a>    <span class="co"># Only plot a subset of steps for clarity</span></span>
<span id="cb12-37"><a href=""></a>    <span class="cf">if</span> (i <span class="sc">%in%</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">3</span>, <span class="dv">5</span>, <span class="dv">10</span>)) {</span>
<span id="cb12-38"><a href=""></a>      posterior_y <span class="ot">&lt;-</span> <span class="fu">dbeta</span>(x, alpha, beta)</span>
<span id="cb12-39"><a href=""></a>      plot_data <span class="ot">&lt;-</span> <span class="fu">rbind</span>(plot_data, <span class="fu">data.frame</span>(</span>
<span id="cb12-40"><a href=""></a>        <span class="at">x =</span> x, <span class="at">y =</span> posterior_y, </span>
<span id="cb12-41"><a href=""></a>        <span class="at">step =</span> i, </span>
<span id="cb12-42"><a href=""></a>        <span class="at">label =</span> <span class="fu">paste0</span>(<span class="st">"After "</span>, i, <span class="st">" observations: Beta("</span>, alpha, <span class="st">","</span>, beta, <span class="st">")"</span>)</span>
<span id="cb12-43"><a href=""></a>      ))</span>
<span id="cb12-44"><a href=""></a>    }</span>
<span id="cb12-45"><a href=""></a>  }</span>
<span id="cb12-46"><a href=""></a>  </span>
<span id="cb12-47"><a href=""></a>  <span class="co"># Convert step to factor for proper ordering</span></span>
<span id="cb12-48"><a href=""></a>  plot_data<span class="sc">$</span>step <span class="ot">&lt;-</span> <span class="fu">factor</span>(plot_data<span class="sc">$</span>step, <span class="at">levels =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="dv">3</span>, <span class="dv">5</span>, <span class="dv">10</span>))</span>
<span id="cb12-49"><a href=""></a>  </span>
<span id="cb12-50"><a href=""></a>  <span class="co"># Create plot</span></span>
<span id="cb12-51"><a href=""></a>  <span class="fu">ggplot</span>(plot_data, <span class="fu">aes</span>(<span class="at">x =</span> x, <span class="at">y =</span> y, <span class="at">color =</span> step, <span class="at">group =</span> step)) <span class="sc">+</span></span>
<span id="cb12-52"><a href=""></a>    <span class="fu">geom_line</span>(<span class="at">size =</span> <span class="fl">1.2</span>) <span class="sc">+</span></span>
<span id="cb12-53"><a href=""></a>    <span class="fu">scale_color_brewer</span>(<span class="at">palette =</span> <span class="st">"Set1"</span>, </span>
<span id="cb12-54"><a href=""></a>                      <span class="at">name =</span> <span class="st">"Update Step"</span>,</span>
<span id="cb12-55"><a href=""></a>                      <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"Prior"</span>, <span class="st">"After 1 obs"</span>, <span class="st">"After 3 obs"</span>, </span>
<span id="cb12-56"><a href=""></a>                                <span class="st">"After 5 obs"</span>, <span class="st">"After 10 obs"</span>)) <span class="sc">+</span></span>
<span id="cb12-57"><a href=""></a>    <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Sequential Updating with Conjugate Prior"</span>,</span>
<span id="cb12-58"><a href=""></a>         <span class="at">subtitle =</span> <span class="st">"Beta-Binomial conjugacy for allele frequency estimation"</span>,</span>
<span id="cb12-59"><a href=""></a>         <span class="at">x =</span> <span class="st">"Allele Frequency"</span>, </span>
<span id="cb12-60"><a href=""></a>         <span class="at">y =</span> <span class="st">"Density"</span>) <span class="sc">+</span></span>
<span id="cb12-61"><a href=""></a>    <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"bottom"</span>)</span>
<span id="cb12-62"><a href=""></a>}</span>
<span id="cb12-63"><a href=""></a></span>
<span id="cb12-64"><a href=""></a><span class="co"># Generate the plot</span></span>
<span id="cb12-65"><a href=""></a><span class="fu">plot_conjugate_updating</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>

</div>
<img data-src="mpg6_files/figure-revealjs/unnamed-chunk-13-1.png" width="960" class="r-stretch"><aside class="notes">
<p>truth was 0.5</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="dirichlet-multinomial-for-multiple-alleles" class="slide level2">
<h2>Dirichlet-Multinomial: For Multiple Alleles</h2>
<p><strong>Model</strong>:<br>
- <strong>Prior</strong>: <span class="math inline">\(\vec{\theta} \sim \text{Dirichlet}(\vec{\alpha})\)</span><br>
- <strong>Likelihood</strong>: <span class="math inline">\(\vec{X}|\vec{\theta} \sim \text{Multinomial}(n, \vec{\theta})\)</span><br>
- <strong>Posterior</strong>: <span class="math inline">\(\vec{\theta}|\vec{X} \sim \text{Dirichlet}(\vec{\alpha} + \vec{X})\)</span></p>
<h3 id="key-intuition">Key Intuition:</h3>
<p>The Dirichlet-Multinomial is just like Beta-Binomial, but for multiple categories instead of just two!</p>
<p><span class="math inline">\(\begin{align}
\text{Prior: } &amp; \vec{\theta} \sim \text{Dirichlet}(\alpha_1, \alpha_2, ..., \alpha_K) \\
\text{Data: } &amp; \vec{X} = (x_1, x_2, ..., x_K) \text{ counts in each category} \\
\text{Posterior: } &amp; \vec{\theta} \sim \text{Dirichlet}(\alpha_1 + x_1, \alpha_2 + x_2, ..., \alpha_K + x_K)
\end{align}\)</span></p>
</section>
<section id="simple-example-1" class="slide level2">
<h2>Simple Example:</h2>
<p>Imagine tracking allele frequencies for three alleles (A, B, C):</p>
<ul>
<li><p><strong>Prior</strong>: Dirichlet(2, 2, 2)</p></li>
<li><p>equally uncertain about all alleles</p></li>
<li><p><strong>Observe</strong>: 10 A’s, 5 B’s, 3 C’s</p></li>
<li><p><strong>Posterior</strong>: Dirichlet(12, 7, 5) - just add counts to prior parameters!</p></li>
</ul>
<aside class="notes">
<ol type="1">
<li>Each observation simply adds 1 to its category’s parameter</li>
<li>$_k$ can be thought of as “pseudo-counts”</li>
<li>Larger prior $$)s = stronger prior beliefs</li>
<li>Sum of ($$)’s = sample size of prior belief</li>
<li>Posterior mean for category k: ($$)</li>
</ol>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="why-this-is-beautiful-1" class="slide level2">
<h2>Why this is beautiful:</h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href=""></a><span class="co"># Create a visualization of Dirichlet-Multinomial</span></span>
<span id="cb13-2"><a href=""></a><span class="fu">set.seed</span>(<span class="dv">234</span>)</span>
<span id="cb13-3"><a href=""></a></span>
<span id="cb13-4"><a href=""></a><span class="co"># Function to generate random points from a Dirichlet distribution</span></span>
<span id="cb13-5"><a href=""></a>rdirichlet_2d <span class="ot">&lt;-</span> <span class="cf">function</span>(n, alpha) {</span>
<span id="cb13-6"><a href=""></a>  x <span class="ot">&lt;-</span> <span class="fu">rgamma</span>(n, alpha[<span class="dv">1</span>], <span class="dv">1</span>)</span>
<span id="cb13-7"><a href=""></a>  y <span class="ot">&lt;-</span> <span class="fu">rgamma</span>(n, alpha[<span class="dv">2</span>], <span class="dv">1</span>)</span>
<span id="cb13-8"><a href=""></a>  z <span class="ot">&lt;-</span> <span class="fu">rgamma</span>(n, alpha[<span class="dv">3</span>], <span class="dv">1</span>)</span>
<span id="cb13-9"><a href=""></a>  s <span class="ot">&lt;-</span> x <span class="sc">+</span> y <span class="sc">+</span> z</span>
<span id="cb13-10"><a href=""></a>  <span class="fu">return</span>(<span class="fu">data.frame</span>(<span class="at">x =</span> x<span class="sc">/</span>s, <span class="at">y =</span> y<span class="sc">/</span>s, <span class="at">z =</span> z<span class="sc">/</span>s))</span>
<span id="cb13-11"><a href=""></a>}</span>
<span id="cb13-12"><a href=""></a></span>
<span id="cb13-13"><a href=""></a><span class="co"># Parameters</span></span>
<span id="cb13-14"><a href=""></a>prior_alpha <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">2</span>, <span class="dv">2</span>)  <span class="co"># Symmetric prior</span></span>
<span id="cb13-15"><a href=""></a>true_props <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fl">0.6</span>, <span class="fl">0.3</span>, <span class="fl">0.1</span>)  <span class="co"># True proportions</span></span>
<span id="cb13-16"><a href=""></a>n_samples <span class="ot">&lt;-</span> <span class="dv">50</span></span>
<span id="cb13-17"><a href=""></a></span>
<span id="cb13-18"><a href=""></a><span class="co"># Generate data</span></span>
<span id="cb13-19"><a href=""></a>counts <span class="ot">&lt;-</span> <span class="fu">rmultinom</span>(<span class="dv">1</span>, n_samples, true_props)</span>
<span id="cb13-20"><a href=""></a></span>
<span id="cb13-21"><a href=""></a><span class="co"># Calculate posterior</span></span>
<span id="cb13-22"><a href=""></a>posterior_alpha <span class="ot">&lt;-</span> prior_alpha <span class="sc">+</span> counts</span>
<span id="cb13-23"><a href=""></a></span>
<span id="cb13-24"><a href=""></a><span class="co"># Generate points for visualization</span></span>
<span id="cb13-25"><a href=""></a>n_points <span class="ot">&lt;-</span> <span class="dv">1000</span></span>
<span id="cb13-26"><a href=""></a>prior_points <span class="ot">&lt;-</span> <span class="fu">rdirichlet_2d</span>(n_points, prior_alpha)</span>
<span id="cb13-27"><a href=""></a>posterior_points <span class="ot">&lt;-</span> <span class="fu">rdirichlet_2d</span>(n_points, posterior_alpha)</span>
<span id="cb13-28"><a href=""></a></span>
<span id="cb13-29"><a href=""></a><span class="co"># Combine data</span></span>
<span id="cb13-30"><a href=""></a>prior_points<span class="sc">$</span>Distribution <span class="ot">&lt;-</span> <span class="st">"Prior"</span></span>
<span id="cb13-31"><a href=""></a>posterior_points<span class="sc">$</span>Distribution <span class="ot">&lt;-</span> <span class="st">"Posterior"</span></span>
<span id="cb13-32"><a href=""></a>all_points <span class="ot">&lt;-</span> <span class="fu">rbind</span>(prior_points, posterior_points)</span>
<span id="cb13-33"><a href=""></a></span>
<span id="cb13-34"><a href=""></a><span class="co"># Create ternary plot data</span></span>
<span id="cb13-35"><a href=""></a><span class="co"># We'll use a 2D projection since ggtern might not be available</span></span>
<span id="cb13-36"><a href=""></a>project_ternary <span class="ot">&lt;-</span> <span class="cf">function</span>(df) {</span>
<span id="cb13-37"><a href=""></a>  <span class="co"># Convert to 2D coordinates</span></span>
<span id="cb13-38"><a href=""></a>  df<span class="sc">$</span>X <span class="ot">&lt;-</span> <span class="fl">0.5</span> <span class="sc">*</span> (<span class="dv">2</span> <span class="sc">*</span> df<span class="sc">$</span>y <span class="sc">+</span> df<span class="sc">$</span>z) <span class="sc">/</span> (df<span class="sc">$</span>x <span class="sc">+</span> df<span class="sc">$</span>y <span class="sc">+</span> df<span class="sc">$</span>z)</span>
<span id="cb13-39"><a href=""></a>  df<span class="sc">$</span>Y <span class="ot">&lt;-</span> (<span class="fu">sqrt</span>(<span class="dv">3</span>)<span class="sc">/</span><span class="dv">2</span>) <span class="sc">*</span> df<span class="sc">$</span>z <span class="sc">/</span> (df<span class="sc">$</span>x <span class="sc">+</span> df<span class="sc">$</span>y <span class="sc">+</span> df<span class="sc">$</span>z)</span>
<span id="cb13-40"><a href=""></a>  <span class="fu">return</span>(df)</span>
<span id="cb13-41"><a href=""></a>}</span>
<span id="cb13-42"><a href=""></a></span>
<span id="cb13-43"><a href=""></a>plot_data <span class="ot">&lt;-</span> <span class="fu">project_ternary</span>(all_points)</span>
<span id="cb13-44"><a href=""></a></span>
<span id="cb13-45"><a href=""></a><span class="co"># Add true value</span></span>
<span id="cb13-46"><a href=""></a>true_point <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">x =</span> true_props[<span class="dv">1</span>], <span class="at">y =</span> true_props[<span class="dv">2</span>], <span class="at">z =</span> true_props[<span class="dv">3</span>],</span>
<span id="cb13-47"><a href=""></a>                        <span class="at">Distribution =</span> <span class="st">"True Value"</span>)</span>
<span id="cb13-48"><a href=""></a>true_point <span class="ot">&lt;-</span> <span class="fu">project_ternary</span>(true_point)</span>
<span id="cb13-49"><a href=""></a></span>
<span id="cb13-50"><a href=""></a><span class="co"># Create plot</span></span>
<span id="cb13-51"><a href=""></a><span class="fu">ggplot</span>(plot_data, <span class="fu">aes</span>(<span class="at">x =</span> X, <span class="at">y =</span> Y, <span class="at">color =</span> Distribution)) <span class="sc">+</span></span>
<span id="cb13-52"><a href=""></a>  <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.3</span>, <span class="at">size =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb13-53"><a href=""></a>  <span class="fu">geom_point</span>(<span class="at">data =</span> true_point, <span class="at">color =</span> <span class="st">"black"</span>, <span class="at">size =</span> <span class="dv">5</span>, <span class="at">shape =</span> <span class="dv">8</span>) <span class="sc">+</span></span>
<span id="cb13-54"><a href=""></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"Prior"</span> <span class="ot">=</span> <span class="st">"blue"</span>, <span class="st">"Posterior"</span> <span class="ot">=</span> <span class="st">"red"</span>, <span class="st">"True Value"</span> <span class="ot">=</span> <span class="st">"black"</span>)) <span class="sc">+</span></span>
<span id="cb13-55"><a href=""></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Dirichlet-Multinomial for Multiple Alleles"</span>,</span>
<span id="cb13-56"><a href=""></a>       <span class="at">subtitle =</span> <span class="fu">paste0</span>(<span class="st">"Prior: Dirichlet("</span>, <span class="fu">paste</span>(prior_alpha, <span class="at">collapse =</span> <span class="st">","</span>), </span>
<span id="cb13-57"><a href=""></a>                       <span class="st">"), Counts: "</span>, <span class="fu">paste</span>(counts, <span class="at">collapse =</span> <span class="st">","</span>)),</span>
<span id="cb13-58"><a href=""></a>       <span class="at">x =</span> <span class="st">""</span>, <span class="at">y =</span> <span class="st">""</span>) <span class="sc">+</span></span>
<span id="cb13-59"><a href=""></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"bottom"</span>,</span>
<span id="cb13-60"><a href=""></a>        <span class="at">axis.text =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb13-61"><a href=""></a>        <span class="at">axis.ticks =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span></span>
<span id="cb13-62"><a href=""></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="dv">0</span>, <span class="at">y =</span> <span class="dv">0</span>, <span class="at">label =</span> <span class="st">"A"</span>) <span class="sc">+</span></span>
<span id="cb13-63"><a href=""></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="dv">1</span>, <span class="at">y =</span> <span class="dv">0</span>, <span class="at">label =</span> <span class="st">"B"</span>) <span class="sc">+</span></span>
<span id="cb13-64"><a href=""></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="fl">0.5</span>, <span class="at">y =</span> <span class="fu">sqrt</span>(<span class="dv">3</span>)<span class="sc">/</span><span class="dv">2</span>, <span class="at">label =</span> <span class="st">"C"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>

</div>
<img data-src="mpg6_files/figure-revealjs/unnamed-chunk-14-1.png" width="960" class="r-stretch"></section>
<section id="conjugate-normal-normal-model" class="slide level2">
<h2>Conjugate Normal-Normal Model</h2>
<ul>
<li>One of the most elegant and widely-used conjugate pairs in Bayesian statistics<br>
</li>
<li>Perfect for analyzing quantitative traits in genomics<br>
</li>
<li>Gives us a mathematical shortcut for updating beliefs</li>
</ul>
</section>
<section id="the-setup" class="slide level2">
<h2>The Setup</h2>
<p>When analyzing a continuous parameter <span class="math inline">\(\mu\)</span> (like an effect size):</p>
<ul>
<li><p><strong>Prior</strong>: <span class="math inline">\(\mu \sim \mathcal{N}(\mu_0, \sigma_0^2)\)</span><br>
</p></li>
<li><p><strong>Likelihood</strong>: <span class="math inline">\(X \sim \mathcal{N}(\mu, \sigma^2)\)</span> where <span class="math inline">\(\sigma^2\)</span> is known<br>
</p></li>
</ul>
<p><strong>Question</strong>: What is <span class="math inline">\(p(\mu|X)\)</span>?</p>
</section>
<section id="the-mathematical-magic" class="slide level2">
<h2>The Mathematical Magic</h2>
<p>The elegance is in the algebraic symmetry:</p>
<p><span class="math display">\[
\begin{align}
p(\mu|X) &amp;\propto p(X|\mu) \times p(\mu)\\
&amp;\propto \exp\left(-\frac{(X-\mu)^2}{2\sigma^2}\right) \times \exp\left(-\frac{(\mu-\mu_0)^2}{2\sigma_0^2}\right)
\end{align}
\]</span></p>
<p>Notice the beautiful symmetry: <span class="math inline">\((X-\mu)^2\)</span> in the likelihood and <span class="math inline">\((\mu-\mu_0)^2\)</span> in the prior.</p>
</section>
<section id="the-key-insight" class="slide level2">
<h2>The Key Insight</h2>
<p>When we expand these terms:</p>
<p><span class="math display">\[
\begin{align}
p(\mu|X) &amp;\propto \exp\left(-\frac{1}{2}\left[\frac{(X-\mu)^2}{\sigma^2} + \frac{(\mu-\mu_0)^2}{\sigma_0^2}\right]\right)\\
&amp;\propto \exp\left(-\frac{1}{2}\left[\frac{\mu^2 - 2\mu X + X^2}{\sigma^2} + \frac{\mu^2 - 2\mu\mu_0 + \mu_0^2}{\sigma_0^2}\right]\right)
\end{align}
\]</span></p>
<p>Collecting terms with <span class="math inline">\(\mu^2\)</span> and <span class="math inline">\(\mu\)</span>…</p>
</section>
<section id="the-posterior-formula" class="slide level2">
<h2>The Posterior Formula</h2>
<p>After completing the square, we get:</p>
<p><span class="math display">\[\mu|X \sim \mathcal{N}(\mu_n, \sigma_n^2)\]</span></p>
<p>Where:</p>
<p><span class="math display">\[\mu_n = \frac{\frac{\mu_0}{\sigma_0^2} + \frac{X}{\sigma^2}}{\frac{1}{\sigma_0^2} + \frac{1}{\sigma^2}} = \frac{\sigma^2\mu_0 + \sigma_0^2 X}{\sigma^2 + \sigma_0^2}\]</span></p>
<p><span class="math display">\[\frac{1}{\sigma_n^2} = \frac{1}{\sigma_0^2} + \frac{1}{\sigma^2}\]</span></p>
</section>
<section id="a-more-intuitive-view" class="slide level2">
<h2>A More Intuitive View</h2>
<p>The posterior mean is a <strong>precision-weighted average</strong> of the prior mean and the data:</p>
<p><span class="math display">\[\mu_n = w\mu_0 + (1-w)X\]</span></p>
<p>Where <span class="math inline">\(w = \frac{\sigma^2}{\sigma^2 + \sigma_0^2} = \frac{\text{data precision}}{\text{total precision}}\)</span></p>
<ul>
<li>When data is precise (small <span class="math inline">\(\sigma^2\)</span>): we trust the data more<br>
</li>
<li>When prior is precise (small <span class="math inline">\(\sigma_0^2\)</span>): we trust the prior more</li>
</ul>
</section>
<section id="multiple-observations" class="slide level2">
<h2>Multiple Observations</h2>
<p>With multiple observations <span class="math inline">\(X_1,...,X_n\)</span>, we get:</p>
<p><span class="math display">\[\mu|(X_1,...,X_n) \sim \mathcal{N}\left(\frac{\frac{\mu_0}{\sigma_0^2} + \frac{n\bar{X}}{\sigma^2}}{\frac{1}{\sigma_0^2} + \frac{n}{\sigma^2}}, \left(\frac{1}{\sigma_0^2} + \frac{n}{\sigma^2}\right)^{-1}\right)\]</span></p>
<ul>
<li>The sample mean <span class="math inline">\(\bar{X}\)</span> is a sufficient statistic<br>
</li>
<li>More data increases precision linearly</li>
</ul>
<div class="callout callout-none no-icon callout-style-simple">
<div class="callout-body">
<div class="callout-content">
<p>Genomics Application: eQTL Effect Sizes</p>
<p>In genomics, we might use this model for:</p>
<ul>
<li><strong>Prior</strong>: Historical effect sizes for similar variants<br>
</li>
<li><strong>Likelihood</strong>: Observed effect in current study<br>
</li>
<li><strong>Posterior</strong>: Updated estimate that balances prior knowledge and new data</li>
</ul>
<p>Example: Effect sizes for expression quantitative trait loci (eQTLs)</p>
</div>
</div>
</div>
</section>
<section id="the-power-of-conjugate-priors" class="slide level2">
<h2>The Power of Conjugate Priors</h2>
<p>Advantages of conjugate Normal-Normal:</p>
<ol type="1">
<li><strong>Analytical solutions</strong> – no MCMC required<br>
</li>
<li><strong>Computational efficiency</strong> – critical for genomic scale<br>
</li>
<li><strong>Interpretable updates</strong> – precision-weighted averages<br>
</li>
<li><strong>Sequential processing</strong> – can update one observation at a time</li>
</ol>
<div class="callout callout-none no-icon callout-style-simple">
<div class="callout-body">
<div class="callout-content">
<ul>
<li><strong>Adaptive Shrinkage (ash)</strong>: Uses a mixture of normals as the prior<br>
</li>
<li><strong>Multivariate Adaptive Shrinkage (mash)</strong>: Extends to correlated effects across conditions<br>
</li>
<li><strong>False Discovery Rate Control</strong>: Through local false discovery rates<br>
</li>
<li><strong>Hierarchical Models</strong>: Building multi-level models with partially pooled estimates</li>
</ul>
</div>
</div>
</div>
<aside class="notes">
<ol type="1">
<li>The posterior is also Normal – that’s conjugacy!<br>
</li>
<li>The posterior mean is a weighted average of prior mean and data<br>
</li>
<li>Weights are determined by relative precisions (1/variance)<br>
</li>
<li>The posterior precision is the sum of the prior and data precisions<br>
</li>
<li>This model provides the foundation for many advanced Bayesian genomic methods</li>
</ol>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="extension-empirical-bayes-for-normal-means" class="slide level2">
<h2>Extension: Empirical Bayes for Normal Means</h2>
<ul>
<li>When we don’t have a specific prior, we can <strong>estimate it from the data</strong><br>
</li>
<li>This approach, known as <strong>Empirical Bayes</strong>, is extremely powerful for genomics<br>
</li>
<li>Applications include: multiple testing, sparse signal detection, and effect size estimation</li>
</ul>
</section>
<section class="slide level2">

</section></section>
<section>
<section id="mixture-models-for-complex-data-what-are-mixture-models" class="title-slide slide level1 center">
<h1>4. Mixture Models for Complex Data: What Are Mixture Models?</h1>
<p>Mixture models are probabilistic models that represent the presence of subpopulations within an overall population:</p>
<p><strong>Used when data come from multiple underlying processes</strong><br>
<strong>Represent heterogeneous populations as mixtures of simpler distributions</strong><br>
</p>
<aside class="notes">
<p>Allow clustering without hard assignments Incorporate uncertainty in group membership</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="mixture-model-mathematical-formulation" class="slide level2">
<h2>Mixture Model: Mathematical Formulation</h2>
<p>A mixture model combines multiple distributions to model complex data:</p>
<p><span class="math display">\[p(x) = \sum_{k=1}^K \pi_k f_k(x|\theta_k)\]</span></p>
<p>Where:</p>
<ul>
<li><span class="math inline">\(p(x)\)</span> is the overall probability density<br>
</li>
<li><span class="math inline">\(K\)</span> is the number of components (subpopulations)<br>
</li>
<li><span class="math inline">\(\pi_k\)</span> are the mixing weights (<span class="math inline">\(\sum_{k=1}^K \pi_k = 1\)</span>)<br>
</li>
<li><span class="math inline">\(f_k(x|\theta_k)\)</span> are the component densities with parameters <span class="math inline">\(\theta_k\)</span></li>
</ul>
</section>
<section id="a-closer-look-at-the-components" class="slide level2">
<h2>A Closer Look at the Components</h2>
<div class="columns">
<div class="column" style="width:40%;">
<p><strong>Key components</strong>:</p>
<ol type="1">
<li><strong>Component distributions</strong> <span class="math inline">\(f_k(x|\theta_k)\)</span> Subpopulation likelihood</li>
<li><strong>Mixing weights</strong> <span class="math inline">\(\pi_k\)</span> <!-- -   Proportion of data from each component\ --> Reflect prior probabilities of group membership</li>
<li><strong>Latent variables</strong> <span class="math inline">\(z_i\)</span> Unobserved component membership <!-- -   $z_i = k$ means data point $i$ came from component $k$ --></li>
</ol>
</div><div class="column" style="width:60%;">
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href=""></a><span class="co"># Create a simple mixture model visualization</span></span>
<span id="cb14-2"><a href=""></a><span class="fu">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb14-3"><a href=""></a></span>
<span id="cb14-4"><a href=""></a><span class="co"># Define mixture parameters</span></span>
<span id="cb14-5"><a href=""></a>means <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="sc">-</span><span class="fl">2.5</span>, <span class="fl">1.5</span>)</span>
<span id="cb14-6"><a href=""></a>sds <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fl">0.7</span>, <span class="fl">1.2</span>)</span>
<span id="cb14-7"><a href=""></a>weights <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fl">0.4</span>, <span class="fl">0.6</span>)</span>
<span id="cb14-8"><a href=""></a></span>
<span id="cb14-9"><a href=""></a><span class="co"># Generate data from mixture</span></span>
<span id="cb14-10"><a href=""></a>n_samples <span class="ot">&lt;-</span> <span class="dv">1000</span></span>
<span id="cb14-11"><a href=""></a>component <span class="ot">&lt;-</span> <span class="fu">sample</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>, n_samples, <span class="at">replace =</span> <span class="cn">TRUE</span>, <span class="at">prob =</span> weights)</span>
<span id="cb14-12"><a href=""></a>x <span class="ot">&lt;-</span> <span class="fu">rnorm</span>(n_samples, <span class="at">mean =</span> means[component], <span class="at">sd =</span> sds[component])</span>
<span id="cb14-13"><a href=""></a></span>
<span id="cb14-14"><a href=""></a><span class="co"># Create data frame for histogram</span></span>
<span id="cb14-15"><a href=""></a>hist_data <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">x =</span> x, <span class="at">component =</span> <span class="fu">factor</span>(component))</span>
<span id="cb14-16"><a href=""></a></span>
<span id="cb14-17"><a href=""></a><span class="co"># Create data for component densities</span></span>
<span id="cb14-18"><a href=""></a>x_seq <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="sc">-</span><span class="dv">6</span>, <span class="dv">6</span>, <span class="at">length.out =</span> <span class="dv">500</span>)</span>
<span id="cb14-19"><a href=""></a>comp1_density <span class="ot">&lt;-</span> weights[<span class="dv">1</span>] <span class="sc">*</span> <span class="fu">dnorm</span>(x_seq, <span class="at">mean =</span> means[<span class="dv">1</span>], <span class="at">sd =</span> sds[<span class="dv">1</span>])</span>
<span id="cb14-20"><a href=""></a>comp2_density <span class="ot">&lt;-</span> weights[<span class="dv">2</span>] <span class="sc">*</span> <span class="fu">dnorm</span>(x_seq, <span class="at">mean =</span> means[<span class="dv">2</span>], <span class="at">sd =</span> sds[<span class="dv">2</span>])</span>
<span id="cb14-21"><a href=""></a>mixture_density <span class="ot">&lt;-</span> comp1_density <span class="sc">+</span> comp2_density</span>
<span id="cb14-22"><a href=""></a></span>
<span id="cb14-23"><a href=""></a>density_data <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb14-24"><a href=""></a>  <span class="at">x =</span> <span class="fu">rep</span>(x_seq, <span class="dv">3</span>),</span>
<span id="cb14-25"><a href=""></a>  <span class="at">y =</span> <span class="fu">c</span>(comp1_density, comp2_density, mixture_density),</span>
<span id="cb14-26"><a href=""></a>  <span class="at">Component =</span> <span class="fu">factor</span>(<span class="fu">rep</span>(<span class="fu">c</span>(<span class="st">"Component 1"</span>, <span class="st">"Component 2"</span>, <span class="st">"Mixture"</span>), <span class="at">each =</span> <span class="fu">length</span>(x_seq)))</span>
<span id="cb14-27"><a href=""></a>)</span>
<span id="cb14-28"><a href=""></a></span>
<span id="cb14-29"><a href=""></a><span class="co"># Create plot</span></span>
<span id="cb14-30"><a href=""></a><span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb14-31"><a href=""></a>  <span class="fu">geom_histogram</span>(<span class="at">data =</span> hist_data, <span class="fu">aes</span>(<span class="at">x =</span> x, <span class="at">y =</span> <span class="fu">after_stat</span>(density), <span class="at">fill =</span> component), </span>
<span id="cb14-32"><a href=""></a>                <span class="at">bins =</span> <span class="dv">50</span>, <span class="at">alpha =</span> <span class="fl">0.5</span>, <span class="at">position =</span> <span class="st">"identity"</span>) <span class="sc">+</span></span>
<span id="cb14-33"><a href=""></a>  <span class="fu">geom_line</span>(<span class="at">data =</span> density_data, <span class="fu">aes</span>(<span class="at">x =</span> x, <span class="at">y =</span> y, <span class="at">color =</span> Component), <span class="at">size =</span> <span class="fl">1.2</span>) <span class="sc">+</span></span>
<span id="cb14-34"><a href=""></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"lightblue"</span>, <span class="st">"lightgreen"</span>), <span class="at">name =</span> <span class="st">"True Component"</span>) <span class="sc">+</span></span>
<span id="cb14-35"><a href=""></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"blue"</span>, <span class="st">"darkgreen"</span>, <span class="st">"red"</span>)) <span class="sc">+</span></span>
<span id="cb14-36"><a href=""></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Gaussian Mixture Model"</span>,</span>
<span id="cb14-37"><a href=""></a>       <span class="at">subtitle =</span> <span class="st">"Two underlying subpopulations create a complex distribution"</span>,</span>
<span id="cb14-38"><a href=""></a>       <span class="at">x =</span> <span class="st">"Value"</span>, </span>
<span id="cb14-39"><a href=""></a>       <span class="at">y =</span> <span class="st">"Density"</span>) <span class="sc">+</span></span>
<span id="cb14-40"><a href=""></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"bottom"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure>
<p><img data-src="mpg6_files/figure-revealjs/unnamed-chunk-15-1.png" width="960"></p>
</figure>
</div>
</div>
</div>
</div></div>
</section>
<section id="likelihood-function-for-mixture-models" class="slide level2">
<h2>Likelihood Function for Mixture Models</h2>
<p>The likelihood of a mixture model for <span class="math inline">\(n\)</span> independent observations <span class="math inline">\(x_1, \ldots, x_n\)</span> is:</p>
<p><span class="math display">\[L(\theta, \pi | x_1, \ldots, x_n) = \prod_{i=1}^n p(x_i) = \prod_{i=1}^n \sum_{k=1}^K \pi_k f_k(x_i|\theta_k)\]</span></p>
<p><strong>Challenge</strong>: The sum inside the product makes this difficult to optimize directly</p>
<p><strong>Solution</strong>: Introduce latent variables <span class="math inline">\(z_i\)</span> and use the EM algorithm</p>
</section>
<section id="the-em-algorithm-in-detail" class="slide level2">
<h2>The EM Algorithm in Detail</h2>
<p>The Expectation-Maximization (EM) algorithm is an iterative method for finding maximum likelihood estimates:</p>
<p><strong>E-step</strong>: Calculate “responsibilities” – the posterior probability that data point <span class="math inline">\(i\)</span> belongs to component <span class="math inline">\(k\)</span>:</p>
<p><span class="math display">\[\gamma_{ik} = P(z_i = k | x_i, \theta) = \frac{\pi_k f_k(x_i|\theta_k)}{\sum_{j=1}^K \pi_j f_j(x_i|\theta_j)}\]</span></p>
<p><strong>M-step</strong>: Update parameters using weighted maximum likelihood:</p>
<p><span class="math display">\[\pi_k^{new} = \frac{1}{n}\sum_{i=1}^n \gamma_{ik}\]</span><br>
<span class="math display">\[\theta_k^{new} = \arg\max_{\theta_k} \sum_{i=1}^n \gamma_{ik} \log f_k(x_i|\theta_k)\]</span></p>
</section>
<section id="em-algorithm-step-by-step-example" class="slide level2">
<h2>EM Algorithm: Step-by-Step Example</h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href=""></a><span class="co"># Simplified EM visualization</span></span>
<span id="cb15-2"><a href=""></a><span class="fu">set.seed</span>(<span class="dv">789</span>)</span>
<span id="cb15-3"><a href=""></a></span>
<span id="cb15-4"><a href=""></a><span class="co"># Generate data from a mixture of two Gaussians</span></span>
<span id="cb15-5"><a href=""></a>n <span class="ot">&lt;-</span> <span class="dv">200</span></span>
<span id="cb15-6"><a href=""></a>true_means <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">2</span>, <span class="dv">2</span>)</span>
<span id="cb15-7"><a href=""></a>true_sds <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fl">0.8</span>, <span class="fl">0.8</span>)</span>
<span id="cb15-8"><a href=""></a>true_weights <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fl">0.4</span>, <span class="fl">0.6</span>)</span>
<span id="cb15-9"><a href=""></a></span>
<span id="cb15-10"><a href=""></a>z <span class="ot">&lt;-</span> <span class="fu">sample</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>, n, <span class="at">replace =</span> <span class="cn">TRUE</span>, <span class="at">prob =</span> true_weights)</span>
<span id="cb15-11"><a href=""></a>x <span class="ot">&lt;-</span> <span class="fu">rnorm</span>(n, <span class="at">mean =</span> true_means[z], <span class="at">sd =</span> true_sds[z])</span>
<span id="cb15-12"><a href=""></a></span>
<span id="cb15-13"><a href=""></a><span class="co"># Create data for visualization</span></span>
<span id="cb15-14"><a href=""></a>iterations <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Initial"</span>, <span class="st">"Iteration 1"</span>, <span class="st">"Iteration 3"</span>, <span class="st">"Final"</span>)</span>
<span id="cb15-15"><a href=""></a>means1 <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>, <span class="sc">-</span><span class="fl">1.5</span>, <span class="sc">-</span><span class="fl">1.8</span>, <span class="sc">-</span><span class="fl">2.0</span>)</span>
<span id="cb15-16"><a href=""></a>means2 <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="fl">1.4</span>, <span class="fl">1.8</span>, <span class="fl">2.0</span>)</span>
<span id="cb15-17"><a href=""></a>sds1 <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fl">1.2</span>, <span class="fl">1.0</span>, <span class="fl">0.9</span>, <span class="fl">0.8</span>)</span>
<span id="cb15-18"><a href=""></a>sds2 <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fl">1.2</span>, <span class="fl">1.0</span>, <span class="fl">0.9</span>, <span class="fl">0.8</span>)</span>
<span id="cb15-19"><a href=""></a>weights1 <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fl">0.5</span>, <span class="fl">0.45</span>, <span class="fl">0.42</span>, <span class="fl">0.4</span>)</span>
<span id="cb15-20"><a href=""></a>weights2 <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fl">0.5</span>, <span class="fl">0.55</span>, <span class="fl">0.58</span>, <span class="fl">0.6</span>)</span>
<span id="cb15-21"><a href=""></a></span>
<span id="cb15-22"><a href=""></a><span class="co"># Create data frame for plotting</span></span>
<span id="cb15-23"><a href=""></a>plot_data <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb15-24"><a href=""></a>x_seq <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="sc">-</span><span class="dv">5</span>, <span class="dv">5</span>, <span class="at">length.out =</span> <span class="dv">200</span>)</span>
<span id="cb15-25"><a href=""></a></span>
<span id="cb15-26"><a href=""></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(iterations)) {</span>
<span id="cb15-27"><a href=""></a>  <span class="co"># Component 1</span></span>
<span id="cb15-28"><a href=""></a>  comp1 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb15-29"><a href=""></a>    <span class="at">x =</span> x_seq,</span>
<span id="cb15-30"><a href=""></a>    <span class="at">y =</span> weights1[i] <span class="sc">*</span> <span class="fu">dnorm</span>(x_seq, <span class="at">mean =</span> means1[i], <span class="at">sd =</span> sds1[i]),</span>
<span id="cb15-31"><a href=""></a>    <span class="at">Component =</span> <span class="st">"Component 1"</span>,</span>
<span id="cb15-32"><a href=""></a>    <span class="at">Iteration =</span> iterations[i]</span>
<span id="cb15-33"><a href=""></a>  )</span>
<span id="cb15-34"><a href=""></a>  </span>
<span id="cb15-35"><a href=""></a>  <span class="co"># Component 2</span></span>
<span id="cb15-36"><a href=""></a>  comp2 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb15-37"><a href=""></a>    <span class="at">x =</span> x_seq,</span>
<span id="cb15-38"><a href=""></a>    <span class="at">y =</span> weights2[i] <span class="sc">*</span> <span class="fu">dnorm</span>(x_seq, <span class="at">mean =</span> means2[i], <span class="at">sd =</span> sds2[i]),</span>
<span id="cb15-39"><a href=""></a>    <span class="at">Component =</span> <span class="st">"Component 2"</span>,</span>
<span id="cb15-40"><a href=""></a>    <span class="at">Iteration =</span> iterations[i]</span>
<span id="cb15-41"><a href=""></a>  )</span>
<span id="cb15-42"><a href=""></a>  </span>
<span id="cb15-43"><a href=""></a>  <span class="co"># Mixture</span></span>
<span id="cb15-44"><a href=""></a>  mixture <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb15-45"><a href=""></a>    <span class="at">x =</span> x_seq,</span>
<span id="cb15-46"><a href=""></a>    <span class="at">y =</span> comp1<span class="sc">$</span>y <span class="sc">+</span> comp2<span class="sc">$</span>y,</span>
<span id="cb15-47"><a href=""></a>    <span class="at">Component =</span> <span class="st">"Mixture"</span>,</span>
<span id="cb15-48"><a href=""></a>    <span class="at">Iteration =</span> iterations[i]</span>
<span id="cb15-49"><a href=""></a>  )</span>
<span id="cb15-50"><a href=""></a>  </span>
<span id="cb15-51"><a href=""></a>  plot_data <span class="ot">&lt;-</span> <span class="fu">rbind</span>(plot_data, comp1, comp2, mixture)</span>
<span id="cb15-52"><a href=""></a>}</span>
<span id="cb15-53"><a href=""></a></span>
<span id="cb15-54"><a href=""></a><span class="co"># Create histogram data</span></span>
<span id="cb15-55"><a href=""></a>hist_data <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">x =</span> x, <span class="at">Iteration =</span> <span class="st">"Data"</span>)</span>
<span id="cb15-56"><a href=""></a></span>
<span id="cb15-57"><a href=""></a><span class="co"># Create plot</span></span>
<span id="cb15-58"><a href=""></a><span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb15-59"><a href=""></a>  <span class="co"># Add histogram for data</span></span>
<span id="cb15-60"><a href=""></a>  <span class="fu">geom_histogram</span>(<span class="at">data =</span> hist_data, <span class="fu">aes</span>(<span class="at">x =</span> x, <span class="at">y =</span> <span class="fu">after_stat</span>(density)), </span>
<span id="cb15-61"><a href=""></a>                <span class="at">bins =</span> <span class="dv">30</span>, <span class="at">fill =</span> <span class="st">"gray80"</span>, <span class="at">color =</span> <span class="st">"black"</span>, <span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb15-62"><a href=""></a>  <span class="co"># Add component and mixture densities</span></span>
<span id="cb15-63"><a href=""></a>  <span class="fu">geom_line</span>(<span class="at">data =</span> plot_data, </span>
<span id="cb15-64"><a href=""></a>           <span class="fu">aes</span>(<span class="at">x =</span> x, <span class="at">y =</span> y, <span class="at">color =</span> Component, <span class="at">group =</span> <span class="fu">interaction</span>(Component, Iteration)),</span>
<span id="cb15-65"><a href=""></a>           <span class="at">size =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb15-66"><a href=""></a>  <span class="co"># Facet by iteration</span></span>
<span id="cb15-67"><a href=""></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span> Iteration, <span class="at">ncol =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb15-68"><a href=""></a>  <span class="co"># Customize colors</span></span>
<span id="cb15-69"><a href=""></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"Component 1"</span> <span class="ot">=</span> <span class="st">"blue"</span>, <span class="st">"Component 2"</span> <span class="ot">=</span> <span class="st">"green"</span>, <span class="st">"Mixture"</span> <span class="ot">=</span> <span class="st">"red"</span>)) <span class="sc">+</span></span>
<span id="cb15-70"><a href=""></a>  <span class="co"># Add labels</span></span>
<span id="cb15-71"><a href=""></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"EM Algorithm Convergence"</span>,</span>
<span id="cb15-72"><a href=""></a>       <span class="at">subtitle =</span> <span class="st">"Mixture model fit improves with each iteration"</span>,</span>
<span id="cb15-73"><a href=""></a>       <span class="at">x =</span> <span class="st">"Value"</span>, </span>
<span id="cb15-74"><a href=""></a>       <span class="at">y =</span> <span class="st">"Density"</span>) <span class="sc">+</span></span>
<span id="cb15-75"><a href=""></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb15-76"><a href=""></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"bottom"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>

</div>
<img data-src="mpg6_files/figure-revealjs/unnamed-chunk-16-1.png" width="960" class="r-stretch"></section>
<section id="worked-example-em-algorithm-step-by-step" class="slide level2">
<h2>Worked Example: EM Algorithm Step-by-Step</h2>
<p>Let’s walk through each step of the EM algorithm for a mixture of two Gaussians:</p>
<ol type="1">
<li><strong>Initialize parameters</strong>:
<ul>
<li>Set initial mixing weights: <span class="math inline">\(\pi_1 = \pi_2 = 0.5\)</span><br>
</li>
<li>Set initial component means: <span class="math inline">\(\mu_1 = -1, \mu_2 = 1\)</span><br>
</li>
<li>Set initial component standard deviations: <span class="math inline">\(\sigma_1 = \sigma_2 = 1\)</span></li>
</ul></li>
<li><strong>E-step</strong>: For each data point <span class="math inline">\(x_i\)</span>, calculate the responsibility of each component:
<ul>
<li><span class="math inline">\(\gamma_{i1} = \frac{\pi_1 N(x_i|\mu_1,\sigma_1^2)}{\pi_1 N(x_i|\mu_1,\sigma_1^2) + \pi_2 N(x_i|\mu_2,\sigma_2^2)}\)</span><br>
</li>
<li><span class="math inline">\(\gamma_{i2} = 1 - \gamma_{i1}\)</span></li>
</ul></li>
</ol>
</section>
<section id="worked-example-em-algorithm-step-by-step-1" class="slide level2">
<h2>Worked Example: EM Algorithm Step-by-Step</h2>
<ol start="3" type="1">
<li><strong>M-step</strong>: Update the parameters using the responsibilities:
<ul>
<li><span class="math inline">\(\pi_1^{new} = \frac{1}{n}\sum_{i=1}^n \gamma_{i1}\)</span> (similarly for <span class="math inline">\(\pi_2^{new}\)</span>)<br>
</li>
<li><span class="math inline">\(\mu_1^{new} = \frac{\sum_{i=1}^n \gamma_{i1}x_i}{\sum_{i=1}^n \gamma_{i1}}\)</span> (similarly for <span class="math inline">\(\mu_2^{new}\)</span>)<br>
</li>
<li><span class="math inline">\((\sigma_1^{new})^2 = \frac{\sum_{i=1}^n \gamma_{i1}(x_i-\mu_1^{new})^2}{\sum_{i=1}^n \gamma_{i1}}\)</span> (similarly for <span class="math inline">\(\sigma_2^{new}\)</span>)</li>
</ul></li>
<li><strong>Repeat</strong> until convergence (parameters stop changing significantly) (link interlude <a href="https://surbut.shinyapps.io/shinyP/" class="uri">https://surbut.shinyapps.io/shinyP/</a> with cool Bayesian intervals)</li>
</ol>
<aside class="notes">
<p>The EM Algorithm: Mathematical Intuition</p>
<p>Key insight: We’re solving a chicken-and-egg problem<br>
If we knew component assignments, parameter estimation would be easy<br>
If we knew parameters, component assignments would be easy<br>
EM iteratively solves both by using expected assignments</p>
<p>E-step (Expectation): Calculate expected component memberships<br>
<span class="math display">\[\gamma_{ik} = P(z_i = k | x_i, \theta^{(t)}) = \frac{\pi_k^{(t)} f_k(x_i|\theta_k^{(t)})}{\sum_{j=1}^K \pi_j^{(t)} f_j(x_i|\theta_j^{(t)})}\]</span><br>
Intuition: “How likely is individual i to belong to population k, given our current parameter estimates?”</p>
<p>M-step (Maximization): Update parameters using weighted averages<br>
For mixing weights:<br>
<span class="math display">\[\pi_k^{(t+1)} = \frac{1}{n}\sum_{i=1}^n \gamma_{ik}\]</span><br>
Intuition: “The new population frequency is the average membership across all individuals”</p>
<p>For component means (Gaussian case):<br>
<span class="math display">\[\mu_k^{(t+1)} = \frac{\sum_{i=1}^n \gamma_{ik}x_i}{\sum_{i=1}^n \gamma_{ik}}\]</span><br>
Intuition: “The new population mean is a weighted average where individuals are weighted by their probability of belonging to this population (link interlude <a href="https://surbut.shinyapps.io/shinyP/" class="uri">https://surbut.shinyapps.io/shinyP/</a> )</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="bayesian-mixture-models" class="slide level2">
<h2>Bayesian Mixture Models</h2>
<p>Bayesian mixture models add priors to the parameters: <span class="math inline">\(p(\theta, \pi | x_1, \ldots, x_n) \propto p(x_1, \ldots, x_n | \theta, \pi) \times p(\theta, \pi)\)</span></p>
<h4 id="common-prior-choices">Common prior choices:<br>
</h4>
<p><span class="math inline">\(\pi \sim \text{Dirichlet}(\alpha_1, \ldots, \alpha_K)\)</span> for mixing weights<br>
<strong>Component-specific priors for</strong> <span class="math inline">\(\theta_k\)</span> (e.g., Normal-Inverse-Gamma for Gaussian components)</p>
<p><strong>Advantages</strong>:<br>
- Handle <strong>uncertainty</strong><br>
- Allow for informed priors from previous studies<br>
- Provide full posterior distribution rather than point estimates</p>
</section>
<section id="mixture-model-applications-in-genomics" class="slide level2">
<h2>Mixture Model Applications in Genomics</h2>
<p><strong>1. Population Structure</strong><br>
- Components = ancestral populations<br>
- Individual genotypes = admixtures of populations<br>
- Example: STRUCTURE, ADMIXTURE software<br>
- Used for: demographic history, association studies, conservation</p>
<p><strong>2. Genetic effect estimation</strong><br>
- (e.g., adaptive shrinkage methods like ash/mash for multiple conditions)</p>
<p><strong>3. Gene Expression Clustering</strong><br>
- Components = cell types/states<br>
- Expression patterns = signatures of cell types<br>
- Example: Single-cell RNA-seq clustering, stay tuned …<br>
</p>
</section>
<section id="why-science-is-still-beautiful" class="slide level2">
<h2>Why science is still beautiful …</h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href=""></a><span class="co"># Create a visualization of population structure</span></span>
<span id="cb16-2"><a href=""></a><span class="fu">set.seed</span>(<span class="dv">456</span>)</span>
<span id="cb16-3"><a href=""></a></span>
<span id="cb16-4"><a href=""></a><span class="co"># Number of individuals and populations</span></span>
<span id="cb16-5"><a href=""></a>n_ind <span class="ot">&lt;-</span> <span class="dv">60</span></span>
<span id="cb16-6"><a href=""></a>n_pop <span class="ot">&lt;-</span> <span class="dv">3</span></span>
<span id="cb16-7"><a href=""></a></span>
<span id="cb16-8"><a href=""></a><span class="co"># Create admixture proportions</span></span>
<span id="cb16-9"><a href=""></a><span class="co"># First 20 individuals mostly from pop1, etc.</span></span>
<span id="cb16-10"><a href=""></a>admixture <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="dv">0</span>, <span class="at">nrow =</span> n_ind, <span class="at">ncol =</span> n_pop)</span>
<span id="cb16-11"><a href=""></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>n_ind) {</span>
<span id="cb16-12"><a href=""></a>  <span class="cf">if</span> (i <span class="sc">&lt;=</span> <span class="dv">20</span>) {</span>
<span id="cb16-13"><a href=""></a>    admixture[i,] <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fl">0.8</span>, <span class="fl">0.1</span>, <span class="fl">0.1</span>) <span class="sc">+</span> <span class="fu">rnorm</span>(<span class="dv">3</span>, <span class="dv">0</span>, <span class="fl">0.05</span>)</span>
<span id="cb16-14"><a href=""></a>  } <span class="cf">else</span> <span class="cf">if</span> (i <span class="sc">&lt;=</span> <span class="dv">40</span>) {</span>
<span id="cb16-15"><a href=""></a>    admixture[i,] <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fl">0.1</span>, <span class="fl">0.8</span>, <span class="fl">0.1</span>) <span class="sc">+</span> <span class="fu">rnorm</span>(<span class="dv">3</span>, <span class="dv">0</span>, <span class="fl">0.05</span>)</span>
<span id="cb16-16"><a href=""></a>  } <span class="cf">else</span> {</span>
<span id="cb16-17"><a href=""></a>    admixture[i,] <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fl">0.1</span>, <span class="fl">0.1</span>, <span class="fl">0.8</span>) <span class="sc">+</span> <span class="fu">rnorm</span>(<span class="dv">3</span>, <span class="dv">0</span>, <span class="fl">0.05</span>)</span>
<span id="cb16-18"><a href=""></a>  }</span>
<span id="cb16-19"><a href=""></a>  <span class="co"># Ensure proportions are positive and sum to 1</span></span>
<span id="cb16-20"><a href=""></a>  admixture[i,] <span class="ot">&lt;-</span> <span class="fu">pmax</span>(admixture[i,], <span class="dv">0</span>)</span>
<span id="cb16-21"><a href=""></a>  admixture[i,] <span class="ot">&lt;-</span> admixture[i,] <span class="sc">/</span> <span class="fu">sum</span>(admixture[i,])</span>
<span id="cb16-22"><a href=""></a>}</span>
<span id="cb16-23"><a href=""></a></span>
<span id="cb16-24"><a href=""></a><span class="co"># Create data frame for plotting</span></span>
<span id="cb16-25"><a href=""></a>admix_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb16-26"><a href=""></a>  <span class="at">Individual =</span> <span class="fu">rep</span>(<span class="dv">1</span><span class="sc">:</span>n_ind, n_pop),</span>
<span id="cb16-27"><a href=""></a>  <span class="at">Population =</span> <span class="fu">factor</span>(<span class="fu">rep</span>(<span class="fu">paste0</span>(<span class="st">"Pop"</span>, <span class="dv">1</span><span class="sc">:</span>n_pop), <span class="at">each =</span> n_ind)),</span>
<span id="cb16-28"><a href=""></a>  <span class="at">Proportion =</span> <span class="fu">c</span>(admixture)</span>
<span id="cb16-29"><a href=""></a>)</span>
<span id="cb16-30"><a href=""></a></span>
<span id="cb16-31"><a href=""></a><span class="co"># Create barplot</span></span>
<span id="cb16-32"><a href=""></a><span class="fu">ggplot</span>(admix_df, <span class="fu">aes</span>(<span class="at">x =</span> Individual, <span class="at">y =</span> Proportion, <span class="at">fill =</span> Population)) <span class="sc">+</span></span>
<span id="cb16-33"><a href=""></a>  <span class="fu">geom_col</span>(<span class="at">width =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb16-34"><a href=""></a>  <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">"Set1"</span>) <span class="sc">+</span></span>
<span id="cb16-35"><a href=""></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Population Structure as a Mixture Model"</span>,</span>
<span id="cb16-36"><a href=""></a>       <span class="at">subtitle =</span> <span class="st">"Each individual is a mixture of ancestral populations"</span>,</span>
<span id="cb16-37"><a href=""></a>       <span class="at">x =</span> <span class="st">"Individual"</span>, </span>
<span id="cb16-38"><a href=""></a>       <span class="at">y =</span> <span class="st">"Ancestry Proportion"</span>) <span class="sc">+</span></span>
<span id="cb16-39"><a href=""></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb16-40"><a href=""></a>        <span class="at">axis.ticks.x =</span> <span class="fu">element_blank</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>

</div>
<img data-src="mpg6_files/figure-revealjs/unnamed-chunk-17-1.png" width="960" class="r-stretch"></section>
<section id="the-structure-model-in-detail" class="slide level2">
<h2>The STRUCTURE Model in Detail</h2>
<p><strong>STRUCTURE</strong>: A Bayesian mixture model for population genetics</p>
<p><strong>Key components</strong>:</p>
<p>Each <em>individual</em> = mixture of <span class="math inline">\(K\)</span> ancestral populations<br>
Each <em>population</em> = distinct allele frequencies<br>
<strong>Goal</strong>: Infer ancestry proportions &amp; population frequencies from observed allele counts</p>
<p><strong>Bayesian formulation</strong>:<br>
<strong>Prior</strong>: <span class="math inline">\(q_{ik} \sim \text{Dirichlet}(\alpha)\)</span> ancestry proportions<br>
<strong>Prior</strong>: <span class="math inline">\(f_{kj} \sim \text{Beta}(\lambda)\)</span> allele frequencices<br>
<strong>Likelihood</strong>: <span class="math inline">\(P(X_{ij} | q_i, f_j)\)</span> conditional likelihood of observed allele, given parameters</p>
<aside class="notes">
<p>Our Data (Haploid Genotypes): Individual M1 M2 M3 * Ind1: A C A * Ind2: G T G * Ind3: A T A We want to find: K=2 ancestral populations q (ancestry) ~ Dirichlet - Because proportions sum to 1 across K populations - K parameters (one for each population) ::: notes - NOT related to number of allele types ::: notes $f (frequencies) ~ Beta $ - Because each marker has 2 possible alleles - Two parameters (success/failure for that allele) - One Beta distribution per marker per population</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="latent-dirichlet-allocation" class="slide level2">
<h2>Latent Dirichlet Allocation</h2>
<p>Topic Models (LDA):</p>
<p>- Document mixture proportions θ ~ Dirichlet(α)</p>
<p>- Topic-word distributions φ ~ Dirichlet(β)</p>
<p>- Words drawn from topics z ~ Multinomial(θ)</p>
<p><strong>STRUCTURE:</strong></p>
<p>- Individual ancestry proportions q ~ Dirichlet(α)</p>
<p>- Population allele frequencies p ~ Dirichlet(β)</p>
<p>- Alleles drawn from populations z ~ Multinomial(q)</p>
</section>
<section id="visualization-if-we-know-the-truth" class="slide level2">
<h2>Visualization: If we know the truth</h2>

<img data-src="mpg6_files/figure-revealjs/unnamed-chunk-18-1.png" width="576" class="r-stretch"></section>
<section id="understanding-the-q-update-at-the-individual-level" class="slide level2">
<h2>Understanding the q Update: At the Individual Level</h2>
<p><strong>Individual 1</strong>: A C A</p>
<p><span class="math inline">\(\log(P(Data|Pop1)) = \log(f_1^A) + \log(f_1^C) + \log(f_1^A)\)</span></p>
<p><span class="math inline">\(\log(P(Data|Pop2)) = \log(f_2^A) + \log(f_2^C) + \log(f_2^A)\)</span></p>
<p>Total for Pop1 = <span class="math inline">\(L_{11} + L_{12} + L_{13}\)</span><br>
Total for Pop1 = <span class="math inline">\(L_{21} + L_{22} + L_{23}\)</span></p>
<aside class="notes">
<p>Here, we’re updating the ancestry proportions (<code>q</code>) for Individual 1. For each marker, we calculate the likelihood of the observed allele under each population’s allele frequency. For example, for Marker 1 (A), we look up the probability of A in Population 1 (<span class="math inline">\(f_1^A\)</span>) and Population 2 (<span class="math inline">\(f_2^A\)</span>). We do this for all markers, summing the log-likelihoods (or likelihoods, depending on the implementation) for each population. These sums are then added to the prior Dirichlet parameters to get the updated parameters for the Dirichlet distribution. The result is a new estimate of Individual 1’s ancestry proportions, reflecting both the prior and the observed data. This process is repeated for each individual in the dataset</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="step-2-dirichlet-update" class="slide level2">
<h2>Step 2: Dirichlet Update</h2>
<p><span class="math display">\[q_1 \sim \text{Dirichlet}(\alpha1 + \sum{L1}, \alpha2 +\sum{L2})\]</span> where:</p>
<p><strong>-</strong> <span class="math inline">\(\sum{L1} = L_{11} + L_{21} + L_{31}\)</span> (added to <span class="math inline">\(\alpha_1\)</span>)<br>
<strong>-</strong> <span class="math inline">\(\sum{L2} = L_{12} + L_{22} + L_{32}\)</span> (added to <span class="math inline">\(\alpha_2\)</span>)</p>
<div class="callout callout-note callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<p><strong>Note</strong></p>
</div>
<div class="callout-content">
<p>The Dirichlet update incorporates log likelihood contributions from both populations to estimate how much of Individual 1’s ancestry comes from each population.</p>
</div>
</div>
</div>
<aside class="notes">
<p>The key difference between this and just adding counts is Simple case: We KNOW which population (just one!) each allele belongs to STRUCTURE case: We DON’T know which population each allele belongs to, so we:Calculate how likely the sequence is under each population Use these likelihoods to update our beliefs about population membership Then use these membership probabilities (q) to update allele frequencies (f). Conjugate Dirichlet-Multinomial Special Case: Prior: Dirichlet(α₁, α₂, α₃) Likelihood: Multinomial(n₁, n₂, n₃) Posterior: Dirichlet(α₁ + n₁, α₂ + n₂, α₃ + n₃) The beauty of conjugate priors is that this complex calculation: adding counts in the conjugate case IS doing likelihood × prior, it’s just a mathematically convenient way to get there!</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="updating-allele-frequencies-f-with-fixed-ancestry-q" class="slide level2">
<h2>Updating Allele Frequencies (f) with Fixed Ancestry (q)</h2>
<p><strong>Marker 1 (Allele A)</strong></p>
<p><strong>Individual 1’s ancestry</strong>: <span class="math inline">\(q_1 = (0.7, 0.3)\)</span></p>
<ul>
<li><p><strong>Population 1</strong>: - Contribution from Individual 1: 0.7 (from <span class="math inline">\(q_{11}\)</span>)<br>
Update: <span class="math inline">\(f_1^A \sim \text{Beta}(\lambda + 0.7, \lambda + 0.3)\)</span></p></li>
<li><p><strong>Population 2</strong>: - Contribution from Individual 1: 0.3 (from <span class="math inline">\(q_{12}\)</span>) Update: <span class="math inline">\(f_2^A \sim \text{Beta}(\lambda + 0.3, \lambda + 0.7)\)</span></p></li>
</ul>
</section>
<section id="marker-2-allele-c" class="slide level2">
<h2>Marker 2 (Allele C)</h2>
<p><strong>Population 1</strong> - Contribution from Individual 1: 0.7 (from <span class="math inline">\(q_{11}\)</span>)<br>
- Update: <span class="math inline">\(f_1^C \sim \text{Beta}(\lambda + 0.7, \lambda + 0.3)\)</span><br>
</p>
<p><strong>Population 2</strong>: - Contribution from Individual 1: 0.3 (from <span class="math inline">\(q_{12}\)</span>)<br>
- Update: <span class="math inline">\(f_2^C \sim \text{Beta}(\lambda + 0.3, \lambda + 0.7)\)</span></p>
<div class="callout callout-note callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<p><strong>Note</strong></p>
</div>
<div class="callout-content">
<p>Note that each allele contributes fractionally to each population’s frequency estimate, weighted by the individual’s ancestry proportions.</p>
</div>
</div>
</div>
</section>
<section id="the-variational-bayes-two-step-building-on-the-em" class="slide level2">
<h2>The Variational Bayes Two-Step… building on the EM</h2>
<ol type="1">
<li><p>Update ancestries given current frequencies: <span class="math inline">\(P(q_i |\text{data}, f) \propto P(\text{data}|q_i,f)P(q_i)\)</span></p></li>
<li><p>Update frequencies given current ancestries: <span class="math inline">\(P(f_k | \text{data}, q) \propto P(\text{data}|q,f_k)P(f_k)\)</span></p></li>
</ol>
</section>
<section id="structure-simulation-ancestry-and-allele-frequency-inference" class="slide level2">
<h2>STRUCTURE Simulation: Ancestry and Allele Frequency Inference</h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href=""></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb17-2"><a href=""></a><span class="fu">library</span>(viridis)</span>
<span id="cb17-3"><a href=""></a><span class="fu">library</span>(patchwork)</span>
<span id="cb17-4"><a href=""></a></span>
<span id="cb17-5"><a href=""></a><span class="fu">set.seed</span>(<span class="dv">2024</span>)</span>
<span id="cb17-6"><a href=""></a></span>
<span id="cb17-7"><a href=""></a><span class="co"># Parameters</span></span>
<span id="cb17-8"><a href=""></a>n_ind <span class="ot">&lt;-</span> <span class="dv">12</span></span>
<span id="cb17-9"><a href=""></a>n_pop <span class="ot">&lt;-</span> <span class="dv">2</span></span>
<span id="cb17-10"><a href=""></a>n_marker <span class="ot">&lt;-</span> <span class="dv">3</span></span>
<span id="cb17-11"><a href=""></a></span>
<span id="cb17-12"><a href=""></a><span class="co"># 1. Simulate population allele frequencies (for each marker, each pop)</span></span>
<span id="cb17-13"><a href=""></a>pop_freqs <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="fu">runif</span>(n_pop <span class="sc">*</span> n_marker, <span class="fl">0.2</span>, <span class="fl">0.8</span>), <span class="at">nrow =</span> n_pop)</span>
<span id="cb17-14"><a href=""></a><span class="fu">colnames</span>(pop_freqs) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"M"</span>, <span class="dv">1</span><span class="sc">:</span>n_marker)</span>
<span id="cb17-15"><a href=""></a><span class="fu">rownames</span>(pop_freqs) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"Pop"</span>, <span class="dv">1</span><span class="sc">:</span>n_pop)</span>
<span id="cb17-16"><a href=""></a></span>
<span id="cb17-17"><a href=""></a><span class="co"># 2. Simulate true ancestry proportions for each individual</span></span>
<span id="cb17-18"><a href=""></a><span class="co"># We'll make 4 mostly Pop1, 4 mostly Pop2, 4 admixed</span></span>
<span id="cb17-19"><a href=""></a>q_true <span class="ot">&lt;-</span> <span class="fu">rbind</span>(</span>
<span id="cb17-20"><a href=""></a>  <span class="fu">t</span>(<span class="fu">replicate</span>(<span class="dv">4</span>, <span class="fu">c</span>(<span class="fu">rbeta</span>(<span class="dv">1</span>, <span class="dv">8</span>, <span class="dv">2</span>), <span class="fu">rbeta</span>(<span class="dv">1</span>, <span class="dv">2</span>, <span class="dv">8</span>)))),   <span class="co"># mostly Pop1</span></span>
<span id="cb17-21"><a href=""></a>  <span class="fu">t</span>(<span class="fu">replicate</span>(<span class="dv">4</span>, <span class="fu">c</span>(<span class="fu">rbeta</span>(<span class="dv">1</span>, <span class="dv">2</span>, <span class="dv">8</span>), <span class="fu">rbeta</span>(<span class="dv">1</span>, <span class="dv">8</span>, <span class="dv">2</span>)))),   <span class="co"># mostly Pop2</span></span>
<span id="cb17-22"><a href=""></a>  <span class="fu">t</span>(<span class="fu">replicate</span>(<span class="dv">4</span>, <span class="fu">c</span>(<span class="fu">rbeta</span>(<span class="dv">1</span>, <span class="dv">5</span>, <span class="dv">5</span>), <span class="fu">rbeta</span>(<span class="dv">1</span>, <span class="dv">5</span>, <span class="dv">5</span>))))    <span class="co"># admixed</span></span>
<span id="cb17-23"><a href=""></a>)</span>
<span id="cb17-24"><a href=""></a>q_true <span class="ot">&lt;-</span> q_true <span class="sc">/</span> <span class="fu">rowSums</span>(q_true)</span>
<span id="cb17-25"><a href=""></a><span class="fu">colnames</span>(q_true) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Pop1"</span>, <span class="st">"Pop2"</span>)</span>
<span id="cb17-26"><a href=""></a></span>
<span id="cb17-27"><a href=""></a><span class="co"># 3. Simulate genotypes for each individual at each marker</span></span>
<span id="cb17-28"><a href=""></a><span class="co"># For simplicity, haploid: each allele is drawn from a population with probability = q</span></span>
<span id="cb17-29"><a href=""></a>geno <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="cn">NA</span>, <span class="at">nrow =</span> n_ind, <span class="at">ncol =</span> n_marker)</span>
<span id="cb17-30"><a href=""></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>n_ind) {</span>
<span id="cb17-31"><a href=""></a>  <span class="cf">for</span> (j <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>n_marker) {</span>
<span id="cb17-32"><a href=""></a>    <span class="co"># Choose population for this allele</span></span>
<span id="cb17-33"><a href=""></a>    pop <span class="ot">&lt;-</span> <span class="fu">sample</span>(<span class="dv">1</span><span class="sc">:</span>n_pop, <span class="dv">1</span>, <span class="at">prob =</span> q_true[i, ])</span>
<span id="cb17-34"><a href=""></a>    <span class="co"># Draw allele: 1 = reference (A), 0 = alt (C)</span></span>
<span id="cb17-35"><a href=""></a>    geno[i, j] <span class="ot">&lt;-</span> <span class="fu">rbinom</span>(<span class="dv">1</span>, <span class="dv">1</span>, pop_freqs[pop, j])</span>
<span id="cb17-36"><a href=""></a>  }</span>
<span id="cb17-37"><a href=""></a>}</span>
<span id="cb17-38"><a href=""></a><span class="fu">colnames</span>(geno) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"M"</span>, <span class="dv">1</span><span class="sc">:</span>n_marker)</span>
<span id="cb17-39"><a href=""></a><span class="fu">rownames</span>(geno) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"Ind"</span>, <span class="dv">1</span><span class="sc">:</span>n_ind)</span>
<span id="cb17-40"><a href=""></a></span>
<span id="cb17-41"><a href=""></a><span class="co"># 4. "Estimate" ancestry proportions over iterations (toy EM/MCMC)</span></span>
<span id="cb17-42"><a href=""></a><span class="co"># We'll start with random q, and update using the observed alleles and pop freqs</span></span>
<span id="cb17-43"><a href=""></a>n_iter <span class="ot">&lt;-</span> <span class="dv">10</span></span>
<span id="cb17-44"><a href=""></a>q_est <span class="ot">&lt;-</span> <span class="fu">array</span>(<span class="cn">NA</span>, <span class="at">dim =</span> <span class="fu">c</span>(n_ind, n_pop, n_iter))</span>
<span id="cb17-45"><a href=""></a>q_est[,,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">apply</span>(<span class="fu">matrix</span>(<span class="fu">runif</span>(n_ind <span class="sc">*</span> n_pop), <span class="at">ncol =</span> n_pop), <span class="dv">1</span>, <span class="cf">function</span>(x) x<span class="sc">/</span><span class="fu">sum</span>(x)))</span>
<span id="cb17-46"><a href=""></a></span>
<span id="cb17-47"><a href=""></a><span class="cf">for</span> (it <span class="cf">in</span> <span class="dv">2</span><span class="sc">:</span>n_iter) {</span>
<span id="cb17-48"><a href=""></a>  <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>n_ind) {</span>
<span id="cb17-49"><a href=""></a>    <span class="co"># For each population, compute likelihood of observed alleles</span></span>
<span id="cb17-50"><a href=""></a>    loglike <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="dv">0</span>, n_pop)</span>
<span id="cb17-51"><a href=""></a>    <span class="cf">for</span> (k <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>n_pop) {</span>
<span id="cb17-52"><a href=""></a>      <span class="cf">for</span> (j <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>n_marker) {</span>
<span id="cb17-53"><a href=""></a>        p <span class="ot">&lt;-</span> pop_freqs[k, j]</span>
<span id="cb17-54"><a href=""></a>        a <span class="ot">&lt;-</span> geno[i, j]</span>
<span id="cb17-55"><a href=""></a>        loglike[k] <span class="ot">&lt;-</span> loglike[k] <span class="sc">+</span> <span class="fu">dbinom</span>(a, <span class="dv">1</span>, p, <span class="at">log=</span><span class="cn">TRUE</span>)</span>
<span id="cb17-56"><a href=""></a>      }</span>
<span id="cb17-57"><a href=""></a>    }</span>
<span id="cb17-58"><a href=""></a>    <span class="co"># Add prior (Dirichlet(1,1) = uniform)</span></span>
<span id="cb17-59"><a href=""></a>    post <span class="ot">&lt;-</span> <span class="fu">exp</span>(loglike <span class="sc">-</span> <span class="fu">max</span>(loglike)) <span class="co"># for stability</span></span>
<span id="cb17-60"><a href=""></a>    post <span class="ot">&lt;-</span> post <span class="sc">/</span> <span class="fu">sum</span>(post)</span>
<span id="cb17-61"><a href=""></a>    <span class="co"># Update q as convex combination (like EM)</span></span>
<span id="cb17-62"><a href=""></a>    q_est[i, , it] <span class="ot">&lt;-</span> <span class="fl">0.7</span> <span class="sc">*</span> q_est[i, , it<span class="dv">-1</span>] <span class="sc">+</span> <span class="fl">0.3</span> <span class="sc">*</span> post</span>
<span id="cb17-63"><a href=""></a>    q_est[i, , it] <span class="ot">&lt;-</span> q_est[i, , it] <span class="sc">/</span> <span class="fu">sum</span>(q_est[i, , it])</span>
<span id="cb17-64"><a href=""></a>  }</span>
<span id="cb17-65"><a href=""></a>}</span>
<span id="cb17-66"><a href=""></a></span>
<span id="cb17-67"><a href=""></a><span class="co"># 5. Prepare data for plotting</span></span>
<span id="cb17-68"><a href=""></a></span>
<span id="cb17-69"><a href=""></a><span class="co"># True ancestry</span></span>
<span id="cb17-70"><a href=""></a>q_true_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(q_true)</span>
<span id="cb17-71"><a href=""></a>q_true_df<span class="sc">$</span>Individual <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="dv">1</span><span class="sc">:</span>n_ind)</span>
<span id="cb17-72"><a href=""></a>q_true_long <span class="ot">&lt;-</span> <span class="fu">pivot_longer</span>(q_true_df, <span class="at">cols =</span> <span class="fu">starts_with</span>(<span class="st">"Pop"</span>), <span class="at">names_to =</span> <span class="st">"Population"</span>, <span class="at">values_to =</span> <span class="st">"Proportion"</span>)</span>
<span id="cb17-73"><a href=""></a></span>
<span id="cb17-74"><a href=""></a><span class="co"># Estimated ancestry at each iteration</span></span>
<span id="cb17-75"><a href=""></a>q_est_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(q_est[,,n_iter])</span>
<span id="cb17-76"><a href=""></a><span class="fu">colnames</span>(q_est_df) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Pop1"</span>, <span class="st">"Pop2"</span>)</span>
<span id="cb17-77"><a href=""></a>q_est_df<span class="sc">$</span>Individual <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="dv">1</span><span class="sc">:</span>n_ind)</span>
<span id="cb17-78"><a href=""></a>q_est_long <span class="ot">&lt;-</span> <span class="fu">pivot_longer</span>(q_est_df, <span class="at">cols =</span> <span class="fu">starts_with</span>(<span class="st">"Pop"</span>), <span class="at">names_to =</span> <span class="st">"Population"</span>, <span class="at">values_to =</span> <span class="st">"Proportion"</span>)</span>
<span id="cb17-79"><a href=""></a></span>
<span id="cb17-80"><a href=""></a><span class="co"># Estimated ancestry over all iterations (for a few individuals)</span></span>
<span id="cb17-81"><a href=""></a>q_iter <span class="ot">&lt;-</span> <span class="fu">lapply</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>, <span class="cf">function</span>(i) {</span>
<span id="cb17-82"><a href=""></a>  <span class="fu">as.data.frame</span>(<span class="fu">t</span>(q_est[i,,])) <span class="sc">%&gt;%</span></span>
<span id="cb17-83"><a href=""></a>    <span class="fu">mutate</span>(<span class="at">Iteration =</span> <span class="dv">1</span><span class="sc">:</span>n_iter, <span class="at">Individual =</span> i)</span>
<span id="cb17-84"><a href=""></a>})</span>
<span id="cb17-85"><a href=""></a>q_iter <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(q_iter)</span>
<span id="cb17-86"><a href=""></a><span class="fu">colnames</span>(q_iter)[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Pop1"</span>, <span class="st">"Pop2"</span>)</span>
<span id="cb17-87"><a href=""></a>q_iter_long <span class="ot">&lt;-</span> <span class="fu">pivot_longer</span>(q_iter, <span class="at">cols =</span> <span class="fu">c</span>(<span class="st">"Pop1"</span>, <span class="st">"Pop2"</span>), <span class="at">names_to =</span> <span class="st">"Population"</span>, <span class="at">values_to =</span> <span class="st">"Proportion"</span>)</span>
<span id="cb17-88"><a href=""></a></span>
<span id="cb17-89"><a href=""></a><span class="co"># 6. Plot</span></span>
<span id="cb17-90"><a href=""></a></span>
<span id="cb17-91"><a href=""></a><span class="co"># STRUCTURE-style barplot: true ancestry</span></span>
<span id="cb17-92"><a href=""></a>p_true <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(q_true_long, <span class="fu">aes</span>(<span class="at">x =</span> Individual, <span class="at">y =</span> Proportion, <span class="at">fill =</span> Population)) <span class="sc">+</span></span>
<span id="cb17-93"><a href=""></a>  <span class="fu">geom_col</span>(<span class="at">width =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb17-94"><a href=""></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"Pop1"</span> <span class="ot">=</span> <span class="st">"blue"</span>, <span class="st">"Pop2"</span> <span class="ot">=</span> <span class="st">"red"</span>)) <span class="sc">+</span></span>
<span id="cb17-95"><a href=""></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"True Ancestry Proportions"</span>, <span class="at">y =</span> <span class="st">"Proportion"</span>, <span class="at">x =</span> <span class="st">"Individual"</span>) <span class="sc">+</span></span>
<span id="cb17-96"><a href=""></a>  <span class="fu">theme_minimal</span>(<span class="at">base_size =</span> <span class="dv">14</span>)</span>
<span id="cb17-97"><a href=""></a></span>
<span id="cb17-98"><a href=""></a><span class="co"># STRUCTURE-style barplot: estimated ancestry (final iteration)</span></span>
<span id="cb17-99"><a href=""></a>p_est <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(q_est_long, <span class="fu">aes</span>(<span class="at">x =</span> Individual, <span class="at">y =</span> Proportion, <span class="at">fill =</span> Population)) <span class="sc">+</span></span>
<span id="cb17-100"><a href=""></a>  <span class="fu">geom_col</span>(<span class="at">width =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb17-101"><a href=""></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"Pop1"</span> <span class="ot">=</span> <span class="st">"blue"</span>, <span class="st">"Pop2"</span> <span class="ot">=</span> <span class="st">"red"</span>)) <span class="sc">+</span></span>
<span id="cb17-102"><a href=""></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Estimated Ancestry (Iteration 10)"</span>, <span class="at">y =</span> <span class="st">"Proportion"</span>, <span class="at">x =</span> <span class="st">"Individual"</span>) <span class="sc">+</span></span>
<span id="cb17-103"><a href=""></a>  <span class="fu">theme_minimal</span>(<span class="at">base_size =</span> <span class="dv">14</span>)</span>
<span id="cb17-104"><a href=""></a></span>
<span id="cb17-105"><a href=""></a><span class="co"># Trajectory plot for a few individuals</span></span>
<span id="cb17-106"><a href=""></a>p_traj <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(q_iter_long, <span class="fu">aes</span>(<span class="at">x =</span> Iteration, <span class="at">y =</span> Proportion, <span class="at">color =</span> Population)) <span class="sc">+</span></span>
<span id="cb17-107"><a href=""></a>  <span class="fu">geom_line</span>(<span class="at">size =</span> <span class="fl">1.2</span>) <span class="sc">+</span></span>
<span id="cb17-108"><a href=""></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span> Individual, <span class="at">nrow =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb17-109"><a href=""></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"Pop1"</span> <span class="ot">=</span> <span class="st">"blue"</span>, <span class="st">"Pop2"</span> <span class="ot">=</span> <span class="st">"red"</span>)) <span class="sc">+</span></span>
<span id="cb17-110"><a href=""></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Ancestry Proportion Trajectories (First 6 Individuals)"</span>, <span class="at">y =</span> <span class="st">"Proportion"</span>, <span class="at">x =</span> <span class="st">"Iteration"</span>) <span class="sc">+</span></span>
<span id="cb17-111"><a href=""></a>  <span class="fu">theme_minimal</span>(<span class="at">base_size =</span> <span class="dv">12</span>)</span>
<span id="cb17-112"><a href=""></a></span>
<span id="cb17-113"><a href=""></a><span class="co"># Heatmap of population allele frequencies</span></span>
<span id="cb17-114"><a href=""></a>pop_freqs_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(pop_freqs)</span>
<span id="cb17-115"><a href=""></a>pop_freqs_df<span class="sc">$</span>Population <span class="ot">&lt;-</span> <span class="fu">rownames</span>(pop_freqs)</span>
<span id="cb17-116"><a href=""></a>pop_freqs_long <span class="ot">&lt;-</span> <span class="fu">pivot_longer</span>(pop_freqs_df, <span class="at">cols =</span> <span class="fu">starts_with</span>(<span class="st">"M"</span>), <span class="at">names_to =</span> <span class="st">"Marker"</span>, <span class="at">values_to =</span> <span class="st">"Frequency"</span>)</span>
<span id="cb17-117"><a href=""></a>p_freq <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(pop_freqs_long, <span class="fu">aes</span>(<span class="at">x =</span> Marker, <span class="at">y =</span> Population, <span class="at">fill =</span> Frequency)) <span class="sc">+</span></span>
<span id="cb17-118"><a href=""></a>  <span class="fu">geom_tile</span>() <span class="sc">+</span></span>
<span id="cb17-119"><a href=""></a>  <span class="fu">scale_fill_viridis</span>(<span class="at">option =</span> <span class="st">"C"</span>) <span class="sc">+</span></span>
<span id="cb17-120"><a href=""></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Population Allele Frequencies"</span>, <span class="at">x =</span> <span class="st">"Marker"</span>, <span class="at">y =</span> <span class="st">"Population"</span>) <span class="sc">+</span></span>
<span id="cb17-121"><a href=""></a>  <span class="fu">theme_minimal</span>(<span class="at">base_size =</span> <span class="dv">12</span>)</span>
<span id="cb17-122"><a href=""></a></span>
<span id="cb17-123"><a href=""></a><span class="co"># Combine plots</span></span>
<span id="cb17-124"><a href=""></a>(p_true <span class="sc">|</span> p_est) <span class="sc">/</span> (p_traj <span class="sc">|</span> p_freq) <span class="sc">+</span> <span class="fu">plot_layout</span>(<span class="at">heights =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="fl">1.2</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>

</div>
<img data-src="mpg6_files/figure-revealjs/unnamed-chunk-19-1.png" width="960" class="r-stretch"></section>
<section id="effect-size-mixtures-in-gwas" class="slide level2">
<h2><strong>Effect Size Mixtures in GWAS</strong></h2>
<p><strong>Problem</strong>: Most variants have no effect, but some do</p>
<p><strong>Solutions</strong>:<br>
- <strong>Spike-and-slab prior</strong>: Mixture of point mass at zero and continuous distribution<br>
- <strong>Scale mixture</strong>: Mixture of normal distributions with different variances<br>
- <strong>Bayesian variable selection</strong>: Latent indicator for whether variant is causal</p>
<!-- **Sound familiar? Multiple hypothesis testing!** -->
<!-- ##  -->
<!-- ```{r, echo=TRUE} -->
<!-- # Create visualization of effect size mixtures -->
<!-- set.seed(789) -->
<!-- # Parameters -->
<!-- n_variants <- 500 -->
<!-- pi0 <- 0.95  # Proportion of null effects -->
<!-- # Generate true effects -->
<!-- is_null <- rbinom(n_variants, 1, pi0) -->
<!-- true_effects <- rep(0, n_variants) -->
<!-- true_effects[is_null == 0] <- rnorm(sum(is_null == 0), 0, 0.5) -->
<!-- true_effects[is_null == 1] <- rnorm(sum(is_null == 1), 0, 2) -->
<!-- # Add noise to create observed effects -->
<!-- observed_effects <- true_effects + rnorm(n_variants, 0, 0.2) -->
<!-- # Create data frame for plotting -->
<!-- effect_df <- data.frame( -->
<!--   Variant = 1:n_variants, -->
<!--   TrueEffect = true_effects, -->
<!--   ObservedEffect = observed_effects, -->
<!--   IsNull = factor(is_null) -->
<!-- ) -->
<!-- # Plot -->
<!-- ggplot(effect_df, aes(x = ObservedEffect, fill = IsNull)) + -->
<!--   geom_density(alpha=0.5) + -->
<!--   scale_fill_manual(values = c("gray70", "red"),  -->
<!--                    labels = c("Null", "True"), -->
<!--                    name = "Variant Type") + -->
<!--   labs(title = "Mixture of Effect Sizes in GWAS", -->
<!--        subtitle = "Most variants have no effect (null)", -->
<!--        x = "Observed Effect Size",  -->
<!--        y = "Count") + -->
<!--   theme_minimal() + -->
<!--   theme(legend.position = "bottom") -->
<!-- ``` -->
<!-- ## Multivariate Normal Mixtures: The mash Approach -->
<!-- ```{r,echo=TRUE} -->
<!-- #| fig-height: 6 -->
<!-- # Create a visualization of multivariate effects -->
<!-- set.seed(345) -->
<!-- # Parameters -->
<!-- n_effects <- 200 -->
<!-- n_conditions <- 4 -->
<!-- # Create different effect patterns -->
<!-- patterns <- list( -->
<!--   "Shared" = rep(1, n_conditions), -->
<!--   "Condition1" = c(1, 0, 0, 0), -->
<!--   "Condition2" = c(0, 1, 0, 0), -->
<!--   "Conditions1&2" = c(1, 1, 0, 0), -->
<!--   "Conditions3&4" = c(0, 0, 1, 1) -->
<!-- ) -->
<!-- # Assign effects to patterns -->
<!-- n_per_pattern <- n_effects / length(patterns) -->
<!-- true_effects <- matrix(0, nrow = n_effects, ncol = n_conditions) -->
<!-- colnames(true_effects) <- paste0("Condition", 1:n_conditions) -->
<!-- current_idx <- 1 -->
<!-- for (p in 1:length(patterns)) { -->
<!--   pattern_name <- names(patterns)[p] -->
<!--   pattern <- patterns[[p]] -->
<!--   for (i in 1:n_per_pattern) { -->
<!--     effect_size <- rnorm(1, 0, 0.5) -->
<!--     true_effects[current_idx, ] <- pattern * effect_size -->
<!--     current_idx <- current_idx + 1 -->
<!--   } -->
<!-- } -->
<!-- # Add noise to create observed effects -->
<!-- observed_effects <- true_effects + matrix(rnorm(n_effects * n_conditions, 0, 0.2), -->
<!--                                         nrow = n_effects) -->
<!-- # Select a few examples for visualization -->
<!-- example_indices <- c(5, 45, 85, 125, 165)  # One from each pattern -->
<!-- example_data <- data.frame( -->
<!--   Effect = rep(paste0("Effect", example_indices), each = n_conditions), -->
<!--   Condition = rep(paste0("Condition", 1:n_conditions), times = length(example_indices)), -->
<!--   TrueEffect = c(t(true_effects[example_indices, ])), -->
<!--   ObservedEffect = c(t(observed_effects[example_indices, ])) -->
<!-- ) -->
<!-- # Reshape for plotting -->
<!-- example_long <- reshape2::melt(example_data,  -->
<!--                              id.vars = c("Effect", "Condition"), -->
<!--                              variable.name = "EffectType", -->
<!--                              value.name = "Value") -->
<!-- # Create plot -->
<!-- ggplot(example_long, aes(x = Condition, y = Value, color = EffectType, group = EffectType)) + -->
<!--   geom_point(size = 3) + -->
<!--   geom_line(size = 1) + -->
<!--   facet_wrap(~ Effect, ncol = 3) + -->
<!--   scale_color_manual(values = c("TrueEffect" = "blue", "ObservedEffect" = "red"), -->
<!--                     labels = c("True Effect", "Observed Effect")) + -->
<!--   labs(title = "Multivariate Effect Patterns", -->
<!--        subtitle = "mash identifies and leverages these patterns", -->
<!--        x = "",  -->
<!--        y = "Effect Size", -->
<!--        color = "") + -->
<!--   theme(legend.position = "bottom") -->
</section></section>
<section>
<section id="bayesian-clinical-trials" class="title-slide slide level1 center">
<h1>5. Bayesian Clinical Trials:</h1>
<p>Observe OR of 1.5O</p>

<img data-src="mpg6_files/figure-revealjs/unnamed-chunk-20-1.png" width="960" class="r-stretch"></section>
<section id="bayes-combining-information" class="slide level2">
<h2>Bayes combining information</h2>
<p><strong>Problem</strong>: The meta analysis example, combining evidence across heterogeneous studies</p>
<p><strong>Model formulation</strong>:<br>
- Let <span class="math inline">\(y_i\)</span> be the observed effect in study <span class="math inline">\(i\)</span><br>
- Let <span class="math inline">\(\sigma_i^2\)</span> be the variance (often known from standard error)<br>
- Let <span class="math inline">\(\theta_i\)</span> be the true effect in study <span class="math inline">\(i\)</span></p>
<p><strong>Hierarchical model</strong>:<br>
<span class="math inline">\(y_i | \theta_i, \sigma_i^2 \sim N(\theta_i, \sigma_i^2)\)</span><br>
<span class="math inline">\(\theta_i | \mu, \tau^2 \sim N(\mu, \tau^2)\)</span><br>
<span class="math inline">\(\mu \sim N(\mu_0, \sigma_0^2)\)</span><br>
<span class="math inline">\(\tau^2 \sim \text{InvGamma}(a, b)\)</span></p>
<aside class="notes">
<ul>
<li><span class="math inline">\(\mu\)</span> is the overall mean effect<br>
</li>
<li><span class="math inline">\(\tau^2\)</span> is the between-study heterogeneity<br>
</li>
<li><span class="math inline">\(\mu_0, \sigma_0^2, a, b\)</span> are mean and ‘intra study’ variation<br>
</li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="putting-it-together" class="slide level2">
<h2>Putting it together</h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href=""></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb18-2"><a href=""></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb18-3"><a href=""></a></span>
<span id="cb18-4"><a href=""></a><span class="co"># Create simulated data for visualization</span></span>
<span id="cb18-5"><a href=""></a><span class="fu">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb18-6"><a href=""></a>n_studies <span class="ot">&lt;-</span> <span class="dv">6</span></span>
<span id="cb18-7"><a href=""></a></span>
<span id="cb18-8"><a href=""></a><span class="co"># Parameters</span></span>
<span id="cb18-9"><a href=""></a>mu_true <span class="ot">&lt;-</span> <span class="fl">0.3</span></span>
<span id="cb18-10"><a href=""></a>tau <span class="ot">&lt;-</span> <span class="fl">0.2</span></span>
<span id="cb18-11"><a href=""></a>mu0 <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb18-12"><a href=""></a>sigma0 <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb18-13"><a href=""></a></span>
<span id="cb18-14"><a href=""></a><span class="co"># Generate study-specific effects</span></span>
<span id="cb18-15"><a href=""></a>theta_i <span class="ot">&lt;-</span> <span class="fu">rnorm</span>(n_studies, mu_true, tau)</span>
<span id="cb18-16"><a href=""></a>sigma_i <span class="ot">&lt;-</span> <span class="fu">runif</span>(n_studies, <span class="fl">0.1</span>, <span class="fl">0.3</span>)</span>
<span id="cb18-17"><a href=""></a>y_i <span class="ot">&lt;-</span> <span class="fu">rnorm</span>(n_studies, theta_i, sigma_i)</span>
<span id="cb18-18"><a href=""></a></span>
<span id="cb18-19"><a href=""></a><span class="co"># Create plot data</span></span>
<span id="cb18-20"><a href=""></a>plot_data <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb18-21"><a href=""></a>  <span class="at">study =</span> <span class="dv">1</span><span class="sc">:</span>n_studies,</span>
<span id="cb18-22"><a href=""></a>  <span class="at">y =</span> y_i,</span>
<span id="cb18-23"><a href=""></a>  <span class="at">theta =</span> theta_i,</span>
<span id="cb18-24"><a href=""></a>  <span class="at">se =</span> sigma_i</span>
<span id="cb18-25"><a href=""></a>)</span>
<span id="cb18-26"><a href=""></a></span>
<span id="cb18-27"><a href=""></a><span class="co"># Create the hierarchical visualization</span></span>
<span id="cb18-28"><a href=""></a><span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb18-29"><a href=""></a>  <span class="co"># Draw the overall prior distribution for μ</span></span>
<span id="cb18-30"><a href=""></a>  <span class="fu">stat_function</span>(<span class="at">fun =</span> dnorm, </span>
<span id="cb18-31"><a href=""></a>               <span class="at">args =</span> <span class="fu">list</span>(<span class="at">mean =</span> mu0, <span class="at">sd =</span> sigma0),</span>
<span id="cb18-32"><a href=""></a>               <span class="fu">aes</span>(<span class="at">color =</span> <span class="st">"Prior for μ"</span>),</span>
<span id="cb18-33"><a href=""></a>               <span class="at">xlim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">2</span>, <span class="dv">2</span>)) <span class="sc">+</span></span>
<span id="cb18-34"><a href=""></a>  </span>
<span id="cb18-35"><a href=""></a>  <span class="co"># Draw the distribution for study effects θᵢ</span></span>
<span id="cb18-36"><a href=""></a>  <span class="fu">stat_function</span>(<span class="at">fun =</span> dnorm,</span>
<span id="cb18-37"><a href=""></a>               <span class="at">args =</span> <span class="fu">list</span>(<span class="at">mean =</span> mu_true, <span class="at">sd =</span> tau),</span>
<span id="cb18-38"><a href=""></a>               <span class="fu">aes</span>(<span class="at">color =</span> <span class="st">"Distribution of θi"</span>),</span>
<span id="cb18-39"><a href=""></a>               <span class="at">xlim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">2</span>, <span class="dv">2</span>)) <span class="sc">+</span></span>
<span id="cb18-40"><a href=""></a>  </span>
<span id="cb18-41"><a href=""></a>  <span class="co"># Add study-specific observations with error bars</span></span>
<span id="cb18-42"><a href=""></a>  <span class="fu">geom_pointrange</span>(<span class="at">data =</span> plot_data,</span>
<span id="cb18-43"><a href=""></a>                 <span class="fu">aes</span>(<span class="at">x =</span> y, <span class="at">y =</span> study, </span>
<span id="cb18-44"><a href=""></a>                     <span class="at">xmin =</span> y <span class="sc">-</span> <span class="fl">1.96</span><span class="sc">*</span>se,</span>
<span id="cb18-45"><a href=""></a>                     <span class="at">xmax =</span> y <span class="sc">+</span> <span class="fl">1.96</span><span class="sc">*</span>se,</span>
<span id="cb18-46"><a href=""></a>                     <span class="at">color =</span> <span class="st">"Observed effects yi"</span>),</span>
<span id="cb18-47"><a href=""></a>                 <span class="at">size =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb18-48"><a href=""></a>  </span>
<span id="cb18-49"><a href=""></a>  <span class="co"># Add true study effects</span></span>
<span id="cb18-50"><a href=""></a>  <span class="fu">geom_point</span>(<span class="at">data =</span> plot_data,</span>
<span id="cb18-51"><a href=""></a>             <span class="fu">aes</span>(<span class="at">x =</span> theta, <span class="at">y =</span> study,</span>
<span id="cb18-52"><a href=""></a>                 <span class="at">color =</span> <span class="st">"True effects θi"</span>),</span>
<span id="cb18-53"><a href=""></a>             <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb18-54"><a href=""></a>  </span>
<span id="cb18-55"><a href=""></a>  <span class="co"># Add vertical line for true mean</span></span>
<span id="cb18-56"><a href=""></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> mu_true, </span>
<span id="cb18-57"><a href=""></a>             <span class="at">linetype =</span> <span class="st">"dashed"</span>, </span>
<span id="cb18-58"><a href=""></a>             <span class="at">color =</span> <span class="st">"darkgreen"</span>) <span class="sc">+</span></span>
<span id="cb18-59"><a href=""></a>  </span>
<span id="cb18-60"><a href=""></a>  <span class="co"># Customize appearance</span></span>
<span id="cb18-61"><a href=""></a>  <span class="fu">scale_color_manual</span>(<span class="at">name =</span> <span class="st">"Components"</span>,</span>
<span id="cb18-62"><a href=""></a>                    <span class="at">values =</span> <span class="fu">c</span>(<span class="st">"Prior for μ"</span> <span class="ot">=</span> <span class="st">"blue"</span>,</span>
<span id="cb18-63"><a href=""></a>                             <span class="st">"Distribution of θi"</span> <span class="ot">=</span> <span class="st">"purple"</span>,</span>
<span id="cb18-64"><a href=""></a>                             <span class="st">"True effects θi"</span> <span class="ot">=</span> <span class="st">"darkgreen"</span>,</span>
<span id="cb18-65"><a href=""></a>                             <span class="st">"Observed effects yi"</span> <span class="ot">=</span> <span class="st">"red"</span>)) <span class="sc">+</span></span>
<span id="cb18-66"><a href=""></a>  </span>
<span id="cb18-67"><a href=""></a>  <span class="co"># Add labels and theme</span></span>
<span id="cb18-68"><a href=""></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Hierarchical Structure of Meta-Analysis Model"</span>,</span>
<span id="cb18-69"><a href=""></a>       <span class="at">subtitle =</span> <span class="fu">paste</span>(<span class="st">"μ0 ="</span>, mu0, <span class="st">", σ0 ="</span>, sigma0, </span>
<span id="cb18-70"><a href=""></a>                       <span class="st">", τ ="</span>, tau, <span class="st">"</span><span class="sc">\n</span><span class="st">True μ ="</span>, mu_true),</span>
<span id="cb18-71"><a href=""></a>       <span class="at">x =</span> <span class="st">"Effect Size"</span>,</span>
<span id="cb18-72"><a href=""></a>       <span class="at">y =</span> <span class="st">"Study"</span>) <span class="sc">+</span></span>
<span id="cb18-73"><a href=""></a>  </span>
<span id="cb18-74"><a href=""></a>  <span class="co"># Add annotations explaining the hierarchy</span></span>
<span id="cb18-75"><a href=""></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="sc">-</span><span class="fl">1.8</span>, <span class="at">y =</span> <span class="fl">7.5</span>,</span>
<span id="cb18-76"><a href=""></a>           <span class="at">label =</span> <span class="st">"yi|θi,sigmai² ~ N(θi,σi²)</span><span class="sc">\n</span></span>
<span id="cb18-77"><a href=""></a><span class="st">           θᵢ|μ,τ² ~ N(μ,τ²)</span><span class="sc">\n</span></span>
<span id="cb18-78"><a href=""></a><span class="st">           μ ~ N(μ1,σ1²)"</span>,</span>
<span id="cb18-79"><a href=""></a>           <span class="at">hjust =</span> <span class="dv">0</span>) <span class="sc">+</span></span>
<span id="cb18-80"><a href=""></a>  </span>
<span id="cb18-81"><a href=""></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb18-82"><a href=""></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"bottom"</span>,</span>
<span id="cb18-83"><a href=""></a>        <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>),</span>
<span id="cb18-84"><a href=""></a>        <span class="at">plot.subtitle =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>

</div>
<img data-src="mpg6_files/figure-revealjs/unnamed-chunk-21-1.png" width="960" class="r-stretch"><aside class="notes">
<p>At the study level, we observe <span class="math inline">\(y_i\)</span> with known variance <span class="math inline">\(\sigma_i^2\)</span> These come from true effects <span class="math inline">\(\theta_i\)</span> that vary around the overall effect <span class="math inline">\(\mu\)</span> with variance <span class="math inline">\(\tau^2\)</span> We put priors on <span class="math inline">\(\mu\)</span> (using <span class="math inline">\(\mu_0, \sigma_0^2\)</span>) and <span class="math inline">\(\tau^2\)</span> (using <span class="math inline">\(a, b\)</span>) The key is that <span class="math inline">\(\tau^2\)</span> represents how much the true effects vary between studies, while <span class="math inline">\(\sigma_i^2\)</span> represents the uncertainty within each study. <span class="math inline">\(\mu_0\)</span> and <span class="math inline">\(\sigma_0^2\)</span> are just prior parameters for our belief about the overall effect.</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="visualization-in-frequentist-terms" class="slide level2">
<h2>Visualization in frequentist terms</h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href=""></a><span class="fu">set.seed</span>(<span class="dv">678</span>)</span>
<span id="cb19-2"><a href=""></a></span>
<span id="cb19-3"><a href=""></a><span class="co"># Parameters</span></span>
<span id="cb19-4"><a href=""></a>n_studies <span class="ot">&lt;-</span> <span class="dv">8</span></span>
<span id="cb19-5"><a href=""></a>true_effect <span class="ot">&lt;-</span> <span class="fl">0.3</span></span>
<span id="cb19-6"><a href=""></a>heterogeneity <span class="ot">&lt;-</span> <span class="fl">0.1</span></span>
<span id="cb19-7"><a href=""></a>clinically_meaningful <span class="ot">&lt;-</span> <span class="fl">0.2</span>  <span class="co"># Define clinically meaningful threshold</span></span>
<span id="cb19-8"><a href=""></a></span>
<span id="cb19-9"><a href=""></a><span class="co"># Generate study-specific effects</span></span>
<span id="cb19-10"><a href=""></a>study_effects <span class="ot">&lt;-</span> <span class="fu">rnorm</span>(n_studies, true_effect, heterogeneity)</span>
<span id="cb19-11"><a href=""></a>study_effects[n_studies] <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fl">0.2</span>  <span class="co"># outlier</span></span>
<span id="cb19-12"><a href=""></a></span>
<span id="cb19-13"><a href=""></a><span class="co"># Generate observed effects with more informative sample sizes</span></span>
<span id="cb19-14"><a href=""></a>sample_sizes <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">150</span>, <span class="dv">200</span>, <span class="dv">300</span>, <span class="dv">400</span>, <span class="dv">450</span>, <span class="dv">200</span>, <span class="dv">150</span>, <span class="dv">100</span>)  <span class="co"># More realistic</span></span>
<span id="cb19-15"><a href=""></a>standard_errors <span class="ot">&lt;-</span> <span class="dv">1</span> <span class="sc">/</span> <span class="fu">sqrt</span>(sample_sizes)</span>
<span id="cb19-16"><a href=""></a>observed_effects <span class="ot">&lt;-</span> <span class="fu">rnorm</span>(n_studies, study_effects, standard_errors)</span>
<span id="cb19-17"><a href=""></a></span>
<span id="cb19-18"><a href=""></a><span class="co"># Bayesian analysis</span></span>
<span id="cb19-19"><a href=""></a><span class="co"># Prior: N(0, 1) - weakly informative</span></span>
<span id="cb19-20"><a href=""></a>prior_mean <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb19-21"><a href=""></a>prior_sd <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb19-22"><a href=""></a></span>
<span id="cb19-23"><a href=""></a><span class="co"># Posterior calculations (using precision-weighted approach)</span></span>
<span id="cb19-24"><a href=""></a>posterior_precision <span class="ot">&lt;-</span> <span class="dv">1</span><span class="sc">/</span>prior_sd<span class="sc">^</span><span class="dv">2</span> <span class="sc">+</span> <span class="fu">sum</span>(<span class="dv">1</span><span class="sc">/</span>standard_errors<span class="sc">^</span><span class="dv">2</span>)</span>
<span id="cb19-25"><a href=""></a>posterior_mean <span class="ot">&lt;-</span> (prior_mean<span class="sc">/</span>prior_sd<span class="sc">^</span><span class="dv">2</span> <span class="sc">+</span> <span class="fu">sum</span>(observed_effects<span class="sc">/</span>standard_errors<span class="sc">^</span><span class="dv">2</span>))<span class="sc">/</span>posterior_precision</span>
<span id="cb19-26"><a href=""></a>posterior_sd <span class="ot">&lt;-</span> <span class="fu">sqrt</span>(<span class="dv">1</span><span class="sc">/</span>posterior_precision)</span>
<span id="cb19-27"><a href=""></a></span>
<span id="cb19-28"><a href=""></a><span class="co"># Calculate probability of clinically meaningful effect</span></span>
<span id="cb19-29"><a href=""></a>prob_meaningful <span class="ot">&lt;-</span> <span class="dv">1</span> <span class="sc">-</span> <span class="fu">pnorm</span>(clinically_meaningful, posterior_mean, posterior_sd)</span>
<span id="cb19-30"><a href=""></a></span>
<span id="cb19-31"><a href=""></a><span class="co"># Create data frame for forest plot</span></span>
<span id="cb19-32"><a href=""></a>meta_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb19-33"><a href=""></a>  <span class="at">Study =</span> <span class="fu">paste0</span>(<span class="st">"Study "</span>, <span class="dv">1</span><span class="sc">:</span>n_studies),</span>
<span id="cb19-34"><a href=""></a>  <span class="at">Effect =</span> observed_effects,</span>
<span id="cb19-35"><a href=""></a>  <span class="at">SE =</span> standard_errors,</span>
<span id="cb19-36"><a href=""></a>  <span class="at">LowerCI =</span> observed_effects <span class="sc">-</span> <span class="fl">1.96</span> <span class="sc">*</span> standard_errors,</span>
<span id="cb19-37"><a href=""></a>  <span class="at">UpperCI =</span> observed_effects <span class="sc">+</span> <span class="fl">1.96</span> <span class="sc">*</span> standard_errors</span>
<span id="cb19-38"><a href=""></a>)</span>
<span id="cb19-39"><a href=""></a></span>
<span id="cb19-40"><a href=""></a><span class="co"># Add Bayesian results</span></span>
<span id="cb19-41"><a href=""></a>bayes_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb19-42"><a href=""></a>  <span class="at">Study =</span> <span class="st">"Bayesian Estimate"</span>,</span>
<span id="cb19-43"><a href=""></a>  <span class="at">Effect =</span> posterior_mean,</span>
<span id="cb19-44"><a href=""></a>  <span class="at">SE =</span> posterior_sd,</span>
<span id="cb19-45"><a href=""></a>  <span class="at">LowerCI =</span> posterior_mean <span class="sc">-</span> <span class="fl">1.96</span> <span class="sc">*</span> posterior_sd,</span>
<span id="cb19-46"><a href=""></a>  <span class="at">UpperCI =</span> posterior_mean <span class="sc">+</span> <span class="fl">1.96</span> <span class="sc">*</span> posterior_sd</span>
<span id="cb19-47"><a href=""></a>)</span>
<span id="cb19-48"><a href=""></a></span>
<span id="cb19-49"><a href=""></a><span class="co"># Combine data</span></span>
<span id="cb19-50"><a href=""></a>plot_df <span class="ot">&lt;-</span> <span class="fu">rbind</span>(meta_df, bayes_df)</span>
<span id="cb19-51"><a href=""></a>plot_df<span class="sc">$</span>Group <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(plot_df<span class="sc">$</span>Study <span class="sc">==</span> <span class="st">"Bayesian Estimate"</span>, <span class="st">"Bayesian"</span>, <span class="st">"Individual Studies"</span>)</span>
<span id="cb19-52"><a href=""></a>plot_df<span class="sc">$</span>Group <span class="ot">&lt;-</span> <span class="fu">factor</span>(plot_df<span class="sc">$</span>Group, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"Individual Studies"</span>, <span class="st">"Bayesian"</span>))</span>
<span id="cb19-53"><a href=""></a></span>
<span id="cb19-54"><a href=""></a></span>
<span id="cb19-55"><a href=""></a></span>
<span id="cb19-56"><a href=""></a><span class="co"># Create two plots side by side</span></span>
<span id="cb19-57"><a href=""></a>p1 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(plot_df, <span class="fu">aes</span>(<span class="at">x =</span> Effect, <span class="at">y =</span> Study, <span class="at">color =</span> Group)) <span class="sc">+</span></span>
<span id="cb19-58"><a href=""></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">color =</span> <span class="st">"gray"</span>) <span class="sc">+</span></span>
<span id="cb19-59"><a href=""></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> clinically_meaningful, <span class="at">linetype =</span> <span class="st">"dotted"</span>, <span class="at">color =</span> <span class="st">"darkgreen"</span>, </span>
<span id="cb19-60"><a href=""></a>             <span class="at">size =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb19-61"><a href=""></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">size =</span> <span class="dv">1</span><span class="sc">/</span>SE)) <span class="sc">+</span></span>
<span id="cb19-62"><a href=""></a>  <span class="fu">geom_errorbarh</span>(<span class="fu">aes</span>(<span class="at">xmin =</span> LowerCI, <span class="at">xmax =</span> UpperCI), <span class="at">height =</span> <span class="fl">0.2</span>) <span class="sc">+</span></span>
<span id="cb19-63"><a href=""></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"Individual Studies"</span> <span class="ot">=</span> <span class="st">"blue"</span>, <span class="st">"Bayesian"</span> <span class="ot">=</span> <span class="st">"red"</span>)) <span class="sc">+</span></span>
<span id="cb19-64"><a href=""></a>  <span class="fu">scale_size_continuous</span>(<span class="at">guide =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb19-65"><a href=""></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Forest Plot"</span>,</span>
<span id="cb19-66"><a href=""></a>       <span class="at">x =</span> <span class="st">"Effect Size"</span>, </span>
<span id="cb19-67"><a href=""></a>       <span class="at">y =</span> <span class="st">""</span>) <span class="sc">+</span></span>
<span id="cb19-68"><a href=""></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb19-69"><a href=""></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"bottom"</span>)</span>
<span id="cb19-70"><a href=""></a></span>
<span id="cb19-71"><a href=""></a><span class="co"># Create density plot for posterior</span></span>
<span id="cb19-72"><a href=""></a>x_range <span class="ot">&lt;-</span> <span class="fu">seq</span>(posterior_mean <span class="sc">-</span> <span class="dv">4</span><span class="sc">*</span>posterior_sd, </span>
<span id="cb19-73"><a href=""></a>               posterior_mean <span class="sc">+</span> <span class="dv">4</span><span class="sc">*</span>posterior_sd, </span>
<span id="cb19-74"><a href=""></a>               <span class="at">length.out =</span> <span class="dv">1000</span>)</span>
<span id="cb19-75"><a href=""></a>posterior_density <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb19-76"><a href=""></a>  <span class="at">x =</span> x_range,</span>
<span id="cb19-77"><a href=""></a>  <span class="at">y =</span> <span class="fu">dnorm</span>(x_range, posterior_mean, posterior_sd)</span>
<span id="cb19-78"><a href=""></a>)</span>
<span id="cb19-79"><a href=""></a></span>
<span id="cb19-80"><a href=""></a>p2 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(posterior_density, <span class="fu">aes</span>(<span class="at">x =</span> x, <span class="at">y =</span> y)) <span class="sc">+</span></span>
<span id="cb19-81"><a href=""></a>  <span class="fu">geom_line</span>(<span class="at">color =</span> <span class="st">"red"</span>, <span class="at">size =</span> <span class="fl">1.2</span>) <span class="sc">+</span></span>
<span id="cb19-82"><a href=""></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">color =</span> <span class="st">"gray"</span>) <span class="sc">+</span></span>
<span id="cb19-83"><a href=""></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> clinically_meaningful, <span class="at">linetype =</span> <span class="st">"dotted"</span>, </span>
<span id="cb19-84"><a href=""></a>             <span class="at">color =</span> <span class="st">"darkgreen"</span>, <span class="at">size =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb19-85"><a href=""></a>  <span class="fu">geom_area</span>(<span class="at">data =</span> <span class="fu">subset</span>(posterior_density, x <span class="sc">&gt;</span> clinically_meaningful),</span>
<span id="cb19-86"><a href=""></a>            <span class="fu">aes</span>(<span class="at">y =</span> y), <span class="at">fill =</span> <span class="st">"red"</span>, <span class="at">alpha =</span> <span class="fl">0.2</span>) <span class="sc">+</span></span>
<span id="cb19-87"><a href=""></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Posterior Distribution"</span>,</span>
<span id="cb19-88"><a href=""></a>       <span class="at">subtitle =</span> <span class="fu">paste0</span>(<span class="st">"P(Effect &gt; "</span>, clinically_meaningful, <span class="st">") = "</span>, </span>
<span id="cb19-89"><a href=""></a>                        <span class="fu">round</span>(prob_meaningful <span class="sc">*</span> <span class="dv">100</span>, <span class="dv">1</span>), <span class="st">"%"</span>),</span>
<span id="cb19-90"><a href=""></a>       <span class="at">x =</span> <span class="st">"Effect Size"</span>,</span>
<span id="cb19-91"><a href=""></a>       <span class="at">y =</span> <span class="st">"Density"</span>) <span class="sc">+</span></span>
<span id="cb19-92"><a href=""></a>  <span class="fu">theme_minimal</span>()</span>
<span id="cb19-93"><a href=""></a></span>
<span id="cb19-94"><a href=""></a><span class="co"># Combine plots</span></span>
<span id="cb19-95"><a href=""></a><span class="fu">library</span>(patchwork)</span>
<span id="cb19-96"><a href=""></a>p1 <span class="sc">+</span> p2 <span class="sc">+</span> </span>
<span id="cb19-97"><a href=""></a>  <span class="fu">plot_layout</span>(<span class="at">widths =</span> <span class="fu">c</span>(<span class="dv">3</span>, <span class="dv">2</span>)) <span class="sc">+</span></span>
<span id="cb19-98"><a href=""></a>  <span class="fu">plot_annotation</span>(</span>
<span id="cb19-99"><a href=""></a>    <span class="at">title =</span> <span class="st">"Bayesian Meta-Analysis"</span>,</span>
<span id="cb19-100"><a href=""></a>    <span class="at">subtitle =</span> <span class="st">"Forest plot and posterior distribution showing probability of clinically meaningful effect"</span></span>
<span id="cb19-101"><a href=""></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>

</div>
<img data-src="mpg6_files/figure-revealjs/unnamed-chunk-22-1.png" width="960" class="r-stretch"><aside class="notes">
<ul>
<li><p>Fixed effects (assumes same effect size)<br>
</p></li>
<li><p>Random effects (allows variation in effect size)<br>
</p></li>
<li><p>Often sensitive to outliers</p></li>
<li><p>Full posterior distribution for all parameters<br>
</p></li>
<li><p>Can incorporate informative priors<br>
</p></li>
<li><p>Naturally handles small studies (shrinkage)<br>
</p></li>
<li><p>Can model outliers explicitly<br>
</p></li>
<li><p>Direct probability statements about effects</p></li>
<li><p>Posterior probability of benefit = 98%<br>
</p></li>
<li><p>95% credible interval for effect size: [0.1, 0.5]<br>
</p></li>
<li><p>90% probability heterogeneity is moderate to high</p></li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="practical-recommendations-for-bayesian-genomics" class="slide level2">
<h2>Practical Recommendations for Bayesian Genomics</h2>
<ol type="1">
<li><strong>Informative vs flat: know your purpose</strong></li>
<li><strong>Report posterior probabilities, not just p-values</strong></li>
<li><strong>Use conjugate models for computational efficiency</strong></li>
<li><strong>Consider mixture models for complex data</strong></li>
<li><strong>Apply decision theory for optimal designs</strong></li>
</ol>
<aside class="notes">
<p>Start with informative priors when possible <strong>- Use previous studies<br>
- Incorporate functional annotations<br>
- Consider evolutionary constraints Report posterior probabilities, not just p-values</strong> - <span class="math inline">\(P(\text{association} | \text{data})\)</span> is more interpretable than <span class="math inline">\(P(\text{data} | \text{no association})\)</span><br>
- Provides direct probability statements about hypotheses Use conjugate models for computational efficiency <strong>- Beta-Binomial for allele frequencies<br>
- Dirichlet-Multinomial for haplotype frequencies<br>
- Normal-Normal for quantitative traits Consider mixture models for complex data</strong> - Population structure<br>
- Heterogeneous effect sizes<br>
- Multiple causal variants Apply decision theory for optimal designs** - Balance false positives and false negatives<br>
- Consider costs of follow-up studies<br>
- Optimize sample allocation</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="summary-when-to-use-bayesian-methods" class="slide level2">
<h2>Summary: When to Use Bayesian Methods</h2>
<table class="caption-top">
<colgroup>
<col style="width: 26%">
<col style="width: 33%">
<col style="width: 40%">
</colgroup>
<thead>
<tr class="header">
<th>Problem</th>
<th>Bayesian Approach</th>
<th>Advantage</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Multiple testing</td>
<td>Posterior probabilities</td>
<td>Direct interpretation, no arbitrary thresholds</td>
</tr>
<tr class="even">
<td>Sparse effects</td>
<td>Mixture priors</td>
<td>Better power, natural sparsity</td>
</tr>
<tr class="odd">
<td>Heterogeneous effects</td>
<td>Hierarchical models</td>
<td>Borrows strength across contexts</td>
</tr>
<tr class="even">
<td>Sequential data</td>
<td>Adaptive designs</td>
<td>Efficiency, early stopping</td>
</tr>
<tr class="odd">
<td>Prior knowledge integration</td>
<td>Informative priors</td>
<td>Improved accuracy, reduced sample size</td>
</tr>
<tr class="even">
<td>Complex dependencies</td>
<td>Bayesian networks</td>
<td>Causal inference, missing data handling</td>
</tr>
</tbody>
</table>
</section>
<section id="resources-for-bayesian-genomics" class="slide level2">
<h2>Resources for Bayesian Genomics</h2>
<p><strong>Software</strong>:<br>
- <strong>R packages</strong>: <code>rstan</code>, <code>brms</code>, <code>mashr</code>, ‘’ <code>BGLR</code>, <code>INLA</code><br>
- <strong>Python</strong>: <code>PyMC3</code>, <code>Stan</code>, <code>Edward</code><br>
- <strong>Specialized</strong>: <code>STRUCTURE</code>, <code>ADMIXTURE</code>, <code>SNPTEST</code></p>
<p><strong>Books</strong>:<br>
<a href="https://sites.stat.columbia.edu/gelman/book/BDA3.pdf">“Bayesian Data Analysis”</a> by Gelman et al.<br>
<a href="https://www.al-edu.com/wp-content/uploads/2014/05/Lindley-D.V.-Understanding-uncertainty-2006.pdf">“Understanding Uncertainty”</a>, by David Lindley<br>
“The Art of Statistics” David Spiegelhalter<br>
“Bayesian Approaches to Clinical Trials” David Spiegelhalter<br>
<a href="https://hastie.su.domains/ElemStatLearn/">“Elements of Statistical Learning”</a> Hastie, Tibshirani et al<br>
</p>
</section>
<section id="online-resources" class="slide level2">
<h2><strong>Online Resources</strong>:</h2>
<ul>
<li><a href="https://stephenslab.github.io/fiveMinuteStats/">Five Minute Stats</a><br>
</li>
<li><a href="https://mc-stan.org/users/documentation/">Stan User Guide</a><br>
</li>
<li>ME! surbut@mgh.harvard.edu</li>
</ul>
<p>All <a href="github.com/surbut/mpgtutorial">slides and codes</a></p>
</section>
<section id="questions" class="slide level2">
<h2>Questions?</h2>
<p>Thank you!!</p>
<p>Where would I be without:</p>
<ul>
<li><strong>Pradeep Natarajan, MD MMSc</strong></li>
<li><strong>Giovanni Parmigiani, PhD</strong></li>
<li><strong>Sasha Gusev, PhD</strong></li>
<li><strong>Matthew Stephens, PhD</strong></li>
</ul>
<div class="quarto-auto-generated-content">
<div class="footer footer-default">

</div>
</div>
</section></section>
    </div>
  </div>

  <script>window.backupDefine = window.define; window.define = undefined;</script>
  <script src="mpg6_files/libs/revealjs/dist/reveal.js"></script>
  <!-- reveal.js plugins -->
  <script src="mpg6_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.js"></script>
  <script src="mpg6_files/libs/revealjs/plugin/pdf-export/pdfexport.js"></script>
  <script src="mpg6_files/libs/revealjs/plugin/reveal-menu/menu.js"></script>
  <script src="mpg6_files/libs/revealjs/plugin/reveal-menu/quarto-menu.js"></script>
  <script src="mpg6_files/libs/revealjs/plugin/quarto-support/support.js"></script>
  

  <script src="mpg6_files/libs/revealjs/plugin/notes/notes.js"></script>
  <script src="mpg6_files/libs/revealjs/plugin/search/search.js"></script>
  <script src="mpg6_files/libs/revealjs/plugin/zoom/zoom.js"></script>
  <script src="mpg6_files/libs/revealjs/plugin/math/math.js"></script>
  <script>window.define = window.backupDefine; window.backupDefine = undefined;</script>

  <script>

      // Full list of configuration options available at:
      // https://revealjs.com/config/
      Reveal.initialize({
'controlsAuto': true,
'previewLinksAuto': false,
'pdfSeparateFragments': false,
'autoAnimateEasing': "ease",
'autoAnimateDuration': 1,
'autoAnimateUnmatched': true,
'menu': {"side":"left","useTextContentForMissingTitles":true,"markers":false,"loadIcons":false,"custom":[{"title":"Tools","icon":"<i class=\"fas fa-gear\"></i>","content":"<ul class=\"slide-menu-items\">\n<li class=\"slide-tool-item active\" data-item=\"0\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.fullscreen(event)\"><kbd>f</kbd> Fullscreen</a></li>\n<li class=\"slide-tool-item\" data-item=\"1\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.speakerMode(event)\"><kbd>s</kbd> Speaker View</a></li>\n<li class=\"slide-tool-item\" data-item=\"2\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.overview(event)\"><kbd>o</kbd> Slide Overview</a></li>\n<li class=\"slide-tool-item\" data-item=\"3\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.togglePdfExport(event)\"><kbd>e</kbd> PDF Export Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"4\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.keyboardHelp(event)\"><kbd>?</kbd> Keyboard Help</a></li>\n</ul>"}],"openButton":true},
'smaller': false,
 
        // Display controls in the bottom right corner
        controls: false,

        // Help the user learn the controls by providing hints, for example by
        // bouncing the down arrow when they first encounter a vertical slide
        controlsTutorial: false,

        // Determines where controls appear, "edges" or "bottom-right"
        controlsLayout: 'edges',

        // Visibility rule for backwards navigation arrows; "faded", "hidden"
        // or "visible"
        controlsBackArrows: 'faded',

        // Display a presentation progress bar
        progress: true,

        // Display the page number of the current slide
        slideNumber: 'c/t',

        // 'all', 'print', or 'speaker'
        showSlideNumber: 'all',

        // Add the current slide number to the URL hash so that reloading the
        // page/copying the URL will return you to the same slide
        hash: true,

        // Start with 1 for the hash rather than 0
        hashOneBasedIndex: false,

        // Flags if we should monitor the hash and change slides accordingly
        respondToHashChanges: true,

        // Push each slide change to the browser history
        history: true,

        // Enable keyboard shortcuts for navigation
        keyboard: true,

        // Enable the slide overview mode
        overview: true,

        // Disables the default reveal.js slide layout (scaling and centering)
        // so that you can use custom CSS layout
        disableLayout: false,

        // Vertical centering of slides
        center: false,

        // Enables touch navigation on devices with touch input
        touch: true,

        // Loop the presentation
        loop: false,

        // Change the presentation direction to be RTL
        rtl: false,

        // see https://revealjs.com/vertical-slides/#navigation-mode
        navigationMode: 'linear',

        // Randomizes the order of slides each time the presentation loads
        shuffle: false,

        // Turns fragments on and off globally
        fragments: true,

        // Flags whether to include the current fragment in the URL,
        // so that reloading brings you to the same fragment position
        fragmentInURL: false,

        // Flags if the presentation is running in an embedded mode,
        // i.e. contained within a limited portion of the screen
        embedded: false,

        // Flags if we should show a help overlay when the questionmark
        // key is pressed
        help: true,

        // Flags if it should be possible to pause the presentation (blackout)
        pause: true,

        // Flags if speaker notes should be visible to all viewers
        showNotes: false,

        // Global override for autoplaying embedded media (null/true/false)
        autoPlayMedia: null,

        // Global override for preloading lazy-loaded iframes (null/true/false)
        preloadIframes: null,

        // Number of milliseconds between automatically proceeding to the
        // next slide, disabled when set to 0, this value can be overwritten
        // by using a data-autoslide attribute on your slides
        autoSlide: 0,

        // Stop auto-sliding after user input
        autoSlideStoppable: true,

        // Use this method for navigation when auto-sliding
        autoSlideMethod: null,

        // Specify the average time in seconds that you think you will spend
        // presenting each slide. This is used to show a pacing timer in the
        // speaker view
        defaultTiming: null,

        // Enable slide navigation via mouse wheel
        mouseWheel: false,

        // The display mode that will be used to show slides
        display: 'block',

        // Hide cursor if inactive
        hideInactiveCursor: true,

        // Time before the cursor is hidden (in ms)
        hideCursorTime: 5000,

        // Opens links in an iframe preview overlay
        previewLinks: false,

        // Transition style (none/fade/slide/convex/concave/zoom)
        transition: 'slide',

        // Transition speed (default/fast/slow)
        transitionSpeed: 'default',

        // Transition style for full page slide backgrounds
        // (none/fade/slide/convex/concave/zoom)
        backgroundTransition: 'none',

        // Number of slides away from the current that are visible
        viewDistance: 3,

        // Number of slides away from the current that are visible on mobile
        // devices. It is advisable to set this to a lower number than
        // viewDistance in order to save resources.
        mobileViewDistance: 2,

        // The "normal" size of the presentation, aspect ratio will be preserved
        // when the presentation is scaled to fit different resolutions. Can be
        // specified using percentage units.
        width: 1050,

        height: 700,

        // Factor of the display size that should remain empty around the content
        margin: 0.1,

        math: {
          mathjax: 'https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.0/MathJax.js',
          config: 'TeX-AMS_HTML-full',
          tex2jax: {
            inlineMath: [['\\(','\\)']],
            displayMath: [['\\[','\\]']],
            balanceBraces: true,
            processEscapes: false,
            processRefs: true,
            processEnvironments: true,
            preview: 'TeX',
            skipTags: ['script','noscript','style','textarea','pre','code'],
            ignoreClass: 'tex2jax_ignore',
            processClass: 'tex2jax_process'
          },
        },

        // reveal.js plugins
        plugins: [QuartoLineHighlight, PdfExport, RevealMenu, QuartoSupport,

          RevealMath,
          RevealNotes,
          RevealSearch,
          RevealZoom
        ]
      });
    </script>
    
    <script>
      // htmlwidgets need to know to resize themselves when slides are shown/hidden.
      // Fire the "slideenter" event (handled by htmlwidgets.js) when the current
      // slide changes (different for each slide format).
      (function () {
        // dispatch for htmlwidgets
        function fireSlideEnter() {
          const event = window.document.createEvent("Event");
          event.initEvent("slideenter", true, true);
          window.document.dispatchEvent(event);
        }

        function fireSlideChanged(previousSlide, currentSlide) {
          fireSlideEnter();

          // dispatch for shiny
          if (window.jQuery) {
            if (previousSlide) {
              window.jQuery(previousSlide).trigger("hidden");
            }
            if (currentSlide) {
              window.jQuery(currentSlide).trigger("shown");
            }
          }
        }

        // hookup for slidy
        if (window.w3c_slidy) {
          window.w3c_slidy.add_observer(function (slide_num) {
            // slide_num starts at position 1
            fireSlideChanged(null, w3c_slidy.slides[slide_num - 1]);
          });
        }

      })();
    </script>

    <script id="quarto-html-after-body" type="application/javascript">
    window.document.addEventListener("DOMContentLoaded", function (event) {
      const toggleBodyColorMode = (bsSheetEl) => {
        const mode = bsSheetEl.getAttribute("data-mode");
        const bodyEl = window.document.querySelector("body");
        if (mode === "dark") {
          bodyEl.classList.add("quarto-dark");
          bodyEl.classList.remove("quarto-light");
        } else {
          bodyEl.classList.add("quarto-light");
          bodyEl.classList.remove("quarto-dark");
        }
      }
      const toggleBodyColorPrimary = () => {
        const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
        if (bsSheetEl) {
          toggleBodyColorMode(bsSheetEl);
        }
      }
      toggleBodyColorPrimary();  
      const tabsets =  window.document.querySelectorAll(".panel-tabset-tabby")
      tabsets.forEach(function(tabset) {
        const tabby = new Tabby('#' + tabset.id);
      });
      const isCodeAnnotation = (el) => {
        for (const clz of el.classList) {
          if (clz.startsWith('code-annotation-')) {                     
            return true;
          }
        }
        return false;
      }
      const onCopySuccess = function(e) {
        // button target
        const button = e.trigger;
        // don't keep focus
        button.blur();
        // flash "checked"
        button.classList.add('code-copy-button-checked');
        var currentTitle = button.getAttribute("title");
        button.setAttribute("title", "Copied!");
        let tooltip;
        if (window.bootstrap) {
          button.setAttribute("data-bs-toggle", "tooltip");
          button.setAttribute("data-bs-placement", "left");
          button.setAttribute("data-bs-title", "Copied!");
          tooltip = new bootstrap.Tooltip(button, 
            { trigger: "manual", 
              customClass: "code-copy-button-tooltip",
              offset: [0, -8]});
          tooltip.show();    
        }
        setTimeout(function() {
          if (tooltip) {
            tooltip.hide();
            button.removeAttribute("data-bs-title");
            button.removeAttribute("data-bs-toggle");
            button.removeAttribute("data-bs-placement");
          }
          button.setAttribute("title", currentTitle);
          button.classList.remove('code-copy-button-checked');
        }, 1000);
        // clear code selection
        e.clearSelection();
      }
      const getTextToCopy = function(trigger) {
          const codeEl = trigger.previousElementSibling.cloneNode(true);
          for (const childEl of codeEl.children) {
            if (isCodeAnnotation(childEl)) {
              childEl.remove();
            }
          }
          return codeEl.innerText;
      }
      const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
        text: getTextToCopy
      });
      clipboard.on('success', onCopySuccess);
      if (window.document.getElementById('quarto-embedded-source-code-modal')) {
        // For code content inside modals, clipBoardJS needs to be initialized with a container option
        // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
        const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
          text: getTextToCopy,
          container: window.document.getElementById('quarto-embedded-source-code-modal')
        });
        clipboardModal.on('success', onCopySuccess);
      }
        var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
        var mailtoRegex = new RegExp(/^mailto:/);
          var filterRegex = new RegExp('/' + window.location.host + '/');
        var isInternal = (href) => {
            return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
        }
        // Inspect non-navigation links and adorn them if external
     	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
        for (var i=0; i<links.length; i++) {
          const link = links[i];
          if (!isInternal(link.href)) {
            // undo the damage that might have been done by quarto-nav.js in the case of
            // links that we want to consider external
            if (link.dataset.originalHref !== undefined) {
              link.href = link.dataset.originalHref;
            }
          }
        }
      function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
        const config = {
          allowHTML: true,
          maxWidth: 500,
          delay: 100,
          arrow: false,
          appendTo: function(el) {
              return el.closest('section.slide') || el.parentElement;
          },
          interactive: true,
          interactiveBorder: 10,
          theme: 'light-border',
          placement: 'bottom-start',
        };
        if (contentFn) {
          config.content = contentFn;
        }
        if (onTriggerFn) {
          config.onTrigger = onTriggerFn;
        }
        if (onUntriggerFn) {
          config.onUntrigger = onUntriggerFn;
        }
          config['offset'] = [0,0];
          config['maxWidth'] = 700;
        window.tippy(el, config); 
      }
      const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
      for (var i=0; i<noterefs.length; i++) {
        const ref = noterefs[i];
        tippyHover(ref, function() {
          // use id or data attribute instead here
          let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
          try { href = new URL(href).hash; } catch {}
          const id = href.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note) {
            return note.innerHTML;
          } else {
            return "";
          }
        });
      }
      const findCites = (el) => {
        const parentEl = el.parentElement;
        if (parentEl) {
          const cites = parentEl.dataset.cites;
          if (cites) {
            return {
              el,
              cites: cites.split(' ')
            };
          } else {
            return findCites(el.parentElement)
          }
        } else {
          return undefined;
        }
      };
      var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
      for (var i=0; i<bibliorefs.length; i++) {
        const ref = bibliorefs[i];
        const citeInfo = findCites(ref);
        if (citeInfo) {
          tippyHover(citeInfo.el, function() {
            var popup = window.document.createElement('div');
            citeInfo.cites.forEach(function(cite) {
              var citeDiv = window.document.createElement('div');
              citeDiv.classList.add('hanging-indent');
              citeDiv.classList.add('csl-entry');
              var biblioDiv = window.document.getElementById('ref-' + cite);
              if (biblioDiv) {
                citeDiv.innerHTML = biblioDiv.innerHTML;
              }
              popup.appendChild(citeDiv);
            });
            return popup.innerHTML;
          });
        }
      }
    });
    </script>
    

</body></html>