<!DOCTYPE html>
<html lang="en"><head>
<script src="mpgbayes_files/libs/clipboard/clipboard.min.js"></script>
<script src="mpgbayes_files/libs/quarto-html/tabby.min.js"></script>
<script src="mpgbayes_files/libs/quarto-html/popper.min.js"></script>
<script src="mpgbayes_files/libs/quarto-html/tippy.umd.min.js"></script>
<link href="mpgbayes_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="mpgbayes_files/libs/quarto-html/light-border.css" rel="stylesheet">
<link href="mpgbayes_files/libs/quarto-html/quarto-html.min.css" rel="stylesheet" data-mode="light">
<link href="mpgbayes_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles"><meta charset="utf-8">
  <meta name="generator" content="quarto-1.5.57">

  <meta name="author" content="Sarah Urbut, MD PhD">
  <title>Bayesian Statistics in Population Genomics</title>
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">
  <link rel="stylesheet" href="mpgbayes_files/libs/revealjs/dist/reset.css">
  <link rel="stylesheet" href="mpgbayes_files/libs/revealjs/dist/reveal.css">
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    div.columns{display: flex; gap: min(4vw, 1.5em);}
    div.column{flex: auto; overflow-x: auto;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    ul.task-list li input[type="checkbox"] {
      width: 0.8em;
      margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
      vertical-align: middle;
    }
    /* CSS for syntax highlighting */
    pre > code.sourceCode { white-space: pre; position: relative; }
    pre > code.sourceCode > span { line-height: 1.25; }
    pre > code.sourceCode > span:empty { height: 1.2em; }
    .sourceCode { overflow: visible; }
    code.sourceCode > span { color: inherit; text-decoration: inherit; }
    div.sourceCode { margin: 1em 0; }
    pre.sourceCode { margin: 0; }
    @media screen {
    div.sourceCode { overflow: auto; }
    }
    @media print {
    pre > code.sourceCode { white-space: pre-wrap; }
    pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
    }
    pre.numberSource code
      { counter-reset: source-line 0; }
    pre.numberSource code > span
      { position: relative; left: -4em; counter-increment: source-line; }
    pre.numberSource code > span > a:first-child::before
      { content: counter(source-line);
        position: relative; left: -1em; text-align: right; vertical-align: baseline;
        border: none; display: inline-block;
        -webkit-touch-callout: none; -webkit-user-select: none;
        -khtml-user-select: none; -moz-user-select: none;
        -ms-user-select: none; user-select: none;
        padding: 0 4px; width: 4em;
        color: #aaaaaa;
      }
    pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
    div.sourceCode
      { color: #003b4f; background-color: #f1f3f5; }
    @media screen {
    pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
    }
    code span { color: #003b4f; } /* Normal */
    code span.al { color: #ad0000; } /* Alert */
    code span.an { color: #5e5e5e; } /* Annotation */
    code span.at { color: #657422; } /* Attribute */
    code span.bn { color: #ad0000; } /* BaseN */
    code span.bu { } /* BuiltIn */
    code span.cf { color: #003b4f; font-weight: bold; } /* ControlFlow */
    code span.ch { color: #20794d; } /* Char */
    code span.cn { color: #8f5902; } /* Constant */
    code span.co { color: #5e5e5e; } /* Comment */
    code span.cv { color: #5e5e5e; font-style: italic; } /* CommentVar */
    code span.do { color: #5e5e5e; font-style: italic; } /* Documentation */
    code span.dt { color: #ad0000; } /* DataType */
    code span.dv { color: #ad0000; } /* DecVal */
    code span.er { color: #ad0000; } /* Error */
    code span.ex { } /* Extension */
    code span.fl { color: #ad0000; } /* Float */
    code span.fu { color: #4758ab; } /* Function */
    code span.im { color: #00769e; } /* Import */
    code span.in { color: #5e5e5e; } /* Information */
    code span.kw { color: #003b4f; font-weight: bold; } /* Keyword */
    code span.op { color: #5e5e5e; } /* Operator */
    code span.ot { color: #003b4f; } /* Other */
    code span.pp { color: #ad0000; } /* Preprocessor */
    code span.sc { color: #5e5e5e; } /* SpecialChar */
    code span.ss { color: #20794d; } /* SpecialString */
    code span.st { color: #20794d; } /* String */
    code span.va { color: #111111; } /* Variable */
    code span.vs { color: #20794d; } /* VerbatimString */
    code span.wa { color: #5e5e5e; font-style: italic; } /* Warning */
  </style>
  <link rel="stylesheet" href="mpgbayes_files/libs/revealjs/dist/theme/quarto.css">
  <link href="mpgbayes_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.css" rel="stylesheet">
  <link href="mpgbayes_files/libs/revealjs/plugin/reveal-menu/menu.css" rel="stylesheet">
  <link href="mpgbayes_files/libs/revealjs/plugin/reveal-menu/quarto-menu.css" rel="stylesheet">
  <link href="mpgbayes_files/libs/revealjs/plugin/quarto-support/footer.css" rel="stylesheet">
  <style type="text/css">

  .callout {
    margin-top: 1em;
    margin-bottom: 1em;  
    border-radius: .25rem;
  }

  .callout.callout-style-simple { 
    padding: 0em 0.5em;
    border-left: solid #acacac .3rem;
    border-right: solid 1px silver;
    border-top: solid 1px silver;
    border-bottom: solid 1px silver;
    display: flex;
  }

  .callout.callout-style-default {
    border-left: solid #acacac .3rem;
    border-right: solid 1px silver;
    border-top: solid 1px silver;
    border-bottom: solid 1px silver;
  }

  .callout .callout-body-container {
    flex-grow: 1;
  }

  .callout.callout-style-simple .callout-body {
    font-size: 1rem;
    font-weight: 400;
  }

  .callout.callout-style-default .callout-body {
    font-size: 0.9rem;
    font-weight: 400;
  }

  .callout.callout-titled.callout-style-simple .callout-body {
    margin-top: 0.2em;
  }

  .callout:not(.callout-titled) .callout-body {
      display: flex;
  }

  .callout:not(.no-icon).callout-titled.callout-style-simple .callout-content {
    padding-left: 1.6em;
  }

  .callout.callout-titled .callout-header {
    padding-top: 0.2em;
    margin-bottom: -0.2em;
  }

  .callout.callout-titled .callout-title  p {
    margin-top: 0.5em;
    margin-bottom: 0.5em;
  }
    
  .callout.callout-titled.callout-style-simple .callout-content  p {
    margin-top: 0;
  }

  .callout.callout-titled.callout-style-default .callout-content  p {
    margin-top: 0.7em;
  }

  .callout.callout-style-simple div.callout-title {
    border-bottom: none;
    font-size: .9rem;
    font-weight: 600;
    opacity: 75%;
  }

  .callout.callout-style-default  div.callout-title {
    border-bottom: none;
    font-weight: 600;
    opacity: 85%;
    font-size: 0.9rem;
    padding-left: 0.5em;
    padding-right: 0.5em;
  }

  .callout.callout-style-default div.callout-content {
    padding-left: 0.5em;
    padding-right: 0.5em;
  }

  .callout.callout-style-simple .callout-icon::before {
    height: 1rem;
    width: 1rem;
    display: inline-block;
    content: "";
    background-repeat: no-repeat;
    background-size: 1rem 1rem;
  }

  .callout.callout-style-default .callout-icon::before {
    height: 0.9rem;
    width: 0.9rem;
    display: inline-block;
    content: "";
    background-repeat: no-repeat;
    background-size: 0.9rem 0.9rem;
  }

  .callout-title {
    display: flex
  }
    
  .callout-icon::before {
    margin-top: 1rem;
    padding-right: .5rem;
  }

  .callout.no-icon::before {
    display: none !important;
  }

  .callout.callout-titled .callout-body > .callout-content > :last-child {
    padding-bottom: 0.5rem;
    margin-bottom: 0;
  }

  .callout.callout-titled .callout-icon::before {
    margin-top: .5rem;
    padding-right: .5rem;
  }

  .callout:not(.callout-titled) .callout-icon::before {
    margin-top: 1rem;
    padding-right: .5rem;
  }

  /* Callout Types */

  div.callout-note {
    border-left-color: #4582ec !important;
  }

  div.callout-note .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAEU0lEQVRYCcVXTWhcVRQ+586kSUMMxkyaElstCto2SIhitS5Ek8xUKV2poatCcVHtUlFQk8mbaaziwpWgglJwVaquitBOfhQXFlqlzSJpFSpIYyXNjBNiTCck7x2/8/LeNDOZxDuEkgOXe++553zfefee+/OYLOXFk3+1LLrRdiO81yNqZ6K9cG0P3MeFaMIQjXssE8Z1JzLO9ls20MBZX7oG8w9GxB0goaPrW5aNMp1yOZIa7Wv6o2ykpLtmAPs/vrG14Z+6d4jpbSKuhdcSyq9wGMPXjonwmESXrriLzFGOdDBLB8Y6MNYBu0dRokSygMA/mrun8MGFN3behm6VVAwg4WR3i6FvYK1T7MHo9BK7ydH+1uurECoouk5MPRyVSBrBHMYwVobG2aOXM07sWrn5qgB60rc6mcwIDJtQrnrEr44kmy+UO9r0u9O5/YbkS9juQckLed3DyW2XV/qWBBB3ptvI8EUY3I9p/67OW+g967TNr3Sotn3IuVlfMLVnsBwH4fsnebJvyGm5GeIUA3jljERmrv49SizPYuq+z7c2H/jlGC+Ghhupn/hcapqmcudB9jwJ/3jvnvu6vu5lVzF1fXyZuZZ7U8nRmVzytvT+H3kilYvH09mLWrQdwFSsFEsxFVs5fK7A0g8gMZjbif4ACpKbjv7gNGaD8bUrlk8x+KRflttr22JEMRUbTUwwDQScyzPgedQHZT0xnx7ujw2jfVfExwYHwOsDTjLdJ2ebmeQIlJ7neo41s/DrsL3kl+W2lWvAga0tR3zueGr6GL78M3ifH0rGXrBC2aAR8uYcIA5gwV8zIE8onoh8u0Fca/ciF7j1uOzEnqcIm59sEXoGc0+z6+H45V1CvAvHcD7THztu669cnp+L0okAeIc6zjbM/24LgGM1gZk7jnRu1aQWoU9sfUOuhrmtaPIO3YY1KLLWZaEO5TKUbMY5zx8W9UJ6elpLwKXbsaZ4EFl7B4bMtDv0iRipKoDQT2sNQI9b1utXFdYisi+wzZ/ri/1m7QfDgEuvgUUEIJPq3DhX/5DWNqIXDOweC2wvIR90Oq3lDpdMIgD2r0dXvGdsEW5H6x6HLRJYU7C69VefO1x8Gde1ZFSJLfWS1jbCnhtOPxmpfv2LXOA2Xk2tvnwKKPFuZ/oRmwBwqRQDcKNeVQkYcOjtWVBuM/JuYw5b6isojIkYxyYAFn5K7ZBF10fea52y8QltAg6jnMqNHFBmGkQ1j+U43HMi2xMar1Nv0zGsf1s8nUsmUtPOOrbFIR8bHFDMB5zL13Gmr/kGlCkUzedTzzmzsaJXhYawnA3UmARpiYj5ooJZiUoxFRtK3X6pgNPv+IZVPcnwbOl6f+aBaO1CNvPW9n9LmCp01nuSaTRF2YxHqZ8DYQT6WsXT+RD6eUztwYLZ8rM+rcPxamv1VQzFUkzFXvkiVrySGQgJNvXHJAxiU3/NwiC03rSf05VBaPtu/Z7/B8Yn/w7eguloAAAAAElFTkSuQmCC');
  }

  div.callout-note.callout-style-default .callout-title {
    background-color: #dae6fb
  }

  div.callout-important {
    border-left-color: #d9534f !important;
  }

  div.callout-important .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAEKklEQVRYCcVXTWhcVRS+575MJym48A+hSRFr00ySRQhURRfd2HYjk2SSTokuBCkU2o0LoSKKraKIBTcuFCoidGFD08nkBzdREbpQ1EDNIv8qSGMFUboImMSZd4/f9zJv8ibJMC8xJQfO3HPPPef7zrvvvnvviIkpC9nsw0UttFunbUhpFzFtarSd6WJkStVMw5xyVqYTvkwfzuf/5FgtkVoB0729j1rjXwThS7Vio+Mo6DNnvLfahoZ+i/o32lULuJ3NNiz7q6+pyAUkJaFF6JwaM2lUJlV0MlnQn5aTRbEu0SEqHUa0A4AdiGuB1kFXRfVyg5d87+Dg4DL6m2TLAub60ilj7A1Ec4odSAc8X95sHh7+ZRPCFo6Fnp7HfU/fBng/hi10CjCnWnJjsxvDNxWw0NfV6Rv5GgP3I3jGWXumdTD/3cbEOP2ZbOZp69yniG3FQ9z1jD7bnBu9Fc2tKGC2q+uAJOQHBDRiZX1x36o7fWBs7J9ownbtO+n0/qWkvW7UPIfc37WgT6ZGR++EOJyeQDSb9UB+DZ1G6DdLDzyS+b/kBCYGsYgJbSQHuThGKRcw5xdeQf8YdNHsc6ePXrlSYMBuSIAFTGAtQo+VuALo4BX83N190NWZWbynBjhOHsmNfFWLeL6v+ynsA58zDvvAC8j5PkbOcXCMg2PZFk3q8MjI7WAG/Dp9AwP7jdGBOOQkAvlFUB+irtm16I1Zw9YBcpGTGXYmk3kQIC/Cds55l+iMI3jqhjAuaoe+am2Jw5GT3Nbz3CkE12NavmzN5+erJW7046n/CH1RO/RVa8lBLozXk9uqykkGAyRXLWlLv5jyp4RFsG5vGVzpDLnIjTWgnRy2Rr+tDKvRc7Y8AyZq10jj8DqXdnIRNtFZb+t/ZRtXcDiVnzpqx8mPcDWxgARUqx0W1QB9MeUZiNrV4qP+Ehc+BpNgATsTX8ozYKL2NtFYAHc84fG7ndxUPr+AR/iQSns7uSUufAymwDOb2+NjK27lEFocm/EE2WpyIy/Hi66MWuMKJn8RvxIcj87IM5Vh9663ziW36kR0HNenXuxmfaD8JC7tfKbrhFr7LiZCrMjrzTeGx+PmkosrkNzW94ObzwocJ7A1HokLolY+AvkTiD/q1H0cN48c5EL8Crkttsa/AXQVDmutfyku0E7jShx49XqV3MFK8IryDhYVbj7Sj2P2eBxwcXoe8T8idsKKPRcnZw1b+slFTubwUwhktrfnAt7J++jwQtLZcm3sr9LQrjRzz6cfMv9aLvgmnAGvpoaGLxM4mAEaLV7iAzQ3oU0IvD5x9ix3yF2RAAuYAOO2f7PEFWCXZ4C9Pb2UsgDeVnFSpbFK7/IWu7TPTvBqzbGdCHOJQSxiEjt6IyZmxQyEJHv6xyQsYk//moVFsN2zP6fRImjfq7/n/wFDguUQFNEwugAAAABJRU5ErkJggg==');
  }

  div.callout-important.callout-style-default .callout-title {
    background-color: #f7dddc
  }

  div.callout-warning {
    border-left-color: #f0ad4e !important;
  }

  div.callout-warning .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAETklEQVRYCeVWW2gcVRg+58yaTUnizqbipZeX4uWhBEniBaoUX1Ioze52t7sRq6APio9V9MEaoWlVsFasRq0gltaAPuxms8lu0gcviE/FFOstVbSIxgcv6SU7EZqmdc7v9+9mJtNks51NTUH84ed889/PP+cmxP+d5FIbMJmNbpREu4WUkiTtCicKny0l1pIKmBzovF2S+hIJHX8iEu3hZJ5lNZGqyRrGSIQpq15AzF28jgpeY6yk6GVdrfFqdrD6Iw+QlB8g0YS2g7dyQmXM/IDhBhT0UCiRf59lfqmmDvzRt6kByV/m4JjtzuaujMUM2c5Z2d6JdKrRb3K2q6mA+oYVz8JnDdKPmmNthzkAk/lN63sYPgevrguc72aZX/L9C6x09GYyxBgCX4NlvyGUHOKELlm5rXeR1kchuChJt4SSwyddZRXgvwMGvYo4QSlk3/zkHD8UHxwVJA6zjZZqP8v8kK8OWLnIZtLyCAJagYC4rTGW/9Pqj92N/c+LUaAj27movwbi19tk/whRCIE7Q9vyI6yvRpftAKVTdUjOW40X3h5OXsKCdmFcx0xlLJoSuQngnrJe7Kcjm4OMq9FlC7CMmScQANuNvjfP3PjGXDBaUQmbp296S5L4DrpbrHN1T87ZVEZVCzg1FF0Ft+dKrlLukI+/c9ENo+TvlTDbYFvuKPtQ9+l052rXrgKoWkDAFnvh0wTOmYn8R5f4k/jN/fZiCM1tQx9jQQ4ANhqG4hiL0qIFTGViG9DKB7GYzgubnpofgYRwO+DFjh0Zin2m4b/97EDkXkc+f6xYAPX0KK2I/7fUQuwzuwo/L3AkcjugPNixC8cHf0FyPjWlItmLxWw4Ou9YsQCr5fijMGoD/zpdRy95HRysyXA74MWOnscpO4j2y3HAVisw85hX5+AFBRSHt4ShfLFkIMXTqyKFc46xdzQM6XbAi702a7sy04J0+feReMFKp5q9esYLCqAZYw/k14E/xcLLsFElaornTuJB0svMuJINy8xkIYuL+xPAlWRceH6+HX7THJ0djLUom46zREu7tTkxwmf/FdOZ/sh6Q8qvEAiHpm4PJ4a/doJe0gH1t+aHRgCzOvBvJedEK5OFE5jpm4AGP2a8Dxe3gGJ/pAutug9Gp6he92CsSsWBaEcxGx0FHytmIpuqGkOpldqNYQK8cSoXvd+xLxXADw0kf6UkJNFtdo5MOgaLjiQOQHcn+A6h5NuL2s0qsC2LOM75PcF3yr5STuBSAcGG+meA14K/CI21HcS4LBT6tv0QAh8Dr5l93AhZzG5ZJ4VxAqdZUEl9z7WJ4aN+svMvwHHL21UKTd1mqvChH7/Za5xzXBBKrUcB0TQ+Ulgkfbi/H/YT5EptrGzsEK7tR1B7ln9BBwckYfMiuSqklSznIuoIIOM42MQO+QnduCoFCI0bpkzjCjddHPN/F+2Yu+sd9bKNpVwHhbS3LluK/0zgfwD0xYI5dXuzlQAAAABJRU5ErkJggg==');
  }

  div.callout-warning.callout-style-default .callout-title {
    background-color: #fcefdc
  }

  div.callout-tip {
    border-left-color: #02b875 !important;
  }

  div.callout-tip .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAADr0lEQVRYCe1XTWgTQRj9ZjZV8a9SPIkKgj8I1bMHsUWrqYLVg4Ue6v9BwZOxSYsIerFao7UiUryIqJcqgtpimhbBXoSCVxUFe9CTiogUrUp2Pt+3aUI2u5vdNh4dmMzOzHvvezuz8xNFM0mjnbXaNu1MvFWRXkXEyE6aYOYJpdW4IXuA4r0fo8qqSMDBU0v1HJUgVieAXxzCsdE/YJTdFcVIZQNMyhruOMJKXYFoLfIfIvVIMWdsrd+Rpd86ZmyzzjJmLStqRn0v8lzkb4rVIXvnpScOJuAn2ACC65FkPzEdEy4TPWRLJ2h7z4cArXzzaOdKlbOvKKX25Wl00jSnrwVxAg3o4dRxhO13RBSdNvH0xSARv3adTXbBdTf64IWO2vH0LT+cv4GR1DJt+DUItaQogeBX/chhbTBxEiZ6gftlDNXTrvT7co4ub5A6gp9HIcHvzTa46OS5fBeP87Qm0fQkr4FsYgVQ7Qg+ZayaDg9jhg1GkWj8RG6lkeSacrrHgDaxdoBiZPg+NXV/KifMuB6//JmYH4CntVEHy/keA6x4h4CU5oFy8GzrBS18cLJMXcljAKB6INjWsRcuZBWVaS3GDrqB7rdapVIeA+isQ57Eev9eCqzqOa81CY05VLd6SamW2wA2H3SiTbnbSxmzfp7WtKZkqy4mdyAlGx7ennghYf8voqp9cLSgKdqNfa6RdRsAAkPwRuJZNbpByn+RrJi1RXTwdi8RQF6ymDwGMAtZ6TVE+4uoKh+MYkcLsT0Hk8eAienbiGdjJHZTpmNjlbFJNKDVAp2fJlYju6IreQxQ08UJDNYdoLSl6AadO+fFuCQqVMB1NJwPm69T04Wv5WhfcWyfXQB+wXRs1pt+nCknRa0LVzSA/2B+a9+zQJadb7IyyV24YAxKp2Jqs3emZTuNnKxsah+uabKbMk7CbTgJx/zIgQYErIeTKRQ9yD9wxVof5YolPHqaWo7TD6tJlh7jQnK5z2n3+fGdggIOx2kaa2YI9QWarc5Ce1ipNWMKeSG4DysFF52KBmTNMmn5HqCFkwy34rDg05gDwgH3bBi+sgFhN/e8QvRn8kbamCOhgrZ9GJhFDgfcMHzFb6BAtjKpFhzTjwv1KCVuxHvCbsSiEz4CANnj84cwHdFXAbAOJ4LTSAawGWFn5tDhLMYz6nWeU2wJfIhmIJBefcd/A5FWQWGgrWzyORZ3Q6HuV+Jf0Bj+BTX69fm1zWgK7By1YTXchFDORywnfQ7GpzOo6S+qECrsx2ifVQAAAABJRU5ErkJggg==');
  }

  div.callout-tip.callout-style-default .callout-title {
    background-color: #ccf1e3
  }

  div.callout-caution {
    border-left-color: #fd7e14 !important;
  }

  div.callout-caution .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAACV0lEQVRYCdVWzWoUQRCuqp2ICBLJXgITZL1EfQDBW/bkzUMUD7klD+ATSHBEfAIfQO+iXsWDxJsHL96EHAwhgzlkg8nBg25XWb0zIb0zs9muYYWkoKeru+vn664fBqElyZNuyh167NXJ8Ut8McjbmEraKHkd7uAnAFku+VWdb3reSmRV8PKSLfZ0Gjn3a6Xlcq9YGb6tADjn+lUfTXtVmaZ1KwBIvFI11rRXlWlatwIAAv2asaa9mlB9wwygiDX26qaw1yYPzFXg2N1GgG0FMF8Oj+VIx7E/03lHx8UhvYyNZLN7BwSPgekXXLribw7w5/c8EF+DBK5idvDVYtEEwMeYefjjLAdEyQ3M9nfOkgnPTEkYU+sxMq0BxNR6jExrAI31H1rzvLEfRIdgcv1XEdj6QTQAS2wtstEALLG1yEZ3QhH6oDX7ExBSFEkFINXH98NTrme5IOaaA7kIfiu2L8A3qhH9zRbukdCqdsA98TdElyeMe5BI8Rs2xHRIsoTSSVFfCFCWGPn9XHb4cdobRIWABNf0add9jakDjQJpJ1bTXOJXnnRXHRf+dNL1ZV1MBRCXhMbaHqGI1JkKIL7+i8uffuP6wVQAzO7+qVEbF6NbS0LJureYcWXUUhH66nLR5rYmva+2tjRFtojkM2aD76HEGAD3tPtKM309FJg5j/K682ywcWJ3PASCcycH/22u+Bh7Aa0ehM2Fu4z0SAE81HF9RkB21c5bEn4Dzw+/qNOyXr3DCTQDMBOdhi4nAgiFDGCinIa2owCEChUwD8qzd03PG+qdW/4fDzjUMcE1ZpIAAAAASUVORK5CYII=');
  }

  div.callout-caution.callout-style-default .callout-title {
    background-color: #ffe5d0
  }

  </style>
  <style type="text/css">
    .reveal div.sourceCode {
      margin: 0;
      overflow: auto;
    }
    .reveal div.hanging-indent {
      margin-left: 1em;
      text-indent: -1em;
    }
    .reveal .slide:not(.center) {
      height: 100%;
    }
    .reveal .slide.scrollable {
      overflow-y: auto;
    }
    .reveal .footnotes {
      height: 100%;
      overflow-y: auto;
    }
    .reveal .slide .absolute {
      position: absolute;
      display: block;
    }
    .reveal .footnotes ol {
      counter-reset: ol;
      list-style-type: none; 
      margin-left: 0;
    }
    .reveal .footnotes ol li:before {
      counter-increment: ol;
      content: counter(ol) ". "; 
    }
    .reveal .footnotes ol li > p:first-child {
      display: inline-block;
    }
    .reveal .slide ul,
    .reveal .slide ol {
      margin-bottom: 0.5em;
    }
    .reveal .slide ul li,
    .reveal .slide ol li {
      margin-top: 0.4em;
      margin-bottom: 0.2em;
    }
    .reveal .slide ul[role="tablist"] li {
      margin-bottom: 0;
    }
    .reveal .slide ul li > *:first-child,
    .reveal .slide ol li > *:first-child {
      margin-block-start: 0;
    }
    .reveal .slide ul li > *:last-child,
    .reveal .slide ol li > *:last-child {
      margin-block-end: 0;
    }
    .reveal .slide .columns:nth-child(3) {
      margin-block-start: 0.8em;
    }
    .reveal blockquote {
      box-shadow: none;
    }
    .reveal .tippy-content>* {
      margin-top: 0.2em;
      margin-bottom: 0.7em;
    }
    .reveal .tippy-content>*:last-child {
      margin-bottom: 0.2em;
    }
    .reveal .slide > img.stretch.quarto-figure-center,
    .reveal .slide > img.r-stretch.quarto-figure-center {
      display: block;
      margin-left: auto;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-left,
    .reveal .slide > img.r-stretch.quarto-figure-left  {
      display: block;
      margin-left: 0;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-right,
    .reveal .slide > img.r-stretch.quarto-figure-right  {
      display: block;
      margin-left: auto;
      margin-right: 0; 
    }
  </style>
</head>
<body class="quarto-light">
  <div class="reveal">
    <div class="slides">

<section id="title-slide" class="quarto-title-block center">
  <h1 class="title">Bayesian Statistics in Population Genomics</h1>
  <p class="subtitle">From P-values to Posterior Probabilities</p>

<div class="quarto-title-authors">
<div class="quarto-title-author">
<div class="quarto-title-author-name">
Sarah Urbut, MD PhD 
</div>
</div>
</div>

</section>
<section id="overview" class="slide level2">
<h2>Overview</h2>
<p>In this primer, we’ll explore key Bayesian concepts critical for modern genomics:</p>
<ol type="1">
<li><strong>P-values vs.&nbsp;Posterior Probabilities</strong>: Why Bayesian thinking helps avoid misinterpretations</li>
<li><strong>Conjugate Models</strong>: Elegant solutions for population genetic inference</li>
<li><strong>Mixture Models</strong>: Powerful tools for complex genomic data</li>
<li><strong>Bayesian Clinical &amp; Adaptive Designs</strong>: Learning and adapting as data accumulates</li>
</ol>
</section>
<section class="slide level2">

</section>
<section>
<section id="p-values-vs.-posterior-probabilities" class="title-slide slide level1 center">
<h1>P-values vs.&nbsp;Posterior Probabilities</h1>

</section>
<section id="the-fallacy-of-p-values" class="slide level2">
<h2>The Fallacy of P-values</h2>
<div>
<ul>
<li class="fragment"><p>P-values answer a <strong>counterfactual question</strong>: “If there were no effect, how surprising would these data be?”</p></li>
<li class="fragment"><p>But researchers want to know: “<strong>What is the probability this association is real?</strong>”</p></li>
<li class="fragment"><p>This disconnect leads to systematic misinterpretation</p></li>
</ul>
</div>
</section>
<section id="a-gwas-example" class="slide level2">
<h2>A GWAS Example</h2>
<div class="columns">
<div class="column" style="width:50%;">
<p><strong>Same evidence, different conclusions:</strong></p>
<ul>
<li><p>SNP with p = 1 × 10⁻⁵ in GWAS</p></li>
<li><p>Traditional: “Nearly significant!”</p></li>
<li><p>Bayesian: “Probably a false positive”</p></li>
</ul>
<p><strong>Why the difference?</strong></p>
</div><div class="column" style="width:50%;">
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href=""></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb1-2"><a href=""></a><span class="fu">library</span>(dplyr)</span>
<span id="cb1-3"><a href=""></a></span>
<span id="cb1-4"><a href=""></a><span class="co"># Function to calculate posterior probability</span></span>
<span id="cb1-5"><a href=""></a>calc_posterior <span class="ot">&lt;-</span> <span class="cf">function</span>(prior, p_value, <span class="at">n_snps=</span><span class="fl">1e6</span>) {</span>
<span id="cb1-6"><a href=""></a>  <span class="co"># Convert p-value to z-score</span></span>
<span id="cb1-7"><a href=""></a>  z <span class="ot">&lt;-</span> <span class="fu">qnorm</span>(p_value<span class="sc">/</span><span class="dv">2</span>, <span class="at">lower.tail =</span> <span class="cn">FALSE</span>)</span>
<span id="cb1-8"><a href=""></a>  <span class="co"># Approximate Bayes factor</span></span>
<span id="cb1-9"><a href=""></a>  bf <span class="ot">&lt;-</span> <span class="fu">exp</span>(z<span class="sc">^</span><span class="dv">2</span><span class="sc">/</span><span class="dv">2</span>)</span>
<span id="cb1-10"><a href=""></a>  <span class="co"># Calculate posterior</span></span>
<span id="cb1-11"><a href=""></a>  posterior <span class="ot">&lt;-</span> (prior <span class="sc">*</span> bf) <span class="sc">/</span> (prior <span class="sc">*</span> bf <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">-</span> prior))</span>
<span id="cb1-12"><a href=""></a>  <span class="fu">return</span>(posterior)</span>
<span id="cb1-13"><a href=""></a>}</span>
<span id="cb1-14"><a href=""></a></span>
<span id="cb1-15"><a href=""></a><span class="co"># Create data</span></span>
<span id="cb1-16"><a href=""></a>p_values <span class="ot">&lt;-</span> <span class="dv">10</span><span class="sc">^</span><span class="fu">seq</span>(<span class="sc">-</span><span class="dv">8</span>, <span class="sc">-</span><span class="dv">2</span>, <span class="fl">0.5</span>)</span>
<span id="cb1-17"><a href=""></a>posteriors <span class="ot">&lt;-</span> <span class="fu">sapply</span>(p_values, <span class="cf">function</span>(p) <span class="fu">calc_posterior</span>(<span class="dv">1</span><span class="sc">/</span><span class="dv">10000</span>, p))</span>
<span id="cb1-18"><a href=""></a></span>
<span id="cb1-19"><a href=""></a><span class="co"># Create plot</span></span>
<span id="cb1-20"><a href=""></a>df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb1-21"><a href=""></a>  <span class="at">p_value =</span> p_values,</span>
<span id="cb1-22"><a href=""></a>  <span class="at">posterior =</span> posteriors</span>
<span id="cb1-23"><a href=""></a>)</span>
<span id="cb1-24"><a href=""></a></span>
<span id="cb1-25"><a href=""></a><span class="fu">ggplot</span>(df, <span class="fu">aes</span>(<span class="at">x =</span> <span class="sc">-</span><span class="fu">log10</span>(p_value), <span class="at">y =</span> posterior)) <span class="sc">+</span></span>
<span id="cb1-26"><a href=""></a>  <span class="fu">geom_line</span>(<span class="at">size =</span> <span class="fl">1.5</span>, <span class="at">color =</span> <span class="st">"blue"</span>) <span class="sc">+</span></span>
<span id="cb1-27"><a href=""></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="fl">0.5</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">color =</span> <span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb1-28"><a href=""></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fl">5e-8</span>), <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">color =</span> <span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb1-29"><a href=""></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="dv">8</span>, <span class="at">y =</span> <span class="fl">0.2</span>, <span class="at">label =</span> <span class="st">"GWAS</span><span class="sc">\n</span><span class="st">significance"</span>, <span class="at">color =</span> <span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb1-30"><a href=""></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="dv">4</span>, <span class="at">y =</span> <span class="fl">0.55</span>, <span class="at">label =</span> <span class="st">"50% posterior</span><span class="sc">\n</span><span class="st">probability"</span>, <span class="at">color =</span> <span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb1-31"><a href=""></a>  <span class="fu">theme_minimal</span>(<span class="at">base_size =</span> <span class="dv">14</span>) <span class="sc">+</span></span>
<span id="cb1-32"><a href=""></a>  <span class="fu">labs</span>(</span>
<span id="cb1-33"><a href=""></a>    <span class="at">title =</span> <span class="st">"P-value vs. Posterior Probability"</span>,</span>
<span id="cb1-34"><a href=""></a>    <span class="at">subtitle =</span> <span class="st">"Prior probability = 1/10,000"</span>,</span>
<span id="cb1-35"><a href=""></a>    <span class="at">x =</span> <span class="st">"-log10(p-value)"</span>,</span>
<span id="cb1-36"><a href=""></a>    <span class="at">y =</span> <span class="st">"Posterior probability of association"</span></span>
<span id="cb1-37"><a href=""></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure>
<p><img data-src="mpgbayes_files/figure-revealjs/unnamed-chunk-1-1.png" width="576"></p>
</figure>
</div>
</div>
</div>
</div></div>
</section>
<section id="bayes-rule-explained" class="slide level2 smaller">
<h2>Bayes’ Rule Explained</h2>
<p><span class="math display">\[ P(H|D) = \frac{P(D|H) \times P(H)}{P(D)} \]</span></p>
<p>In GWAS context:</p>
<p><span class="math display">\[ P(\text{Real association}|\text{Data}) = \frac{P(\text{Data}|\text{Real association}) \times P(\text{Real association})}{P(\text{Data})} \]</span></p>
<div>
<ul>
<li class="fragment">The <strong>prior probability</strong> matters tremendously:
<ul>
<li class="fragment"><strong>Rare diseases</strong>: Stronger priors for genes in relevant pathways</li>
<li class="fragment"><strong>Common traits</strong>: Most variants have tiny or no effects</li>
<li class="fragment"><strong>eQTLs</strong>: Higher prior near transcription start sites</li>
</ul></li>
</ul>
</div>
</section>
<section id="effect-of-different-priors" class="slide level2 smaller">
<h2>Effect of Different Priors</h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href=""></a><span class="co"># Calculate posteriors for different priors</span></span>
<span id="cb2-2"><a href=""></a>priors <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">1</span><span class="sc">/</span><span class="dv">1000</span>, <span class="dv">1</span><span class="sc">/</span><span class="dv">10000</span>, <span class="dv">1</span><span class="sc">/</span><span class="dv">100000</span>)</span>
<span id="cb2-3"><a href=""></a>prior_labels <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Strong prior (1/1,000)"</span>, </span>
<span id="cb2-4"><a href=""></a>                 <span class="st">"Moderate prior (1/10,000)"</span>, </span>
<span id="cb2-5"><a href=""></a>                 <span class="st">"Weak prior (1/100,000)"</span>)</span>
<span id="cb2-6"><a href=""></a></span>
<span id="cb2-7"><a href=""></a>results <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb2-8"><a href=""></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(priors)) {</span>
<span id="cb2-9"><a href=""></a>  posteriors <span class="ot">&lt;-</span> <span class="fu">sapply</span>(p_values, <span class="cf">function</span>(p) <span class="fu">calc_posterior</span>(priors[i], p))</span>
<span id="cb2-10"><a href=""></a>  results <span class="ot">&lt;-</span> <span class="fu">rbind</span>(results, <span class="fu">data.frame</span>(</span>
<span id="cb2-11"><a href=""></a>    <span class="at">p_value =</span> p_values,</span>
<span id="cb2-12"><a href=""></a>    <span class="at">posterior =</span> posteriors,</span>
<span id="cb2-13"><a href=""></a>    <span class="at">prior =</span> prior_labels[i]</span>
<span id="cb2-14"><a href=""></a>  ))</span>
<span id="cb2-15"><a href=""></a>}</span>
<span id="cb2-16"><a href=""></a></span>
<span id="cb2-17"><a href=""></a><span class="co"># Create plot</span></span>
<span id="cb2-18"><a href=""></a><span class="fu">ggplot</span>(results, <span class="fu">aes</span>(<span class="at">x =</span> <span class="sc">-</span><span class="fu">log10</span>(p_value), <span class="at">y =</span> posterior, <span class="at">color =</span> prior)) <span class="sc">+</span></span>
<span id="cb2-19"><a href=""></a>  <span class="fu">geom_line</span>(<span class="at">size =</span> <span class="fl">1.5</span>) <span class="sc">+</span></span>
<span id="cb2-20"><a href=""></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="fl">0.5</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">color =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb2-21"><a href=""></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fl">5e-8</span>), <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">color =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb2-22"><a href=""></a>  <span class="fu">theme_minimal</span>(<span class="at">base_size =</span> <span class="dv">14</span>) <span class="sc">+</span></span>
<span id="cb2-23"><a href=""></a>  <span class="fu">scale_color_brewer</span>(<span class="at">palette =</span> <span class="st">"Set1"</span>) <span class="sc">+</span></span>
<span id="cb2-24"><a href=""></a>  <span class="fu">labs</span>(</span>
<span id="cb2-25"><a href=""></a>    <span class="at">title =</span> <span class="st">"Effect of Prior Probabilities on Posterior Inference"</span>,</span>
<span id="cb2-26"><a href=""></a>    <span class="at">x =</span> <span class="st">"-log10(p-value)"</span>,</span>
<span id="cb2-27"><a href=""></a>    <span class="at">y =</span> <span class="st">"Posterior probability of association"</span>,</span>
<span id="cb2-28"><a href=""></a>    <span class="at">color =</span> <span class="st">"Prior probability"</span></span>
<span id="cb2-29"><a href=""></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>

</div>
<img data-src="mpgbayes_files/figure-revealjs/unnamed-chunk-2-1.png" width="960" class="r-stretch"><div class="fragment">
<p><strong>Key insight</strong>: In genomics, informative priors come from: - Functional annotations (coding, regulatory, etc.) - Evolutionary conservation - Previous studies in related traits</p>
</div>
</section>
<section id="likelihood-ratios-the-bridge" class="slide level2 smaller">
<h2>Likelihood Ratios: The Bridge</h2>
<p>The likelihood ratio (LR) directly quantifies evidence:</p>
<p><span class="math display">\[ LR = \frac{P(D|H_1)}{P(D|H_0)} \]</span></p>
<div class="columns">
<div class="column" style="width:50%;">
<p><strong>Jeffreys’ scale for LR:</strong></p>
<table class="caption-top">
<thead>
<tr class="header">
<th>LR</th>
<th>Evidence</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>1-3</td>
<td>Barely worth mentioning</td>
</tr>
<tr class="even">
<td>3-10</td>
<td>Substantial</td>
</tr>
<tr class="odd">
<td>10-30</td>
<td>Strong</td>
</tr>
<tr class="even">
<td>30-100</td>
<td>Very strong</td>
</tr>
<tr class="odd">
<td>&gt;100</td>
<td>Decisive</td>
</tr>
</tbody>
</table>
</div><div class="column" style="width:50%;">
<p><strong>Converting p-values to LR:</strong></p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href=""></a>p_vals <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fl">0.05</span>, <span class="fl">0.01</span>, <span class="fl">0.001</span>, <span class="fl">1e-5</span>, <span class="fl">1e-8</span>)</span>
<span id="cb3-2"><a href=""></a>z_scores <span class="ot">&lt;-</span> <span class="fu">qnorm</span>(p_vals<span class="sc">/</span><span class="dv">2</span>, <span class="at">lower.tail =</span> <span class="cn">FALSE</span>)</span>
<span id="cb3-3"><a href=""></a>approx_lr <span class="ot">&lt;-</span> <span class="fu">exp</span>(z_scores<span class="sc">^</span><span class="dv">2</span><span class="sc">/</span><span class="dv">2</span>)</span>
<span id="cb3-4"><a href=""></a></span>
<span id="cb3-5"><a href=""></a><span class="fu">data.frame</span>(</span>
<span id="cb3-6"><a href=""></a>  <span class="at">p_value =</span> p_vals,</span>
<span id="cb3-7"><a href=""></a>  <span class="at">approx_LR =</span> <span class="fu">round</span>(approx_lr, <span class="dv">1</span>)</span>
<span id="cb3-8"><a href=""></a>) <span class="sc">%&gt;%</span></span>
<span id="cb3-9"><a href=""></a>  knitr<span class="sc">::</span><span class="fu">kable</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<table class="caption-top">
<thead>
<tr class="header">
<th style="text-align: right;">p_value</th>
<th style="text-align: right;">approx_LR</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: right;">5e-02</td>
<td style="text-align: right;">6.8</td>
</tr>
<tr class="even">
<td style="text-align: right;">1e-02</td>
<td style="text-align: right;">27.6</td>
</tr>
<tr class="odd">
<td style="text-align: right;">1e-03</td>
<td style="text-align: right;">224.5</td>
</tr>
<tr class="even">
<td style="text-align: right;">1e-05</td>
<td style="text-align: right;">17252.5</td>
</tr>
<tr class="odd">
<td style="text-align: right;">0e+00</td>
<td style="text-align: right;">13532796.7</td>
</tr>
</tbody>
</table>
</div>
</div>
</div></div>
<div class="fragment">
<p><strong>Advantage</strong>: LR is independent of priors, providing a direct measure of evidence strength</p>
</div>
</section>
<section class="slide level2">

</section></section>
<section>
<section id="conjugate-models-in-population-genomics" class="title-slide slide level1 center">
<h1>Conjugate Models in Population Genomics</h1>

</section>
<section id="conjugate-distributions-why-they-matter" class="slide level2 smaller">
<h2>Conjugate Distributions: Why They Matter</h2>
<div>
<ul>
<li class="fragment"><p><strong>Conjugate prior</strong>: When prior and posterior come from the same family of distributions</p></li>
<li class="fragment"><p><strong>Why important in genomics</strong>:</p>
<ul>
<li class="fragment">Analytically tractable (closed-form solutions)</li>
<li class="fragment">Intuitive interpretation of parameters</li>
<li class="fragment">Computational efficiency for large-scale analyses</li>
<li class="fragment">Natural way to incorporate previous knowledge</li>
</ul></li>
</ul>
</div>
</section>
<section id="beta-binomial-the-population-geneticists-friend" class="slide level2 smaller">
<h2>Beta-Binomial: The Population Geneticist’s Friend</h2>
<div class="columns">
<div class="column" style="width:50%;">
<p><strong>The model:</strong> - <strong>Prior</strong>: Beta(α, β) for allele frequency - <strong>Data</strong>: Binomial sampling (observed alleles) - <strong>Posterior</strong>: Beta(α + counts, β + non-counts)</p>
<p><strong>Perfect for:</strong> - Allele frequency estimation - Mutation rate estimation - Selection coefficient inference</p>
</div><div class="column" style="width:50%;">
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href=""></a><span class="co"># Function to plot Beta distributions</span></span>
<span id="cb4-2"><a href=""></a>plot_beta <span class="ot">&lt;-</span> <span class="cf">function</span>(alpha, beta, label) {</span>
<span id="cb4-3"><a href=""></a>  x <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="at">length.out =</span> <span class="dv">500</span>)</span>
<span id="cb4-4"><a href=""></a>  y <span class="ot">&lt;-</span> <span class="fu">dbeta</span>(x, alpha, beta)</span>
<span id="cb4-5"><a href=""></a>  <span class="fu">data.frame</span>(<span class="at">x =</span> x, <span class="at">y =</span> y, <span class="at">distribution =</span> label)</span>
<span id="cb4-6"><a href=""></a>}</span>
<span id="cb4-7"><a href=""></a></span>
<span id="cb4-8"><a href=""></a><span class="co"># Create example data</span></span>
<span id="cb4-9"><a href=""></a>prior_data <span class="ot">&lt;-</span> <span class="fu">plot_beta</span>(<span class="dv">1</span>, <span class="dv">1</span>, <span class="st">"Prior: Beta(1,1)"</span>)</span>
<span id="cb4-10"><a href=""></a>likelihood_data <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb4-11"><a href=""></a>  <span class="at">x =</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="at">length.out =</span> <span class="dv">500</span>),</span>
<span id="cb4-12"><a href=""></a>  <span class="at">y =</span> <span class="fu">dbinom</span>(<span class="dv">70</span>, <span class="dv">100</span>, <span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="at">length.out =</span> <span class="dv">500</span>)),</span>
<span id="cb4-13"><a href=""></a>  <span class="at">distribution =</span> <span class="st">"Likelihood: 70/100 reference alleles"</span></span>
<span id="cb4-14"><a href=""></a>)</span>
<span id="cb4-15"><a href=""></a>likelihood_data<span class="sc">$</span>y <span class="ot">&lt;-</span> likelihood_data<span class="sc">$</span>y <span class="sc">/</span> <span class="fu">max</span>(likelihood_data<span class="sc">$</span>y) <span class="sc">*</span> <span class="fu">max</span>(prior_data<span class="sc">$</span>y) <span class="sc">*</span> <span class="fl">0.8</span></span>
<span id="cb4-16"><a href=""></a>posterior_data <span class="ot">&lt;-</span> <span class="fu">plot_beta</span>(<span class="dv">1</span><span class="sc">+</span><span class="dv">70</span>, <span class="dv">1</span><span class="sc">+</span><span class="dv">30</span>, <span class="st">"Posterior: Beta(71,31)"</span>)</span>
<span id="cb4-17"><a href=""></a></span>
<span id="cb4-18"><a href=""></a><span class="co"># Combine data</span></span>
<span id="cb4-19"><a href=""></a>plot_data <span class="ot">&lt;-</span> <span class="fu">rbind</span>(prior_data, posterior_data)</span>
<span id="cb4-20"><a href=""></a></span>
<span id="cb4-21"><a href=""></a><span class="co"># Create plot</span></span>
<span id="cb4-22"><a href=""></a><span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb4-23"><a href=""></a>  <span class="fu">geom_line</span>(<span class="at">data =</span> plot_data, <span class="fu">aes</span>(<span class="at">x =</span> x, <span class="at">y =</span> y, <span class="at">color =</span> distribution), <span class="at">size =</span> <span class="fl">1.5</span>) <span class="sc">+</span></span>
<span id="cb4-24"><a href=""></a>  <span class="fu">geom_line</span>(<span class="at">data =</span> likelihood_data, <span class="fu">aes</span>(<span class="at">x =</span> x, <span class="at">y =</span> y, <span class="at">color =</span> distribution), </span>
<span id="cb4-25"><a href=""></a>            <span class="at">size =</span> <span class="fl">1.5</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb4-26"><a href=""></a>  <span class="fu">theme_minimal</span>(<span class="at">base_size =</span> <span class="dv">14</span>) <span class="sc">+</span></span>
<span id="cb4-27"><a href=""></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"blue"</span>, <span class="st">"red"</span>, <span class="st">"darkgreen"</span>)) <span class="sc">+</span></span>
<span id="cb4-28"><a href=""></a>  <span class="fu">labs</span>(</span>
<span id="cb4-29"><a href=""></a>    <span class="at">title =</span> <span class="st">"Beta-Binomial Conjugacy"</span>,</span>
<span id="cb4-30"><a href=""></a>    <span class="at">x =</span> <span class="st">"Allele Frequency"</span>,</span>
<span id="cb4-31"><a href=""></a>    <span class="at">y =</span> <span class="st">"Density"</span>,</span>
<span id="cb4-32"><a href=""></a>    <span class="at">color =</span> <span class="st">""</span></span>
<span id="cb4-33"><a href=""></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure>
<p><img data-src="mpgbayes_files/figure-revealjs/unnamed-chunk-4-1.png" width="960"></p>
</figure>
</div>
</div>
</div>
</div></div>
</section>
<section id="beta-binomial-as-pseudo-counts" class="slide level2 smaller">
<h2>Beta-Binomial as “Pseudo-counts”</h2>
<p>The Beta prior parameters (α, β) can be interpreted as <strong>pseudo-counts</strong>:</p>
<ul>
<li><strong>Beta(1, 1)</strong>: Uniform prior (no pseudo-counts)</li>
<li><strong>Beta(0.5, 0.5)</strong>: Jeffreys prior (favors extremes)</li>
<li><strong>Beta(10, 30)</strong>: Strong prior centered at 0.25</li>
</ul>
<div class="fragment">
<p><strong>In practice</strong>: α and β represent “prior observations”</p>
<p>For a sample with 20 A alleles and 30 B alleles: - With Beta(1, 1) prior: Posterior is Beta(21, 31) - Estimate = 21/(21+31) = 0.404 (close to data)</p>
<ul>
<li>With Beta(10, 30) prior: Posterior is Beta(30, 60)</li>
<li>Estimate = 30/(30+60) = 0.333 (pulled toward prior)</li>
</ul>
</div>
</section>
<section id="conjugate-priors-in-structure" class="slide level2 smaller">
<h2>Conjugate Priors in STRUCTURE</h2>
<p>STRUCTURE uses Dirichlet-Multinomial conjugacy:</p>
<div>
<ol type="1">
<li class="fragment"><strong>For each population k and locus j</strong>:
<ul>
<li class="fragment">Prior on allele frequencies: Dirichlet(α₁, α₂, …, αᵦ)</li>
<li class="fragment">Data: Observed allele counts in that population</li>
<li class="fragment">Posterior: Updated Dirichlet with added counts</li>
</ul></li>
<li class="fragment"><strong>For each individual i</strong>:
<ul>
<li class="fragment">Prior on population membership: Dirichlet(λ₁, λ₂, …, λₖ)</li>
<li class="fragment">Data: Genotype information across loci</li>
<li class="fragment">Posterior: Updated admixture proportions</li>
</ul></li>
</ol>
</div>
<div class="fragment">
<p><strong>Why conjugacy matters</strong>: Enables efficient Gibbs sampling in the MCMC algorithm - essential for making STRUCTURE computationally feasible</p>
</div>
</section>
<section id="dirichlet-prior-parameters-matter" class="slide level2 smaller">
<h2>Dirichlet Prior Parameters Matter</h2>
<div class="columns">
<div class="column" style="width:60%;">
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href=""></a><span class="fu">library</span>(tidyr)</span>
<span id="cb5-2"><a href=""></a></span>
<span id="cb5-3"><a href=""></a><span class="co"># Function to sample from a Dirichlet distribution</span></span>
<span id="cb5-4"><a href=""></a>rdirichlet <span class="ot">&lt;-</span> <span class="cf">function</span>(n, alpha) {</span>
<span id="cb5-5"><a href=""></a>  k <span class="ot">&lt;-</span> <span class="fu">length</span>(alpha)</span>
<span id="cb5-6"><a href=""></a>  r <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="fu">rgamma</span>(n<span class="sc">*</span>k, <span class="at">shape=</span>alpha, <span class="at">rate=</span><span class="dv">1</span>), <span class="at">ncol=</span>k, <span class="at">byrow=</span><span class="cn">TRUE</span>)</span>
<span id="cb5-7"><a href=""></a>  r <span class="ot">&lt;-</span> r <span class="sc">/</span> <span class="fu">rowSums</span>(r)</span>
<span id="cb5-8"><a href=""></a>  <span class="fu">return</span>(r)</span>
<span id="cb5-9"><a href=""></a>}</span>
<span id="cb5-10"><a href=""></a></span>
<span id="cb5-11"><a href=""></a><span class="co"># Sample from different Dirichlet priors</span></span>
<span id="cb5-12"><a href=""></a><span class="fu">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb5-13"><a href=""></a>n_samples <span class="ot">&lt;-</span> <span class="dv">1000</span></span>
<span id="cb5-14"><a href=""></a></span>
<span id="cb5-15"><a href=""></a><span class="co"># Very small alpha (sparse)</span></span>
<span id="cb5-16"><a href=""></a>alpha_small <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fl">0.1</span>, <span class="fl">0.1</span>, <span class="fl">0.1</span>)  </span>
<span id="cb5-17"><a href=""></a>samples_small <span class="ot">&lt;-</span> <span class="fu">rdirichlet</span>(n_samples, alpha_small)</span>
<span id="cb5-18"><a href=""></a></span>
<span id="cb5-19"><a href=""></a><span class="co"># Unity alpha (uniform)</span></span>
<span id="cb5-20"><a href=""></a>alpha_unity <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>)</span>
<span id="cb5-21"><a href=""></a>samples_unity <span class="ot">&lt;-</span> <span class="fu">rdirichlet</span>(n_samples, alpha_unity)</span>
<span id="cb5-22"><a href=""></a></span>
<span id="cb5-23"><a href=""></a><span class="co"># Large alpha (concentrated)</span></span>
<span id="cb5-24"><a href=""></a>alpha_large <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">10</span>, <span class="dv">10</span>, <span class="dv">10</span>) </span>
<span id="cb5-25"><a href=""></a>samples_large <span class="ot">&lt;-</span> <span class="fu">rdirichlet</span>(n_samples, alpha_large)</span>
<span id="cb5-26"><a href=""></a></span>
<span id="cb5-27"><a href=""></a><span class="co"># Create data frames</span></span>
<span id="cb5-28"><a href=""></a>df_small <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(samples_small)</span>
<span id="cb5-29"><a href=""></a>df_small<span class="sc">$</span>prior <span class="ot">&lt;-</span> <span class="st">"Sparse: Dirichlet(0.1,0.1,0.1)"</span></span>
<span id="cb5-30"><a href=""></a><span class="fu">names</span>(df_small)[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>] <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Pop1"</span>, <span class="st">"Pop2"</span>, <span class="st">"Pop3"</span>)</span>
<span id="cb5-31"><a href=""></a></span>
<span id="cb5-32"><a href=""></a>df_unity <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(samples_unity)</span>
<span id="cb5-33"><a href=""></a>df_unity<span class="sc">$</span>prior <span class="ot">&lt;-</span> <span class="st">"Uniform: Dirichlet(1,1,1)"</span></span>
<span id="cb5-34"><a href=""></a><span class="fu">names</span>(df_unity)[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>] <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Pop1"</span>, <span class="st">"Pop2"</span>, <span class="st">"Pop3"</span>)</span>
<span id="cb5-35"><a href=""></a></span>
<span id="cb5-36"><a href=""></a>df_large <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(samples_large)</span>
<span id="cb5-37"><a href=""></a>df_large<span class="sc">$</span>prior <span class="ot">&lt;-</span> <span class="st">"Concentrated: Dirichlet(10,10,10)"</span></span>
<span id="cb5-38"><a href=""></a><span class="fu">names</span>(df_large)[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>] <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Pop1"</span>, <span class="st">"Pop2"</span>, <span class="st">"Pop3"</span>)</span>
<span id="cb5-39"><a href=""></a></span>
<span id="cb5-40"><a href=""></a><span class="co"># Combine data</span></span>
<span id="cb5-41"><a href=""></a>df_all <span class="ot">&lt;-</span> <span class="fu">rbind</span>(df_small, df_unity, df_large)</span>
<span id="cb5-42"><a href=""></a></span>
<span id="cb5-43"><a href=""></a><span class="co"># Convert to long format for ggplot</span></span>
<span id="cb5-44"><a href=""></a>df_long <span class="ot">&lt;-</span> df_all <span class="sc">%&gt;%</span></span>
<span id="cb5-45"><a href=""></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">c</span>(<span class="st">"Pop1"</span>, <span class="st">"Pop2"</span>, <span class="st">"Pop3"</span>), </span>
<span id="cb5-46"><a href=""></a>               <span class="at">names_to =</span> <span class="st">"Population"</span>, </span>
<span id="cb5-47"><a href=""></a>               <span class="at">values_to =</span> <span class="st">"Proportion"</span>)</span>
<span id="cb5-48"><a href=""></a></span>
<span id="cb5-49"><a href=""></a><span class="co"># Create ggplot</span></span>
<span id="cb5-50"><a href=""></a><span class="fu">ggplot</span>(df_long, <span class="fu">aes</span>(<span class="at">x =</span> Proportion, <span class="at">fill =</span> Population)) <span class="sc">+</span></span>
<span id="cb5-51"><a href=""></a>  <span class="fu">geom_density</span>(<span class="at">alpha =</span> <span class="fl">0.6</span>) <span class="sc">+</span></span>
<span id="cb5-52"><a href=""></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span> prior, <span class="at">ncol =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb5-53"><a href=""></a>  <span class="fu">theme_minimal</span>(<span class="at">base_size =</span> <span class="dv">12</span>) <span class="sc">+</span></span>
<span id="cb5-54"><a href=""></a>  <span class="fu">labs</span>(</span>
<span id="cb5-55"><a href=""></a>    <span class="at">title =</span> <span class="st">"Effect of Dirichlet Prior Parameters"</span>,</span>
<span id="cb5-56"><a href=""></a>    <span class="at">x =</span> <span class="st">"Population Proportion"</span>,</span>
<span id="cb5-57"><a href=""></a>    <span class="at">y =</span> <span class="st">"Density"</span></span>
<span id="cb5-58"><a href=""></a>  ) <span class="sc">+</span></span>
<span id="cb5-59"><a href=""></a>  <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">"Set2"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure>
<p><img data-src="mpgbayes_files/figure-revealjs/unnamed-chunk-5-1.png" width="960"></p>
</figure>
</div>
</div>
</div>
</div><div class="column" style="width:40%;">
<p><strong>Interpreting the priors</strong>:</p>
<ul>
<li><p><strong>Sparse prior</strong>: Favors solutions where each individual belongs to mostly one population</p></li>
<li><p><strong>Uniform prior</strong>: No preference for admixture level</p></li>
<li><p><strong>Concentrated prior</strong>: Favors solutions where individuals have similar proportions of each ancestry</p></li>
</ul>
<p><strong>In STRUCTURE</strong>: - For population allele frequencies, typically use small α (0.1-1) - For admixture proportions, λ controls expected admixture level</p>
</div></div>
</section>
<section class="slide level2">

</section></section>
<section>
<section id="mixture-models-in-genomics" class="title-slide slide level1 center">
<h1>Mixture Models in Genomics</h1>

</section>
<section id="mixture-models-the-concept" class="slide level2 smaller">
<h2>Mixture Models: The Concept</h2>
<p>A mixture model assumes data comes from multiple distinct distributions:</p>
<p><span class="math display">\[ p(x) = \sum_{k=1}^K \pi_k f_k(x|\theta_k) \]</span></p>
<p>Where: - <span class="math inline">\(\pi_k\)</span> are the mixing weights (<span class="math inline">\(\sum \pi_k = 1\)</span>) - <span class="math inline">\(f_k\)</span> are the component distributions - <span class="math inline">\(\theta_k\)</span> are the parameters of each component</p>
<div class="fragment">
<p><strong>Key genomic applications</strong>: - Cell type deconvolution - Population structure (STRUCTURE) - Gene expression clustering - QTL effect sharing across tissues (mash)</p>
</div>
</section>
<section id="em-algorithm-the-workhorse" class="slide level2 smaller">
<h2>EM Algorithm: The Workhorse</h2>
<div class="columns">
<div class="column" style="width:60%;">
<p>The EM algorithm iteratively:</p>
<ol type="1">
<li><p><strong>E-step</strong>: Calculate responsibilities (posterior probability each data point comes from each component) <span class="math display">\[\gamma_{ik} = \frac{\pi_k f_k(x_i|\theta_k)}{\sum_j \pi_j f_j(x_i|\theta_j)}\]</span></p></li>
<li><p><strong>M-step</strong>: Update parameters using weighted data <span class="math display">\[\pi_k^{new} = \frac{1}{n}\sum_{i=1}^n \gamma_{ik}\]</span> <span class="math display">\[\theta_k^{new} = \arg\max_{\theta_k} \sum_{i=1}^n \gamma_{ik} \log f_k(x_i|\theta_k)\]</span></p></li>
</ol>
</div><div class="column" style="width:40%;">
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href=""></a><span class="co"># Simulate data from a mixture of two normal distributions</span></span>
<span id="cb6-2"><a href=""></a><span class="fu">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb6-3"><a href=""></a>n <span class="ot">&lt;-</span> <span class="dv">500</span></span>
<span id="cb6-4"><a href=""></a>z <span class="ot">&lt;-</span> <span class="fu">rbinom</span>(n, <span class="dv">1</span>, <span class="fl">0.6</span>)  <span class="co"># Component memberships</span></span>
<span id="cb6-5"><a href=""></a>x <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(z <span class="sc">==</span> <span class="dv">0</span>, <span class="fu">rnorm</span>(n, <span class="sc">-</span><span class="dv">2</span>, <span class="fl">0.7</span>), <span class="fu">rnorm</span>(n, <span class="dv">2</span>, <span class="dv">1</span>))</span>
<span id="cb6-6"><a href=""></a></span>
<span id="cb6-7"><a href=""></a><span class="co"># Create plot data</span></span>
<span id="cb6-8"><a href=""></a>hist_data <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">x =</span> x)</span>
<span id="cb6-9"><a href=""></a></span>
<span id="cb6-10"><a href=""></a><span class="co"># Create plot</span></span>
<span id="cb6-11"><a href=""></a><span class="fu">ggplot</span>(hist_data, <span class="fu">aes</span>(<span class="at">x =</span> x)) <span class="sc">+</span></span>
<span id="cb6-12"><a href=""></a>  <span class="fu">geom_histogram</span>(<span class="fu">aes</span>(<span class="at">y =</span> <span class="fu">after_stat</span>(density)), </span>
<span id="cb6-13"><a href=""></a>                 <span class="at">bins =</span> <span class="dv">30</span>, </span>
<span id="cb6-14"><a href=""></a>                 <span class="at">fill =</span> <span class="st">"lightblue"</span>, </span>
<span id="cb6-15"><a href=""></a>                 <span class="at">color =</span> <span class="st">"black"</span>, </span>
<span id="cb6-16"><a href=""></a>                 <span class="at">alpha =</span> <span class="fl">0.7</span>) <span class="sc">+</span></span>
<span id="cb6-17"><a href=""></a>  <span class="fu">stat_function</span>(<span class="at">fun =</span> <span class="cf">function</span>(x) <span class="fl">0.6</span> <span class="sc">*</span> <span class="fu">dnorm</span>(x, <span class="dv">2</span>, <span class="dv">1</span>), </span>
<span id="cb6-18"><a href=""></a>                <span class="fu">aes</span>(<span class="at">color =</span> <span class="st">"Component 1"</span>), </span>
<span id="cb6-19"><a href=""></a>                <span class="at">size =</span> <span class="fl">1.2</span>) <span class="sc">+</span></span>
<span id="cb6-20"><a href=""></a>  <span class="fu">stat_function</span>(<span class="at">fun =</span> <span class="cf">function</span>(x) <span class="fl">0.4</span> <span class="sc">*</span> <span class="fu">dnorm</span>(x, <span class="sc">-</span><span class="dv">2</span>, <span class="fl">0.7</span>), </span>
<span id="cb6-21"><a href=""></a>                <span class="fu">aes</span>(<span class="at">color =</span> <span class="st">"Component 2"</span>), </span>
<span id="cb6-22"><a href=""></a>                <span class="at">size =</span> <span class="fl">1.2</span>) <span class="sc">+</span></span>
<span id="cb6-23"><a href=""></a>  <span class="fu">stat_function</span>(<span class="at">fun =</span> <span class="cf">function</span>(x) <span class="fl">0.6</span> <span class="sc">*</span> <span class="fu">dnorm</span>(x, <span class="dv">2</span>, <span class="dv">1</span>) <span class="sc">+</span> <span class="fl">0.4</span> <span class="sc">*</span> <span class="fu">dnorm</span>(x, <span class="sc">-</span><span class="dv">2</span>, <span class="fl">0.7</span>), </span>
<span id="cb6-24"><a href=""></a>                <span class="fu">aes</span>(<span class="at">color =</span> <span class="st">"Mixture"</span>), </span>
<span id="cb6-25"><a href=""></a>                <span class="at">size =</span> <span class="fl">1.5</span>) <span class="sc">+</span></span>
<span id="cb6-26"><a href=""></a>  <span class="fu">theme_minimal</span>(<span class="at">base_size =</span> <span class="dv">12</span>) <span class="sc">+</span></span>
<span id="cb6-27"><a href=""></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"Component 1"</span> <span class="ot">=</span> <span class="st">"red"</span>, </span>
<span id="cb6-28"><a href=""></a>                              <span class="st">"Component 2"</span> <span class="ot">=</span> <span class="st">"blue"</span>, </span>
<span id="cb6-29"><a href=""></a>                              <span class="st">"Mixture"</span> <span class="ot">=</span> <span class="st">"purple"</span>)) <span class="sc">+</span></span>
<span id="cb6-30"><a href=""></a>  <span class="fu">labs</span>(</span>
<span id="cb6-31"><a href=""></a>    <span class="at">title =</span> <span class="st">"Gaussian Mixture Model"</span>,</span>
<span id="cb6-32"><a href=""></a>    <span class="at">x =</span> <span class="st">"Value"</span>,</span>
<span id="cb6-33"><a href=""></a>    <span class="at">y =</span> <span class="st">"Density"</span>,</span>
<span id="cb6-34"><a href=""></a>    <span class="at">color =</span> <span class="st">""</span></span>
<span id="cb6-35"><a href=""></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure>
<p><img data-src="mpgbayes_files/figure-revealjs/unnamed-chunk-6-1.png" width="960"></p>
</figure>
</div>
</div>
</div>
</div></div>
</section>
<section id="cell-type-deconvolution-example" class="slide level2 smaller">
<h2>Cell Type Deconvolution Example</h2>
<div class="columns">
<div class="column" style="width:50%;">
<p><strong>Problem</strong>: Tissue samples contain multiple cell types</p>
<p><strong>Mixture model interpretation</strong>: - Each cell type has a distinct expression profile - Observed expression is a weighted sum of profiles - Weights = cell type proportions</p>
<p><strong>Components</strong>: - <span class="math inline">\(f_k(x|\theta_k)\)</span> = Expression profile of cell type k - <span class="math inline">\(\pi_k\)</span> = Proportion of cell type k in sample</p>
</div><div class="column" style="width:50%;">
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href=""></a><span class="co"># Simulate gene expression data for deconvolution</span></span>
<span id="cb7-2"><a href=""></a><span class="fu">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb7-3"><a href=""></a>n_genes <span class="ot">&lt;-</span> <span class="dv">500</span></span>
<span id="cb7-4"><a href=""></a>n_cell_types <span class="ot">&lt;-</span> <span class="dv">3</span></span>
<span id="cb7-5"><a href=""></a>n_samples <span class="ot">&lt;-</span> <span class="dv">50</span></span>
<span id="cb7-6"><a href=""></a></span>
<span id="cb7-7"><a href=""></a><span class="co"># True cell type proportions for each sample</span></span>
<span id="cb7-8"><a href=""></a>cell_props <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="fu">runif</span>(n_samples <span class="sc">*</span> n_cell_types), <span class="at">ncol =</span> n_cell_types)</span>
<span id="cb7-9"><a href=""></a>cell_props <span class="ot">&lt;-</span> cell_props <span class="sc">/</span> <span class="fu">rowSums</span>(cell_props)</span>
<span id="cb7-10"><a href=""></a><span class="fu">colnames</span>(cell_props) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"CellType1"</span>, <span class="st">"CellType2"</span>, <span class="st">"CellType3"</span>)</span>
<span id="cb7-11"><a href=""></a></span>
<span id="cb7-12"><a href=""></a><span class="co"># True expression profiles for each cell type</span></span>
<span id="cb7-13"><a href=""></a>cell_profiles <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="fu">rnorm</span>(n_genes <span class="sc">*</span> n_cell_types, <span class="at">mean =</span> <span class="dv">5</span>, <span class="at">sd =</span> <span class="dv">1</span>), </span>
<span id="cb7-14"><a href=""></a>                        <span class="at">nrow =</span> n_genes)</span>
<span id="cb7-15"><a href=""></a><span class="co"># Make some genes specific to cell types</span></span>
<span id="cb7-16"><a href=""></a>cell_profiles[<span class="dv">1</span><span class="sc">:</span><span class="dv">50</span>, <span class="dv">1</span>] <span class="ot">&lt;-</span> cell_profiles[<span class="dv">1</span><span class="sc">:</span><span class="dv">50</span>, <span class="dv">1</span>] <span class="sc">+</span> <span class="dv">3</span>  <span class="co"># Cell type 1 markers</span></span>
<span id="cb7-17"><a href=""></a>cell_profiles[<span class="dv">51</span><span class="sc">:</span><span class="dv">100</span>, <span class="dv">2</span>] <span class="ot">&lt;-</span> cell_profiles[<span class="dv">51</span><span class="sc">:</span><span class="dv">100</span>, <span class="dv">2</span>] <span class="sc">+</span> <span class="dv">3</span>  <span class="co"># Cell type 2 markers</span></span>
<span id="cb7-18"><a href=""></a>cell_profiles[<span class="dv">101</span><span class="sc">:</span><span class="dv">150</span>, <span class="dv">3</span>] <span class="ot">&lt;-</span> cell_profiles[<span class="dv">101</span><span class="sc">:</span><span class="dv">150</span>, <span class="dv">3</span>] <span class="sc">+</span> <span class="dv">3</span>  <span class="co"># Cell type 3 markers</span></span>
<span id="cb7-19"><a href=""></a></span>
<span id="cb7-20"><a href=""></a><span class="co"># Generate mixed samples</span></span>
<span id="cb7-21"><a href=""></a>mixed_expression <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="dv">0</span>, <span class="at">nrow =</span> n_genes, <span class="at">ncol =</span> n_samples)</span>
<span id="cb7-22"><a href=""></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>n_samples) {</span>
<span id="cb7-23"><a href=""></a>  mixed_expression[, i] <span class="ot">&lt;-</span> cell_profiles <span class="sc">%*%</span> cell_props[i, ] <span class="sc">+</span> </span>
<span id="cb7-24"><a href=""></a>                           <span class="fu">rnorm</span>(n_genes, <span class="dv">0</span>, <span class="fl">0.1</span>)  <span class="co"># Add noise</span></span>
<span id="cb7-25"><a href=""></a>}</span>
<span id="cb7-26"><a href=""></a></span>
<span id="cb7-27"><a href=""></a><span class="co"># Create heatmap data</span></span>
<span id="cb7-28"><a href=""></a>example_genes <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">25</span>, <span class="dv">75</span>, <span class="dv">125</span>)  <span class="co"># Marker genes for each cell type</span></span>
<span id="cb7-29"><a href=""></a>example_data <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb7-30"><a href=""></a>  <span class="at">Gene =</span> <span class="fu">rep</span>(<span class="fu">paste0</span>(<span class="st">"Gene"</span>, example_genes), <span class="at">each =</span> n_samples),</span>
<span id="cb7-31"><a href=""></a>  <span class="at">Sample =</span> <span class="fu">rep</span>(<span class="dv">1</span><span class="sc">:</span>n_samples, <span class="at">times =</span> <span class="fu">length</span>(example_genes)),</span>
<span id="cb7-32"><a href=""></a>  <span class="at">Expression =</span> <span class="fu">c</span>(mixed_expression[example_genes[<span class="dv">1</span>], ],</span>
<span id="cb7-33"><a href=""></a>                mixed_expression[example_genes[<span class="dv">2</span>], ],</span>
<span id="cb7-34"><a href=""></a>                mixed_expression[example_genes[<span class="dv">3</span>], ])</span>
<span id="cb7-35"><a href=""></a>)</span>
<span id="cb7-36"><a href=""></a></span>
<span id="cb7-37"><a href=""></a><span class="co"># Add cell type proportions to the data</span></span>
<span id="cb7-38"><a href=""></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>n_cell_types) {</span>
<span id="cb7-39"><a href=""></a>  example_data[[<span class="fu">paste0</span>(<span class="st">"Prop"</span>, i)]] <span class="ot">&lt;-</span> <span class="fu">rep</span>(cell_props[, i], <span class="at">times =</span> <span class="fu">length</span>(example_genes))</span>
<span id="cb7-40"><a href=""></a>}</span>
<span id="cb7-41"><a href=""></a></span>
<span id="cb7-42"><a href=""></a><span class="co"># Create plot</span></span>
<span id="cb7-43"><a href=""></a><span class="fu">ggplot</span>(example_data, <span class="fu">aes</span>(<span class="at">x =</span> Sample, <span class="at">y =</span> Expression, <span class="at">color =</span> Gene)) <span class="sc">+</span></span>
<span id="cb7-44"><a href=""></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">3</span>, <span class="at">alpha =</span> <span class="fl">0.7</span>) <span class="sc">+</span></span>
<span id="cb7-45"><a href=""></a>  <span class="fu">geom_line</span>(<span class="at">size =</span> <span class="dv">1</span>, <span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb7-46"><a href=""></a>  <span class="fu">theme_minimal</span>(<span class="at">base_size =</span> <span class="dv">12</span>) <span class="sc">+</span></span>
<span id="cb7-47"><a href=""></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span></span>
<span id="cb7-48"><a href=""></a>  <span class="fu">labs</span>(</span>
<span id="cb7-49"><a href=""></a>    <span class="at">title =</span> <span class="st">"Gene Expression in Mixed Tissue Samples"</span>,</span>
<span id="cb7-50"><a href=""></a>    <span class="at">subtitle =</span> <span class="st">"Expression correlates with cell type proportions"</span>,</span>
<span id="cb7-51"><a href=""></a>    <span class="at">x =</span> <span class="st">"Samples"</span>,</span>
<span id="cb7-52"><a href=""></a>    <span class="at">y =</span> <span class="st">"Expression Level"</span></span>
<span id="cb7-53"><a href=""></a>  ) <span class="sc">+</span></span>
<span id="cb7-54"><a href=""></a>  <span class="fu">scale_color_brewer</span>(<span class="at">palette =</span> <span class="st">"Set1"</span>) <span class="sc">+</span></span>
<span id="cb7-55"><a href=""></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span> Gene, <span class="at">ncol =</span> <span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure>
<p><img data-src="mpgbayes_files/figure-revealjs/unnamed-chunk-7-1.png" width="960"></p>
</figure>
</div>
</div>
</div>
</div></div>
</section>
<section id="mixture-models-mash-approach" class="slide level2 smaller">
<h2>Mixture Models: mash Approach</h2>
<p>The <code>mash</code> method (Urbut, Stephens, et al.) uses mixture models to share information across conditions:</p>
<div class="columns">
<div class="column" style="width:45%;">
<p><strong>Key components</strong>:</p>
<ol type="1">
<li><p><strong>Component distributions</strong>: Different patterns of effects across tissues/conditions</p>
<ul>
<li>Shared effects</li>
<li>Tissue-specific effects</li>
<li>Correlated tissues effects</li>
</ul></li>
<li><p><strong>Mixture weights</strong>: Learned from data, revealing which patterns are common</p></li>
<li><p><strong>Result</strong>: Improved effect estimates by borrowing strength appropriately</p></li>
</ol>
</div><div class="column" style="width:55%;">
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href=""></a><span class="co"># Simulate mash-like data for visualization</span></span>
<span id="cb8-2"><a href=""></a><span class="fu">set.seed</span>(<span class="dv">456</span>)</span>
<span id="cb8-3"><a href=""></a>n_effects <span class="ot">&lt;-</span> <span class="dv">200</span></span>
<span id="cb8-4"><a href=""></a>n_tissues <span class="ot">&lt;-</span> <span class="dv">4</span></span>
<span id="cb8-5"><a href=""></a></span>
<span id="cb8-6"><a href=""></a><span class="co"># Create different effect patterns for visualization</span></span>
<span id="cb8-7"><a href=""></a>effect_patterns <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb8-8"><a href=""></a>  <span class="st">"Shared"</span> <span class="ot">=</span> <span class="fu">rep</span>(<span class="dv">1</span>, n_tissues),</span>
<span id="cb8-9"><a href=""></a>  <span class="st">"Tissue1_specific"</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>),</span>
<span id="cb8-10"><a href=""></a>  <span class="st">"Tissue2_specific"</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="dv">0</span>, <span class="dv">0</span>),</span>
<span id="cb8-11"><a href=""></a>  <span class="st">"Tissues_1_2"</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">0</span>, <span class="dv">0</span>),</span>
<span id="cb8-12"><a href=""></a>  <span class="st">"Tissues_3_4"</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">1</span>, <span class="dv">1</span>)</span>
<span id="cb8-13"><a href=""></a>)</span>
<span id="cb8-14"><a href=""></a></span>
<span id="cb8-15"><a href=""></a><span class="co"># Create matrix to store true effects</span></span>
<span id="cb8-16"><a href=""></a>true_effects <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="dv">0</span>, <span class="at">nrow =</span> n_effects, <span class="at">ncol =</span> n_tissues)</span>
<span id="cb8-17"><a href=""></a><span class="fu">colnames</span>(true_effects) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"Tissue"</span>, <span class="dv">1</span><span class="sc">:</span>n_tissues)</span>
<span id="cb8-18"><a href=""></a></span>
<span id="cb8-19"><a href=""></a><span class="co"># Assign effects based on patterns</span></span>
<span id="cb8-20"><a href=""></a>n_shared <span class="ot">&lt;-</span> <span class="dv">50</span></span>
<span id="cb8-21"><a href=""></a>n_pattern <span class="ot">&lt;-</span> (n_effects <span class="sc">-</span> n_shared) <span class="sc">/</span> (<span class="fu">length</span>(effect_patterns) <span class="sc">-</span> <span class="dv">1</span>)</span>
<span id="cb8-22"><a href=""></a></span>
<span id="cb8-23"><a href=""></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>n_shared) {</span>
<span id="cb8-24"><a href=""></a>  effect_size <span class="ot">&lt;-</span> <span class="fu">rnorm</span>(<span class="dv">1</span>, <span class="dv">0</span>, <span class="fl">0.5</span>)</span>
<span id="cb8-25"><a href=""></a>  true_effects[i, ] <span class="ot">&lt;-</span> effect_patterns[[<span class="dv">1</span>]] <span class="sc">*</span> effect_size</span>
<span id="cb8-26"><a href=""></a>}</span>
<span id="cb8-27"><a href=""></a></span>
<span id="cb8-28"><a href=""></a>current_idx <span class="ot">&lt;-</span> n_shared <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb8-29"><a href=""></a><span class="cf">for</span> (p <span class="cf">in</span> <span class="dv">2</span><span class="sc">:</span><span class="fu">length</span>(effect_patterns)) {</span>
<span id="cb8-30"><a href=""></a>  <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>n_pattern) {</span>
<span id="cb8-31"><a href=""></a>    effect_size <span class="ot">&lt;-</span> <span class="fu">rnorm</span>(<span class="dv">1</span>, <span class="dv">0</span>, <span class="fl">0.5</span>)</span>
<span id="cb8-32"><a href=""></a>    true_effects[current_idx, ] <span class="ot">&lt;-</span> effect_patterns[[p]] <span class="sc">*</span> effect_size</span>
<span id="cb8-33"><a href=""></a>    current_idx <span class="ot">&lt;-</span> current_idx <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb8-34"><a href=""></a>  }</span>
<span id="cb8-35"><a href=""></a>}</span>
<span id="cb8-36"><a href=""></a></span>
<span id="cb8-37"><a href=""></a><span class="co"># Create observed data with noise</span></span>
<span id="cb8-38"><a href=""></a>observed_effects <span class="ot">&lt;-</span> true_effects <span class="sc">+</span> <span class="fu">matrix</span>(<span class="fu">rnorm</span>(n_effects <span class="sc">*</span> n_tissues, <span class="dv">0</span>, <span class="fl">0.2</span>),</span>
<span id="cb8-39"><a href=""></a>                                        <span class="at">nrow =</span> n_effects)</span>
<span id="cb8-40"><a href=""></a></span>
<span id="cb8-41"><a href=""></a><span class="co"># Prepare data for plotting correlation structure</span></span>
<span id="cb8-42"><a href=""></a>correlation_matrix <span class="ot">&lt;-</span> <span class="fu">cor</span>(observed_effects)</span>
<span id="cb8-43"><a href=""></a></span>
<span id="cb8-44"><a href=""></a><span class="co"># Convert correlation matrix to long format for plotting</span></span>
<span id="cb8-45"><a href=""></a>corr_data <span class="ot">&lt;-</span> <span class="fu">expand.grid</span>(<span class="at">Tissue1 =</span> <span class="fu">colnames</span>(correlation_matrix),</span>
<span id="cb8-46"><a href=""></a>                        <span class="at">Tissue2 =</span> <span class="fu">colnames</span>(correlation_matrix))</span>
<span id="cb8-47"><a href=""></a>corr_data<span class="sc">$</span>Correlation <span class="ot">&lt;-</span> <span class="fu">c</span>(correlation_matrix)</span>
<span id="cb8-48"><a href=""></a></span>
<span id="cb8-49"><a href=""></a><span class="co"># Create heatmap</span></span>
<span id="cb8-50"><a href=""></a><span class="fu">ggplot</span>(corr_data, <span class="fu">aes</span>(<span class="at">x =</span> Tissue1, <span class="at">y =</span> Tissue2, <span class="at">fill =</span> Correlation)) <span class="sc">+</span></span>
<span id="cb8-51"><a href=""></a>  <span class="fu">geom_tile</span>() <span class="sc">+</span></span>
<span id="cb8-52"><a href=""></a>  <span class="fu">scale_fill_gradient2</span>(<span class="at">low =</span> <span class="st">"blue"</span>, <span class="at">mid =</span> <span class="st">"white"</span>, <span class="at">high =</span> <span class="st">"red"</span>,</span>
<span id="cb8-53"><a href=""></a>                      <span class="at">midpoint =</span> <span class="dv">0</span>, <span class="at">limits =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>, <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb8-54"><a href=""></a>  <span class="fu">theme_minimal</span>(<span class="at">base_size =</span> <span class="dv">14</span>) <span class="sc">+</span></span>
<span id="cb8-55"><a href=""></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb8-56"><a href=""></a>  <span class="fu">labs</span>(</span>
<span id="cb8-57"><a href=""></a>    <span class="at">title =</span> <span class="st">"Correlation Structure Across Tissues"</span>,</span>
<span id="cb8-58"><a href=""></a>    <span class="at">subtitle =</span> <span class="st">"mash leverages these patterns"</span>,</span>
<span id="cb8-59"><a href=""></a>    <span class="at">x =</span> <span class="st">""</span>,</span>
<span id="cb8-60"><a href=""></a>    <span class="at">y =</span> <span class="st">""</span></span>
<span id="cb8-61"><a href=""></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure>
<p><img data-src="mpgbayes_files/figure-revealjs/unnamed-chunk-8-1.png" width="960"></p>
</figure>
</div>
</div>
</div>
</div></div>
<div class="fragment">
<p><strong>Key insight</strong>: mash automatically learns which patterns of sharing are common and uses this to improve effect estimates</p>
</div>
</section>
<section class="slide level2">

</section></section>
<section>
<section id="bayesian-clinical-adaptive-designs" class="title-slide slide level1 center">
<h1>Bayesian Clinical &amp; Adaptive Designs</h1>

</section>
<section id="adaptive-designs-learn-as-you-go" class="slide level2 smaller">
<h2>Adaptive Designs: Learn As You Go</h2>
<div>
<ul>
<li class="fragment"><strong>Traditional approach</strong>: Fixed design, analyze only at end</li>
<li class="fragment"><strong>Bayesian adaptive approach</strong>: Update and adjust as data accumulates</li>
</ul>
</div>
<div class="fragment">
<p><strong>Key advantages in genomics</strong>: - Focus resources on promising candidates - Stop early when evidence is compelling - Adjust study parameters based on interim results</p>
</div>
</section>
<section id="adaptive-clinical-trial-example" class="slide level2 smaller">
<h2>Adaptive Clinical Trial Example</h2>
<div class="columns">
<div class="column" style="width:50%;">
<p><strong>Traditional vs.&nbsp;Adaptive</strong>:</p>
<ul>
<li><p><strong>Traditional</strong>: Fixed number of patients in treatment &amp; control</p></li>
<li><p><strong>Adaptive</strong>:</p>
<ul>
<li>Start with small groups</li>
<li>Calculate P(treatment &gt; control)</li>
<li>If high probability, assign more patients to treatment</li>
<li>If low probability, stop for futility</li>
</ul></li>
</ul>
</div><div class="column" style="width:50%;">
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href=""></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb9-2"><a href=""></a></span>
<span id="cb9-3"><a href=""></a><span class="co"># Simulate adaptive trial</span></span>
<span id="cb9-4"><a href=""></a><span class="fu">set.seed</span>(<span class="dv">42</span>)</span>
<span id="cb9-5"><a href=""></a></span>
<span id="cb9-6"><a href=""></a><span class="co"># True treatment effect</span></span>
<span id="cb9-7"><a href=""></a>true_effect <span class="ot">&lt;-</span> <span class="fl">0.12</span>  <span class="co"># 12% improvement</span></span>
<span id="cb9-8"><a href=""></a></span>
<span id="cb9-9"><a href=""></a><span class="co"># Trial parameters</span></span>
<span id="cb9-10"><a href=""></a>max_patients <span class="ot">&lt;-</span> <span class="dv">500</span></span>
<span id="cb9-11"><a href=""></a>interims <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="dv">50</span>, max_patients, <span class="at">by =</span> <span class="dv">50</span>)</span>
<span id="cb9-12"><a href=""></a>success_threshold <span class="ot">&lt;-</span> <span class="fl">0.95</span>  <span class="co"># Probability treatment is better</span></span>
<span id="cb9-13"><a href=""></a>futility_threshold <span class="ot">&lt;-</span> <span class="fl">0.1</span>  <span class="co"># Probability treatment is better</span></span>
<span id="cb9-14"><a href=""></a></span>
<span id="cb9-15"><a href=""></a><span class="co"># Prior for control rate: Beta(10, 40) - centered around 20%</span></span>
<span id="cb9-16"><a href=""></a>prior_alpha_control <span class="ot">&lt;-</span> <span class="dv">10</span></span>
<span id="cb9-17"><a href=""></a>prior_beta_control <span class="ot">&lt;-</span> <span class="dv">40</span></span>
<span id="cb9-18"><a href=""></a></span>
<span id="cb9-19"><a href=""></a><span class="co"># Prior for treatment rate: Same as control (skeptical)</span></span>
<span id="cb9-20"><a href=""></a>prior_alpha_treatment <span class="ot">&lt;-</span> <span class="dv">10</span></span>
<span id="cb9-21"><a href=""></a>prior_beta_treatment <span class="ot">&lt;-</span> <span class="dv">40</span></span>
<span id="cb9-22"><a href=""></a></span>
<span id="cb9-23"><a href=""></a><span class="co"># Calculate posterior probability of superiority through trial</span></span>
<span id="cb9-24"><a href=""></a>calc_prob_superior <span class="ot">&lt;-</span> <span class="cf">function</span>(control_a, control_b, treat_a, treat_b, <span class="at">n_samples =</span> <span class="dv">10000</span>) {</span>
<span id="cb9-25"><a href=""></a>  <span class="co"># Monte Carlo estimation</span></span>
<span id="cb9-26"><a href=""></a>  control_samples <span class="ot">&lt;-</span> <span class="fu">rbeta</span>(n_samples, control_a, control_b)</span>
<span id="cb9-27"><a href=""></a>  treat_samples <span class="ot">&lt;-</span> <span class="fu">rbeta</span>(n_samples, treat_a, treat_b)</span>
<span id="cb9-28"><a href=""></a>  <span class="fu">mean</span>(treat_samples <span class="sc">&gt;</span> control_samples)</span>
<span id="cb9-29"><a href=""></a>}</span>
<span id="cb9-30"><a href=""></a></span>
<span id="cb9-31"><a href=""></a><span class="co"># Simulate adaptive trial</span></span>
<span id="cb9-32"><a href=""></a>trial_data <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb9-33"><a href=""></a>  <span class="at">Patients =</span> interims,</span>
<span id="cb9-34"><a href=""></a>  <span class="at">ProbSuperior =</span> <span class="cn">NA</span>,</span>
<span id="cb9-35"><a href=""></a>  <span class="at">ControlRate =</span> <span class="cn">NA</span>,</span>
<span id="cb9-36"><a href=""></a>  <span class="at">TreatmentRate =</span> <span class="cn">NA</span>,</span>
<span id="cb9-37"><a href=""></a>  <span class="at">Decision =</span> <span class="cn">NA</span></span>
<span id="cb9-38"><a href=""></a>)</span>
<span id="cb9-39"><a href=""></a></span>
<span id="cb9-40"><a href=""></a><span class="co"># Simulate control and treatment outcomes</span></span>
<span id="cb9-41"><a href=""></a>control_results <span class="ot">&lt;-</span> <span class="fu">rbinom</span>(max_patients, <span class="dv">1</span>, <span class="fl">0.2</span>)  <span class="co"># 20% success rate</span></span>
<span id="cb9-42"><a href=""></a>treat_results <span class="ot">&lt;-</span> <span class="fu">rbinom</span>(max_patients, <span class="dv">1</span>, <span class="fl">0.2</span> <span class="sc">+</span> true_effect)  <span class="co"># 32% success rate</span></span>
<span id="cb9-43"><a href=""></a></span>
<span id="cb9-44"><a href=""></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(interims)) {</span>
<span id="cb9-45"><a href=""></a>  n <span class="ot">&lt;-</span> interims[i]</span>
<span id="cb9-46"><a href=""></a>  </span>
<span id="cb9-47"><a href=""></a>  <span class="co"># Current data</span></span>
<span id="cb9-48"><a href=""></a>  current_control <span class="ot">&lt;-</span> control_results[<span class="dv">1</span><span class="sc">:</span>n<span class="sc">/</span><span class="dv">2</span>]  <span class="co"># Half in control</span></span>
<span id="cb9-49"><a href=""></a>  current_treat <span class="ot">&lt;-</span> treat_results[<span class="dv">1</span><span class="sc">:</span>n<span class="sc">/</span><span class="dv">2</span>]      <span class="co"># Half in treatment</span></span>
<span id="cb9-50"><a href=""></a>  </span>
<span id="cb9-51"><a href=""></a>  <span class="co"># Update posteriors</span></span>
<span id="cb9-52"><a href=""></a>  post_alpha_control <span class="ot">&lt;-</span> prior_alpha_control <span class="sc">+</span> <span class="fu">sum</span>(current_control)</span>
<span id="cb9-53"><a href=""></a>  post_beta_control <span class="ot">&lt;-</span> prior_beta_control <span class="sc">+</span> <span class="fu">length</span>(current_control) <span class="sc">-</span> <span class="fu">sum</span>(current_control)</span>
<span id="cb9-54"><a href=""></a>  </span>
<span id="cb9-55"><a href=""></a>  post_alpha_treat <span class="ot">&lt;-</span> prior_alpha_treatment <span class="sc">+</span> <span class="fu">sum</span>(current_treat)</span>
<span id="cb9-56"><a href=""></a>  post_beta_treat <span class="ot">&lt;-</span> prior_beta_treatment <span class="sc">+</span> <span class="fu">length</span>(current_treat) <span class="sc">-</span> <span class="fu">sum</span>(current_treat)</span>
<span id="cb9-57"><a href=""></a>  </span>
<span id="cb9-58"><a href=""></a>  <span class="co"># Calculate probability of superiority</span></span>
<span id="cb9-59"><a href=""></a>  prob_superior <span class="ot">&lt;-</span> <span class="fu">calc_prob_superior</span>(</span>
<span id="cb9-60"><a href=""></a>    post_alpha_control, post_beta_control,</span>
<span id="cb9-61"><a href=""></a>    post_alpha_treat, post_beta_treat</span>
<span id="cb9-62"><a href=""></a>  )</span>
<span id="cb9-63"><a href=""></a>  </span>
<span id="cb9-64"><a href=""></a>  <span class="co"># Record results</span></span>
<span id="cb9-65"><a href=""></a>  trial_data<span class="sc">$</span>ProbSuperior[i] <span class="ot">&lt;-</span> prob_superior</span>
<span id="cb9-66"><a href=""></a>  trial_data<span class="sc">$</span>ControlRate[i] <span class="ot">&lt;-</span> post_alpha_control <span class="sc">/</span> (post_alpha_control <span class="sc">+</span> post_beta_control)</span>
<span id="cb9-67"><a href=""></a>  trial_data<span class="sc">$</span>TreatmentRate[i] <span class="ot">&lt;-</span> post_alpha_treat <span class="sc">/</span> (post_alpha_treat <span class="sc">+</span> post_beta_treat)</span>
<span id="cb9-68"><a href=""></a>  </span>
<span id="cb9-69"><a href=""></a>  <span class="co"># Make decision</span></span>
<span id="cb9-70"><a href=""></a>  <span class="cf">if</span> (prob_superior <span class="sc">&gt;=</span> success_threshold) {</span>
<span id="cb9-71"><a href=""></a>    trial_data<span class="sc">$</span>Decision[i] <span class="ot">&lt;-</span> <span class="st">"Success"</span></span>
<span id="cb9-72"><a href=""></a>    <span class="cf">if</span> (i <span class="sc">&lt;</span> <span class="fu">length</span>(interims)) {</span>
<span id="cb9-73"><a href=""></a>      trial_data<span class="sc">$</span>Decision[(i<span class="sc">+</span><span class="dv">1</span>)<span class="sc">:</span><span class="fu">length</span>(interims)] <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb9-74"><a href=""></a>    }</span>
<span id="cb9-75"><a href=""></a>    <span class="cf">break</span></span>
<span id="cb9-76"><a href=""></a>  } <span class="cf">else</span> <span class="cf">if</span> (prob_superior <span class="sc">&lt;=</span> futility_threshold) {</span>
<span id="cb9-77"><a href=""></a>    trial_data<span class="sc">$</span>Decision[i] <span class="ot">&lt;-</span> <span class="st">"Futility"</span></span>
<span id="cb9-78"><a href=""></a>    <span class="cf">if</span> (i <span class="sc">&lt;</span> <span class="fu">length</span>(interims)) {</span>
<span id="cb9-79"><a href=""></a>      trial_data<span class="sc">$</span>Decision[(i<span class="sc">+</span><span class="dv">1</span>)<span class="sc">:</span><span class="fu">length</span>(interims)] <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb9-80"><a href=""></a>    }</span>
<span id="cb9-81"><a href=""></a>    <span class="cf">break</span></span>
<span id="cb9-82"><a href=""></a>  } <span class="cf">else</span> {</span>
<span id="cb9-83"><a href=""></a>    trial_data<span class="sc">$</span>Decision[i] <span class="ot">&lt;-</span> <span class="st">"Continue"</span></span>
<span id="cb9-84"><a href=""></a>  }</span>
<span id="cb9-85"><a href=""></a>}</span>
<span id="cb9-86"><a href=""></a></span>
<span id="cb9-87"><a href=""></a><span class="co"># Create plot</span></span>
<span id="cb9-88"><a href=""></a><span class="fu">ggplot</span>(trial_data, <span class="fu">aes</span>(<span class="at">x =</span> Patients, <span class="at">y =</span> ProbSuperior)) <span class="sc">+</span></span>
<span id="cb9-89"><a href=""></a>  <span class="fu">geom_line</span>(<span class="at">size =</span> <span class="fl">1.2</span>, <span class="at">color =</span> <span class="st">"blue"</span>) <span class="sc">+</span></span>
<span id="cb9-90"><a href=""></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">3</span>, <span class="fu">aes</span>(<span class="at">color =</span> Decision)) <span class="sc">+</span></span>
<span id="cb9-91"><a href=""></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> success_threshold, <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">color =</span> <span class="st">"green"</span>) <span class="sc">+</span></span>
<span id="cb9-92"><a href=""></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> futility_threshold, <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">color =</span> <span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb9-93"><a href=""></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> max_patients <span class="sc">*</span> <span class="fl">0.8</span>, <span class="at">y =</span> success_threshold <span class="sc">+</span> <span class="fl">0.02</span>, </span>
<span id="cb9-94"><a href=""></a>           <span class="at">label =</span> <span class="st">"Success threshold"</span>, <span class="at">color =</span> <span class="st">"green"</span>) <span class="sc">+</span></span>
<span id="cb9-95"><a href=""></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> max_patients <span class="sc">*</span> <span class="fl">0.8</span>, <span class="at">y =</span> futility_threshold <span class="sc">-</span> <span class="fl">0.02</span>, </span>
<span id="cb9-96"><a href=""></a>           <span class="at">label =</span> <span class="st">"Futility threshold"</span>, <span class="at">color =</span> <span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb9-97"><a href=""></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"Continue"</span> <span class="ot">=</span> <span class="st">"orange"</span>, </span>
<span id="cb9-98"><a href=""></a>                               <span class="st">"Success"</span> <span class="ot">=</span> <span class="st">"green"</span>, </span>
<span id="cb9-99"><a href=""></a>                               <span class="st">"Futility"</span> <span class="ot">=</span> <span class="st">"red"</span>)) <span class="sc">+</span></span>
<span id="cb9-100"><a href=""></a>  <span class="fu">theme_minimal</span>(<span class="at">base_size =</span> <span class="dv">12</span>) <span class="sc">+</span></span>
<span id="cb9-101"><a href=""></a>  <span class="fu">labs</span>(</span>
<span id="cb9-102"><a href=""></a>    <span class="at">title =</span> <span class="st">"Bayesian Adaptive Clinical Trial"</span>,</span>
<span id="cb9-103"><a href=""></a>    <span class="at">subtitle =</span> <span class="st">"Posterior probability treatment is superior to control"</span>,</span>
<span id="cb9-104"><a href=""></a>    <span class="at">x =</span> <span class="st">"Number of Patients Enrolled"</span>,</span>
<span id="cb9-105"><a href=""></a>    <span class="at">y =</span> <span class="st">"P(Treatment &gt; Control)"</span></span>
<span id="cb9-106"><a href=""></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure>
<p><img data-src="mpgbayes_files/figure-revealjs/unnamed-chunk-9-1.png" width="960"></p>
</figure>
</div>
</div>
</div>
</div></div>
</section>
<section id="population-genomics-applications" class="slide level2 smaller">
<h2>Population Genomics Applications</h2>
<p><strong>Adaptive sequencing designs</strong>:</p>
<div>
<ol type="1">
<li class="fragment"><strong>Candidate gene sequencing</strong>:
<ul>
<li class="fragment">Sequence initial set of genes in a subset of samples</li>
<li class="fragment">Calculate posterior probability of association for each gene</li>
<li class="fragment">Focus remaining resources on promising genes</li>
<li class="fragment">Add related genes based on pathway analysis</li>
</ul></li>
<li class="fragment"><strong>Exome-to-genome expansion</strong>:
<ul>
<li class="fragment">Start with exome sequencing in all samples</li>
<li class="fragment">For variants with high posterior probability, sequence surrounding regions</li>
<li class="fragment">Iteratively expand to full genome in regions of interest</li>
</ul></li>
<li class="fragment"><strong>Sample size re-estimation</strong>:
<ul>
<li class="fragment">Update power calculations based on observed effect sizes</li>
<li class="fragment">Increase sample size only where needed</li>
</ul></li>
</ol>
</div>
</section>
<section id="using-informative-priors-from-previous-studies" class="slide level2 smaller">
<h2>Using Informative Priors from Previous Studies</h2>
<div class="columns">
<div class="column" style="width:50%;">
<p><strong>Combining data across studies</strong>:</p>
<ol type="1">
<li><strong>Previous study as prior</strong>:
<ul>
<li>Convert previous results to Beta/Normal prior</li>
<li>Current study provides likelihood</li>
<li>Posterior combines both sources</li>
</ul></li>
<li><strong>Meta-analysis alternative</strong>:
<ul>
<li>Each study analyzed separately</li>
<li>Results combined at the end</li>
<li>Less efficient use of information</li>
</ul></li>
</ol>
</div><div class="column" style="width:50%;">
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href=""></a><span class="co"># Demonstrate using previous study as prior</span></span>
<span id="cb10-2"><a href=""></a><span class="fu">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb10-3"><a href=""></a></span>
<span id="cb10-4"><a href=""></a><span class="co"># Previous study data (small study)</span></span>
<span id="cb10-5"><a href=""></a>prev_n <span class="ot">&lt;-</span> <span class="dv">50</span></span>
<span id="cb10-6"><a href=""></a>prev_effect <span class="ot">&lt;-</span> <span class="fl">0.15</span></span>
<span id="cb10-7"><a href=""></a>prev_se <span class="ot">&lt;-</span> <span class="fl">0.1</span></span>
<span id="cb10-8"><a href=""></a></span>
<span id="cb10-9"><a href=""></a><span class="co"># Current study (larger)</span></span>
<span id="cb10-10"><a href=""></a>current_n <span class="ot">&lt;-</span> <span class="dv">200</span></span>
<span id="cb10-11"><a href=""></a>true_effect <span class="ot">&lt;-</span> <span class="fl">0.12</span></span>
<span id="cb10-12"><a href=""></a>current_se <span class="ot">&lt;-</span> <span class="fl">0.05</span></span>
<span id="cb10-13"><a href=""></a></span>
<span id="cb10-14"><a href=""></a><span class="co"># Simulate observed effects</span></span>
<span id="cb10-15"><a href=""></a>prev_observed <span class="ot">&lt;-</span> <span class="fu">rnorm</span>(<span class="dv">1</span>, prev_effect, prev_se)</span>
<span id="cb10-16"><a href=""></a>current_observed <span class="ot">&lt;-</span> <span class="fu">rnorm</span>(<span class="dv">1</span>, true_effect, current_se)</span>
<span id="cb10-17"><a href=""></a></span>
<span id="cb10-18"><a href=""></a><span class="co"># Approaches to combine:</span></span>
<span id="cb10-19"><a href=""></a><span class="co"># 1. Flat prior + current data only</span></span>
<span id="cb10-20"><a href=""></a>flat_prior_mean <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb10-21"><a href=""></a>flat_prior_sd <span class="ot">&lt;-</span> <span class="dv">10</span>  <span class="co"># Very diffuse</span></span>
<span id="cb10-22"><a href=""></a></span>
<span id="cb10-23"><a href=""></a><span class="co"># 2. Using previous study as prior</span></span>
<span id="cb10-24"><a href=""></a>informative_prior_mean <span class="ot">&lt;-</span> prev_observed</span>
<span id="cb10-25"><a href=""></a>informative_prior_sd <span class="ot">&lt;-</span> prev_se</span>
<span id="cb10-26"><a href=""></a></span>
<span id="cb10-27"><a href=""></a><span class="co"># Calculate posteriors</span></span>
<span id="cb10-28"><a href=""></a><span class="co"># Flat prior</span></span>
<span id="cb10-29"><a href=""></a>flat_posterior_precision <span class="ot">&lt;-</span> <span class="dv">1</span><span class="sc">/</span>flat_prior_sd<span class="sc">^</span><span class="dv">2</span> <span class="sc">+</span> <span class="dv">1</span><span class="sc">/</span>current_se<span class="sc">^</span><span class="dv">2</span></span>
<span id="cb10-30"><a href=""></a>flat_posterior_sd <span class="ot">&lt;-</span> <span class="fu">sqrt</span>(<span class="dv">1</span><span class="sc">/</span>flat_posterior_precision)</span>
<span id="cb10-31"><a href=""></a>flat_posterior_mean <span class="ot">&lt;-</span> (flat_prior_mean<span class="sc">/</span>flat_prior_sd<span class="sc">^</span><span class="dv">2</span> <span class="sc">+</span> </span>
<span id="cb10-32"><a href=""></a>                       current_observed<span class="sc">/</span>current_se<span class="sc">^</span><span class="dv">2</span>) <span class="sc">/</span> flat_posterior_precision</span>
<span id="cb10-33"><a href=""></a></span>
<span id="cb10-34"><a href=""></a><span class="co"># Informative prior</span></span>
<span id="cb10-35"><a href=""></a>inf_posterior_precision <span class="ot">&lt;-</span> <span class="dv">1</span><span class="sc">/</span>informative_prior_sd<span class="sc">^</span><span class="dv">2</span> <span class="sc">+</span> <span class="dv">1</span><span class="sc">/</span>current_se<span class="sc">^</span><span class="dv">2</span></span>
<span id="cb10-36"><a href=""></a>inf_posterior_sd <span class="ot">&lt;-</span> <span class="fu">sqrt</span>(<span class="dv">1</span><span class="sc">/</span>inf_posterior_precision)</span>
<span id="cb10-37"><a href=""></a>inf_posterior_mean <span class="ot">&lt;-</span> (informative_prior_mean<span class="sc">/</span>informative_prior_sd<span class="sc">^</span><span class="dv">2</span> <span class="sc">+</span> </span>
<span id="cb10-38"><a href=""></a>                     current_observed<span class="sc">/</span>current_se<span class="sc">^</span><span class="dv">2</span>) <span class="sc">/</span> inf_posterior_precision</span>
<span id="cb10-39"><a href=""></a></span>
<span id="cb10-40"><a href=""></a><span class="co"># 3. Fixed meta-analysis (inverse variance weighted)</span></span>
<span id="cb10-41"><a href=""></a>meta_weight_prev <span class="ot">&lt;-</span> <span class="dv">1</span><span class="sc">/</span>prev_se<span class="sc">^</span><span class="dv">2</span></span>
<span id="cb10-42"><a href=""></a>meta_weight_current <span class="ot">&lt;-</span> <span class="dv">1</span><span class="sc">/</span>current_se<span class="sc">^</span><span class="dv">2</span></span>
<span id="cb10-43"><a href=""></a>meta_mean <span class="ot">&lt;-</span> (meta_weight_prev <span class="sc">*</span> prev_observed <span class="sc">+</span> </span>
<span id="cb10-44"><a href=""></a>             meta_weight_current <span class="sc">*</span> current_observed) <span class="sc">/</span> </span>
<span id="cb10-45"><a href=""></a>            (meta_weight_prev <span class="sc">+</span> meta_weight_current)</span>
<span id="cb10-46"><a href=""></a>meta_se <span class="ot">&lt;-</span> <span class="fu">sqrt</span>(<span class="dv">1</span><span class="sc">/</span>(meta_weight_prev <span class="sc">+</span> meta_weight_current))</span>
<span id="cb10-47"><a href=""></a></span>
<span id="cb10-48"><a href=""></a><span class="co"># Plot the results</span></span>
<span id="cb10-49"><a href=""></a>x_range <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="fl">0.3</span>, <span class="at">length.out =</span> <span class="dv">1000</span>)</span>
<span id="cb10-50"><a href=""></a>prior_density <span class="ot">&lt;-</span> <span class="fu">dnorm</span>(x_range, informative_prior_mean, informative_prior_sd)</span>
<span id="cb10-51"><a href=""></a>likelihood_density <span class="ot">&lt;-</span> <span class="fu">dnorm</span>(x_range, current_observed, current_se)</span>
<span id="cb10-52"><a href=""></a>flat_posterior_density <span class="ot">&lt;-</span> <span class="fu">dnorm</span>(x_range, flat_posterior_mean, flat_posterior_sd)</span>
<span id="cb10-53"><a href=""></a>inf_posterior_density <span class="ot">&lt;-</span> <span class="fu">dnorm</span>(x_range, inf_posterior_mean, inf_posterior_sd)</span>
<span id="cb10-54"><a href=""></a>meta_density <span class="ot">&lt;-</span> <span class="fu">dnorm</span>(x_range, meta_mean, meta_se)</span>
<span id="cb10-55"><a href=""></a></span>
<span id="cb10-56"><a href=""></a>plot_data <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb10-57"><a href=""></a>  <span class="at">x =</span> <span class="fu">rep</span>(x_range, <span class="dv">5</span>),</span>
<span id="cb10-58"><a href=""></a>  <span class="at">y =</span> <span class="fu">c</span>(prior_density, likelihood_density, flat_posterior_density, </span>
<span id="cb10-59"><a href=""></a>       inf_posterior_density, meta_density),</span>
<span id="cb10-60"><a href=""></a>  <span class="at">Distribution =</span> <span class="fu">factor</span>(<span class="fu">rep</span>(<span class="fu">c</span>(<span class="st">"Previous Study (Prior)"</span>, <span class="st">"Current Study (Likelihood)"</span>, </span>
<span id="cb10-61"><a href=""></a>                           <span class="st">"Posterior (Flat Prior)"</span>, <span class="st">"Posterior (Informative)"</span>, </span>
<span id="cb10-62"><a href=""></a>                           <span class="st">"Fixed Meta-analysis"</span>), </span>
<span id="cb10-63"><a href=""></a>                       <span class="at">each =</span> <span class="fu">length</span>(x_range)))</span>
<span id="cb10-64"><a href=""></a>)</span>
<span id="cb10-65"><a href=""></a></span>
<span id="cb10-66"><a href=""></a><span class="co"># Plot</span></span>
<span id="cb10-67"><a href=""></a><span class="fu">ggplot</span>(plot_data, <span class="fu">aes</span>(<span class="at">x =</span> x, <span class="at">y =</span> y, <span class="at">color =</span> Distribution)) <span class="sc">+</span></span>
<span id="cb10-68"><a href=""></a>  <span class="fu">geom_line</span>(<span class="at">size =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb10-69"><a href=""></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> true_effect, <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">color =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb10-70"><a href=""></a>  <span class="fu">theme_minimal</span>(<span class="at">base_size =</span> <span class="dv">12</span>) <span class="sc">+</span></span>
<span id="cb10-71"><a href=""></a>  <span class="fu">scale_color_brewer</span>(<span class="at">palette =</span> <span class="st">"Set1"</span>) <span class="sc">+</span></span>
<span id="cb10-72"><a href=""></a>  <span class="fu">labs</span>(</span>
<span id="cb10-73"><a href=""></a>    <span class="at">title =</span> <span class="st">"Combining Information Across Studies"</span>,</span>
<span id="cb10-74"><a href=""></a>    <span class="at">subtitle =</span> <span class="fu">paste</span>(<span class="st">"True effect ="</span>, true_effect),</span>
<span id="cb10-75"><a href=""></a>    <span class="at">x =</span> <span class="st">"Effect Size"</span>,</span>
<span id="cb10-76"><a href=""></a>    <span class="at">y =</span> <span class="st">"Density"</span></span>
<span id="cb10-77"><a href=""></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure>
<p><img data-src="mpgbayes_files/figure-revealjs/unnamed-chunk-10-1.png" width="960"></p>
</figure>
</div>
</div>
</div>
</div></div>
</section>
<section id="bayesian-fdr-control-in-gwas" class="slide level2 smaller">
<h2>Bayesian FDR Control in GWAS</h2>
<div class="columns">
<div class="column" style="width:40%;">
<p><strong>Traditional FDR</strong>: - Sort p-values - Apply B-H procedure - Control expected proportion of false discoveries</p>
<p><strong>Bayesian FDR</strong>: - Calculate posterior probability of association for each variant - Sort by posterior probability - Include variants until expected FDR reaches threshold</p>
<p><strong>Advantage</strong>: Directly incorporates prior information like: - Functional annotations - MAF - LD structure</p>
</div><div class="column" style="width:60%;">
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href=""></a><span class="co"># Simulate GWAS results</span></span>
<span id="cb11-2"><a href=""></a><span class="fu">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb11-3"><a href=""></a>n_variants <span class="ot">&lt;-</span> <span class="dv">5000</span></span>
<span id="cb11-4"><a href=""></a>n_causal <span class="ot">&lt;-</span> <span class="dv">100</span></span>
<span id="cb11-5"><a href=""></a></span>
<span id="cb11-6"><a href=""></a><span class="co"># Generate true status (1 = causal, 0 = null)</span></span>
<span id="cb11-7"><a href=""></a>true_status <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="dv">1</span>, n_causal), <span class="fu">rep</span>(<span class="dv">0</span>, n_variants <span class="sc">-</span> n_causal))</span>
<span id="cb11-8"><a href=""></a></span>
<span id="cb11-9"><a href=""></a><span class="co"># Generate z-scores</span></span>
<span id="cb11-10"><a href=""></a>null_z <span class="ot">&lt;-</span> <span class="fu">rnorm</span>(n_variants <span class="sc">-</span> n_causal, <span class="dv">0</span>, <span class="dv">1</span>)</span>
<span id="cb11-11"><a href=""></a>causal_z <span class="ot">&lt;-</span> <span class="fu">rnorm</span>(n_causal, <span class="dv">0</span>, <span class="dv">1</span>) <span class="sc">+</span> <span class="fu">runif</span>(n_causal, <span class="dv">2</span>, <span class="dv">5</span>)  <span class="co"># Add effect</span></span>
<span id="cb11-12"><a href=""></a>z_scores <span class="ot">&lt;-</span> <span class="fu">c</span>(causal_z, null_z)</span>
<span id="cb11-13"><a href=""></a></span>
<span id="cb11-14"><a href=""></a><span class="co"># Calculate p-values</span></span>
<span id="cb11-15"><a href=""></a>p_values <span class="ot">&lt;-</span> <span class="dv">2</span> <span class="sc">*</span> <span class="fu">pnorm</span>(<span class="sc">-</span><span class="fu">abs</span>(z_scores))</span>
<span id="cb11-16"><a href=""></a></span>
<span id="cb11-17"><a href=""></a><span class="co"># Calculate Bayes factors</span></span>
<span id="cb11-18"><a href=""></a>bf <span class="ot">&lt;-</span> <span class="fu">exp</span>(z_scores<span class="sc">^</span><span class="dv">2</span><span class="sc">/</span><span class="dv">2</span>)</span>
<span id="cb11-19"><a href=""></a></span>
<span id="cb11-20"><a href=""></a><span class="co"># Calculate posterior with uniform prior</span></span>
<span id="cb11-21"><a href=""></a>prior_prob <span class="ot">&lt;-</span> n_causal <span class="sc">/</span> n_variants</span>
<span id="cb11-22"><a href=""></a>posterior_prob <span class="ot">&lt;-</span> (prior_prob <span class="sc">*</span> bf) <span class="sc">/</span> (prior_prob <span class="sc">*</span> bf <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">-</span> prior_prob))</span>
<span id="cb11-23"><a href=""></a></span>
<span id="cb11-24"><a href=""></a><span class="co"># Apply Benjamini-Hochberg</span></span>
<span id="cb11-25"><a href=""></a>alpha <span class="ot">&lt;-</span> <span class="fl">0.1</span></span>
<span id="cb11-26"><a href=""></a>sorted_p <span class="ot">&lt;-</span> <span class="fu">sort</span>(p_values)</span>
<span id="cb11-27"><a href=""></a>ranks <span class="ot">&lt;-</span> <span class="dv">1</span><span class="sc">:</span>n_variants</span>
<span id="cb11-28"><a href=""></a>bh_threshold <span class="ot">&lt;-</span> <span class="fu">max</span>(sorted_p[sorted_p <span class="sc">&lt;=</span> (ranks <span class="sc">/</span> n_variants) <span class="sc">*</span> alpha])</span>
<span id="cb11-29"><a href=""></a>bh_significant <span class="ot">&lt;-</span> p_values <span class="sc">&lt;=</span> bh_threshold</span>
<span id="cb11-30"><a href=""></a></span>
<span id="cb11-31"><a href=""></a><span class="co"># Apply Bayesian FDR control</span></span>
<span id="cb11-32"><a href=""></a>sorted_probs <span class="ot">&lt;-</span> <span class="fu">sort</span>(<span class="dv">1</span> <span class="sc">-</span> posterior_prob, <span class="at">decreasing =</span> <span class="cn">FALSE</span>)</span>
<span id="cb11-33"><a href=""></a>cumulative_fdr <span class="ot">&lt;-</span> <span class="fu">cumsum</span>(sorted_probs) <span class="sc">/</span> (<span class="dv">1</span><span class="sc">:</span>n_variants)</span>
<span id="cb11-34"><a href=""></a>bayes_threshold <span class="ot">&lt;-</span> <span class="fu">min</span>(posterior_prob[posterior_prob <span class="sc">&gt;=</span> </span>
<span id="cb11-35"><a href=""></a>                                     <span class="fu">min</span>(posterior_prob[cumulative_fdr <span class="sc">&lt;=</span> alpha])])</span>
<span id="cb11-36"><a href=""></a>bayes_significant <span class="ot">&lt;-</span> posterior_prob <span class="sc">&gt;=</span> bayes_threshold</span>
<span id="cb11-37"><a href=""></a></span>
<span id="cb11-38"><a href=""></a><span class="co"># Create data for comparison</span></span>
<span id="cb11-39"><a href=""></a>results <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb11-40"><a href=""></a>  <span class="at">z_score =</span> z_scores,</span>
<span id="cb11-41"><a href=""></a>  <span class="at">p_value =</span> p_values,</span>
<span id="cb11-42"><a href=""></a>  <span class="at">posterior =</span> posterior_prob,</span>
<span id="cb11-43"><a href=""></a>  <span class="at">true_status =</span> <span class="fu">as.factor</span>(true_status),</span>
<span id="cb11-44"><a href=""></a>  <span class="at">bh_sig =</span> bh_significant,</span>
<span id="cb11-45"><a href=""></a>  <span class="at">bayes_sig =</span> bayes_significant</span>
<span id="cb11-46"><a href=""></a>)</span>
<span id="cb11-47"><a href=""></a></span>
<span id="cb11-48"><a href=""></a><span class="co"># Calculate FDR and power</span></span>
<span id="cb11-49"><a href=""></a>bh_fdr <span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="sc">!</span>true_status <span class="sc">&amp;</span> bh_significant) <span class="sc">/</span> <span class="fu">max</span>(<span class="dv">1</span>, <span class="fu">sum</span>(bh_significant))</span>
<span id="cb11-50"><a href=""></a>bayes_fdr <span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="sc">!</span>true_status <span class="sc">&amp;</span> bayes_significant) <span class="sc">/</span> <span class="fu">max</span>(<span class="dv">1</span>, <span class="fu">sum</span>(bayes_significant))</span>
<span id="cb11-51"><a href=""></a>bh_power <span class="ot">&lt;-</span> <span class="fu">sum</span>(true_status <span class="sc">&amp;</span> bh_significant) <span class="sc">/</span> <span class="fu">sum</span>(true_status)</span>
<span id="cb11-52"><a href=""></a>bayes_power <span class="ot">&lt;-</span> <span class="fu">sum</span>(true_status <span class="sc">&amp;</span> bayes_significant) <span class="sc">/</span> <span class="fu">sum</span>(true_status)</span>
<span id="cb11-53"><a href=""></a></span>
<span id="cb11-54"><a href=""></a><span class="co"># Plot comparison</span></span>
<span id="cb11-55"><a href=""></a><span class="fu">ggplot</span>(results, <span class="fu">aes</span>(<span class="at">x =</span> <span class="sc">-</span><span class="fu">log10</span>(p_value), <span class="at">y =</span> posterior, <span class="at">color =</span> true_status)) <span class="sc">+</span></span>
<span id="cb11-56"><a href=""></a>  <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.6</span>) <span class="sc">+</span></span>
<span id="cb11-57"><a href=""></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> bayes_threshold, <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">color =</span> <span class="st">"blue"</span>) <span class="sc">+</span></span>
<span id="cb11-58"><a href=""></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(bh_threshold), <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">color =</span> <span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb11-59"><a href=""></a>  <span class="fu">theme_minimal</span>(<span class="at">base_size =</span> <span class="dv">12</span>) <span class="sc">+</span></span>
<span id="cb11-60"><a href=""></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"0"</span> <span class="ot">=</span> <span class="st">"gray"</span>, <span class="st">"1"</span> <span class="ot">=</span> <span class="st">"orange"</span>), </span>
<span id="cb11-61"><a href=""></a>                    <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"0"</span> <span class="ot">=</span> <span class="st">"Null"</span>, <span class="st">"1"</span> <span class="ot">=</span> <span class="st">"Causal"</span>)) <span class="sc">+</span></span>
<span id="cb11-62"><a href=""></a>  <span class="fu">labs</span>(</span>
<span id="cb11-63"><a href=""></a>    <span class="at">title =</span> <span class="st">"Bayesian vs. Traditional FDR Control"</span>,</span>
<span id="cb11-64"><a href=""></a>    <span class="at">subtitle =</span> <span class="fu">paste0</span>(<span class="st">"BH FDR = "</span>, <span class="fu">round</span>(bh_fdr, <span class="dv">3</span>), <span class="st">", Power = "</span>, <span class="fu">round</span>(bh_power, <span class="dv">3</span>),</span>
<span id="cb11-65"><a href=""></a>                    <span class="st">"</span><span class="sc">\n</span><span class="st">Bayes FDR = "</span>, <span class="fu">round</span>(bayes_fdr, <span class="dv">3</span>), <span class="st">", Power = "</span>, <span class="fu">round</span>(bayes_power, <span class="dv">3</span>)),</span>
<span id="cb11-66"><a href=""></a>    <span class="at">x =</span> <span class="st">"-log10(p-value)"</span>,</span>
<span id="cb11-67"><a href=""></a>    <span class="at">y =</span> <span class="st">"Posterior Probability"</span>,</span>
<span id="cb11-68"><a href=""></a>    <span class="at">color =</span> <span class="st">"True Status"</span></span>
<span id="cb11-69"><a href=""></a>  ) <span class="sc">+</span></span>
<span id="cb11-70"><a href=""></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="dv">2</span>, <span class="at">y =</span> bayes_threshold <span class="sc">+</span> <span class="fl">0.05</span>, </span>
<span id="cb11-71"><a href=""></a>           <span class="at">label =</span> <span class="st">"Bayesian threshold"</span>, <span class="at">color =</span> <span class="st">"blue"</span>) <span class="sc">+</span></span>
<span id="cb11-72"><a href=""></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="sc">-</span><span class="fu">log10</span>(bh_threshold) <span class="sc">+</span> <span class="fl">0.5</span>, <span class="at">y =</span> <span class="fl">0.1</span>, </span>
<span id="cb11-73"><a href=""></a>           <span class="at">label =</span> <span class="st">"BH threshold"</span>, <span class="at">color =</span> <span class="st">"red"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure>
<p><img data-src="mpgbayes_files/figure-revealjs/unnamed-chunk-11-1.png" width="960"></p>
</figure>
</div>
</div>
</div>
</div></div>
</section>
<section id="recommended-resources" class="slide level2 smaller">
<h2>Recommended Resources</h2>
<div class="columns">
<div class="column" style="width:50%;">
<p><strong>Books &amp; Articles</strong>: - “Statistical Rethinking” by McElreath - “Bayesian Data Analysis” by Gelman et al.&nbsp;- Five Minute Statistics: http://stephens999.github.io/fiveMinuteStats/ - Urbut et al.&nbsp;“Flexible statistical methods for estimating and testing effects in genomic studies with multiple conditions” (2019)</p>
</div><div class="column" style="width:50%;">
<p><strong>Software &amp; Packages</strong>: - Stan/rstan: Comprehensive Bayesian modeling - STRUCTURE: Population inference - mash: Multi-condition effect sharing - ashr: Adaptive shrinkage - INLA: Fast Bayesian inference</p>
</div></div>
<div class="fragment">
<p><strong>Key papers</strong>: - Stephens &amp; Balding (2009) “Bayesian statistical methods for genetic association studies” - Guan &amp; Stephens (2008) “Practical Issues in Imputation-Based Association Mapping” - Servin &amp; Stephens (2007) “Imputation-based analysis of association studies”</p>
</div>
</section>
<section id="summary-why-bayesian-methods-matter" class="slide level2 smaller">
<h2>Summary: Why Bayesian Methods Matter</h2>
<div>
<ol type="1">
<li class="fragment"><strong>Direct probability statements</strong> about hypotheses
<ul>
<li class="fragment">“There is a 95% probability this variant affects the trait”</li>
</ul></li>
<li class="fragment"><strong>Natural incorporation of prior knowledge</strong>
<ul>
<li class="fragment">Functional annotations, conservation, previous studies</li>
</ul></li>
<li class="fragment"><strong>Honest handling of uncertainty</strong>
<ul>
<li class="fragment">Credible intervals with correct interpretation</li>
</ul></li>
<li class="fragment"><strong>More efficient use of data</strong>
<ul>
<li class="fragment">Borrow strength across contexts (genes, tissues, populations)</li>
</ul></li>
<li class="fragment"><strong>Adaptive learning framework</strong>
<ul>
<li class="fragment">Update beliefs as data accumulates</li>
</ul></li>
</ol>
</div>
</section>
<section id="thank-you" class="slide level2">
<h2>Thank You!</h2>
<p><strong>Questions?</strong></p>
</section>
<section id="appendix-technical-details" class="slide level2">
<h2>Appendix: Technical Details</h2>
<div class="quarto-auto-generated-content">
<div class="footer footer-default">

</div>
</div>
</section></section>
    </div>
  </div>

  <script>window.backupDefine = window.define; window.define = undefined;</script>
  <script src="mpgbayes_files/libs/revealjs/dist/reveal.js"></script>
  <!-- reveal.js plugins -->
  <script src="mpgbayes_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.js"></script>
  <script src="mpgbayes_files/libs/revealjs/plugin/pdf-export/pdfexport.js"></script>
  <script src="mpgbayes_files/libs/revealjs/plugin/reveal-menu/menu.js"></script>
  <script src="mpgbayes_files/libs/revealjs/plugin/reveal-menu/quarto-menu.js"></script>
  <script src="mpgbayes_files/libs/revealjs/plugin/quarto-support/support.js"></script>
  

  <script src="mpgbayes_files/libs/revealjs/plugin/notes/notes.js"></script>
  <script src="mpgbayes_files/libs/revealjs/plugin/search/search.js"></script>
  <script src="mpgbayes_files/libs/revealjs/plugin/zoom/zoom.js"></script>
  <script src="mpgbayes_files/libs/revealjs/plugin/math/math.js"></script>
  <script>window.define = window.backupDefine; window.backupDefine = undefined;</script>

  <script>

      // Full list of configuration options available at:
      // https://revealjs.com/config/
      Reveal.initialize({
'controlsAuto': true,
'previewLinksAuto': false,
'pdfSeparateFragments': false,
'autoAnimateEasing': "ease",
'autoAnimateDuration': 1,
'autoAnimateUnmatched': true,
'menu': {"side":"left","useTextContentForMissingTitles":true,"markers":false,"loadIcons":false,"custom":[{"title":"Tools","icon":"<i class=\"fas fa-gear\"></i>","content":"<ul class=\"slide-menu-items\">\n<li class=\"slide-tool-item active\" data-item=\"0\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.fullscreen(event)\"><kbd>f</kbd> Fullscreen</a></li>\n<li class=\"slide-tool-item\" data-item=\"1\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.speakerMode(event)\"><kbd>s</kbd> Speaker View</a></li>\n<li class=\"slide-tool-item\" data-item=\"2\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.overview(event)\"><kbd>o</kbd> Slide Overview</a></li>\n<li class=\"slide-tool-item\" data-item=\"3\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.togglePdfExport(event)\"><kbd>e</kbd> PDF Export Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"4\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.keyboardHelp(event)\"><kbd>?</kbd> Keyboard Help</a></li>\n</ul>"}],"openButton":true},
'smaller': false,
 
        // Display controls in the bottom right corner
        controls: false,

        // Help the user learn the controls by providing hints, for example by
        // bouncing the down arrow when they first encounter a vertical slide
        controlsTutorial: false,

        // Determines where controls appear, "edges" or "bottom-right"
        controlsLayout: 'edges',

        // Visibility rule for backwards navigation arrows; "faded", "hidden"
        // or "visible"
        controlsBackArrows: 'faded',

        // Display a presentation progress bar
        progress: true,

        // Display the page number of the current slide
        slideNumber: 'c/t',

        // 'all', 'print', or 'speaker'
        showSlideNumber: 'all',

        // Add the current slide number to the URL hash so that reloading the
        // page/copying the URL will return you to the same slide
        hash: true,

        // Start with 1 for the hash rather than 0
        hashOneBasedIndex: false,

        // Flags if we should monitor the hash and change slides accordingly
        respondToHashChanges: true,

        // Push each slide change to the browser history
        history: true,

        // Enable keyboard shortcuts for navigation
        keyboard: true,

        // Enable the slide overview mode
        overview: true,

        // Disables the default reveal.js slide layout (scaling and centering)
        // so that you can use custom CSS layout
        disableLayout: false,

        // Vertical centering of slides
        center: false,

        // Enables touch navigation on devices with touch input
        touch: true,

        // Loop the presentation
        loop: false,

        // Change the presentation direction to be RTL
        rtl: false,

        // see https://revealjs.com/vertical-slides/#navigation-mode
        navigationMode: 'linear',

        // Randomizes the order of slides each time the presentation loads
        shuffle: false,

        // Turns fragments on and off globally
        fragments: true,

        // Flags whether to include the current fragment in the URL,
        // so that reloading brings you to the same fragment position
        fragmentInURL: false,

        // Flags if the presentation is running in an embedded mode,
        // i.e. contained within a limited portion of the screen
        embedded: false,

        // Flags if we should show a help overlay when the questionmark
        // key is pressed
        help: true,

        // Flags if it should be possible to pause the presentation (blackout)
        pause: true,

        // Flags if speaker notes should be visible to all viewers
        showNotes: false,

        // Global override for autoplaying embedded media (null/true/false)
        autoPlayMedia: null,

        // Global override for preloading lazy-loaded iframes (null/true/false)
        preloadIframes: null,

        // Number of milliseconds between automatically proceeding to the
        // next slide, disabled when set to 0, this value can be overwritten
        // by using a data-autoslide attribute on your slides
        autoSlide: 0,

        // Stop auto-sliding after user input
        autoSlideStoppable: true,

        // Use this method for navigation when auto-sliding
        autoSlideMethod: null,

        // Specify the average time in seconds that you think you will spend
        // presenting each slide. This is used to show a pacing timer in the
        // speaker view
        defaultTiming: null,

        // Enable slide navigation via mouse wheel
        mouseWheel: false,

        // The display mode that will be used to show slides
        display: 'block',

        // Hide cursor if inactive
        hideInactiveCursor: true,

        // Time before the cursor is hidden (in ms)
        hideCursorTime: 5000,

        // Opens links in an iframe preview overlay
        previewLinks: false,

        // Transition style (none/fade/slide/convex/concave/zoom)
        transition: 'fade',

        // Transition speed (default/fast/slow)
        transitionSpeed: 'default',

        // Transition style for full page slide backgrounds
        // (none/fade/slide/convex/concave/zoom)
        backgroundTransition: 'none',

        // Number of slides away from the current that are visible
        viewDistance: 3,

        // Number of slides away from the current that are visible on mobile
        // devices. It is advisable to set this to a lower number than
        // viewDistance in order to save resources.
        mobileViewDistance: 2,

        // The "normal" size of the presentation, aspect ratio will be preserved
        // when the presentation is scaled to fit different resolutions. Can be
        // specified using percentage units.
        width: 1050,

        height: 700,

        // Factor of the display size that should remain empty around the content
        margin: 0.1,

        math: {
          mathjax: 'https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.0/MathJax.js',
          config: 'TeX-AMS_HTML-full',
          tex2jax: {
            inlineMath: [['\\(','\\)']],
            displayMath: [['\\[','\\]']],
            balanceBraces: true,
            processEscapes: false,
            processRefs: true,
            processEnvironments: true,
            preview: 'TeX',
            skipTags: ['script','noscript','style','textarea','pre','code'],
            ignoreClass: 'tex2jax_ignore',
            processClass: 'tex2jax_process'
          },
        },

        // reveal.js plugins
        plugins: [QuartoLineHighlight, PdfExport, RevealMenu, QuartoSupport,

          RevealMath,
          RevealNotes,
          RevealSearch,
          RevealZoom
        ]
      });
    </script>
    
    <script>
      // htmlwidgets need to know to resize themselves when slides are shown/hidden.
      // Fire the "slideenter" event (handled by htmlwidgets.js) when the current
      // slide changes (different for each slide format).
      (function () {
        // dispatch for htmlwidgets
        function fireSlideEnter() {
          const event = window.document.createEvent("Event");
          event.initEvent("slideenter", true, true);
          window.document.dispatchEvent(event);
        }

        function fireSlideChanged(previousSlide, currentSlide) {
          fireSlideEnter();

          // dispatch for shiny
          if (window.jQuery) {
            if (previousSlide) {
              window.jQuery(previousSlide).trigger("hidden");
            }
            if (currentSlide) {
              window.jQuery(currentSlide).trigger("shown");
            }
          }
        }

        // hookup for slidy
        if (window.w3c_slidy) {
          window.w3c_slidy.add_observer(function (slide_num) {
            // slide_num starts at position 1
            fireSlideChanged(null, w3c_slidy.slides[slide_num - 1]);
          });
        }

      })();
    </script>

    <script id="quarto-html-after-body" type="application/javascript">
    window.document.addEventListener("DOMContentLoaded", function (event) {
      const toggleBodyColorMode = (bsSheetEl) => {
        const mode = bsSheetEl.getAttribute("data-mode");
        const bodyEl = window.document.querySelector("body");
        if (mode === "dark") {
          bodyEl.classList.add("quarto-dark");
          bodyEl.classList.remove("quarto-light");
        } else {
          bodyEl.classList.add("quarto-light");
          bodyEl.classList.remove("quarto-dark");
        }
      }
      const toggleBodyColorPrimary = () => {
        const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
        if (bsSheetEl) {
          toggleBodyColorMode(bsSheetEl);
        }
      }
      toggleBodyColorPrimary();  
      const tabsets =  window.document.querySelectorAll(".panel-tabset-tabby")
      tabsets.forEach(function(tabset) {
        const tabby = new Tabby('#' + tabset.id);
      });
      const isCodeAnnotation = (el) => {
        for (const clz of el.classList) {
          if (clz.startsWith('code-annotation-')) {                     
            return true;
          }
        }
        return false;
      }
      const onCopySuccess = function(e) {
        // button target
        const button = e.trigger;
        // don't keep focus
        button.blur();
        // flash "checked"
        button.classList.add('code-copy-button-checked');
        var currentTitle = button.getAttribute("title");
        button.setAttribute("title", "Copied!");
        let tooltip;
        if (window.bootstrap) {
          button.setAttribute("data-bs-toggle", "tooltip");
          button.setAttribute("data-bs-placement", "left");
          button.setAttribute("data-bs-title", "Copied!");
          tooltip = new bootstrap.Tooltip(button, 
            { trigger: "manual", 
              customClass: "code-copy-button-tooltip",
              offset: [0, -8]});
          tooltip.show();    
        }
        setTimeout(function() {
          if (tooltip) {
            tooltip.hide();
            button.removeAttribute("data-bs-title");
            button.removeAttribute("data-bs-toggle");
            button.removeAttribute("data-bs-placement");
          }
          button.setAttribute("title", currentTitle);
          button.classList.remove('code-copy-button-checked');
        }, 1000);
        // clear code selection
        e.clearSelection();
      }
      const getTextToCopy = function(trigger) {
          const codeEl = trigger.previousElementSibling.cloneNode(true);
          for (const childEl of codeEl.children) {
            if (isCodeAnnotation(childEl)) {
              childEl.remove();
            }
          }
          return codeEl.innerText;
      }
      const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
        text: getTextToCopy
      });
      clipboard.on('success', onCopySuccess);
      if (window.document.getElementById('quarto-embedded-source-code-modal')) {
        // For code content inside modals, clipBoardJS needs to be initialized with a container option
        // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
        const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
          text: getTextToCopy,
          container: window.document.getElementById('quarto-embedded-source-code-modal')
        });
        clipboardModal.on('success', onCopySuccess);
      }
        var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
        var mailtoRegex = new RegExp(/^mailto:/);
          var filterRegex = new RegExp('/' + window.location.host + '/');
        var isInternal = (href) => {
            return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
        }
        // Inspect non-navigation links and adorn them if external
     	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
        for (var i=0; i<links.length; i++) {
          const link = links[i];
          if (!isInternal(link.href)) {
            // undo the damage that might have been done by quarto-nav.js in the case of
            // links that we want to consider external
            if (link.dataset.originalHref !== undefined) {
              link.href = link.dataset.originalHref;
            }
          }
        }
      function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
        const config = {
          allowHTML: true,
          maxWidth: 500,
          delay: 100,
          arrow: false,
          appendTo: function(el) {
              return el.closest('section.slide') || el.parentElement;
          },
          interactive: true,
          interactiveBorder: 10,
          theme: 'light-border',
          placement: 'bottom-start',
        };
        if (contentFn) {
          config.content = contentFn;
        }
        if (onTriggerFn) {
          config.onTrigger = onTriggerFn;
        }
        if (onUntriggerFn) {
          config.onUntrigger = onUntriggerFn;
        }
          config['offset'] = [0,0];
          config['maxWidth'] = 700;
        window.tippy(el, config); 
      }
      const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
      for (var i=0; i<noterefs.length; i++) {
        const ref = noterefs[i];
        tippyHover(ref, function() {
          // use id or data attribute instead here
          let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
          try { href = new URL(href).hash; } catch {}
          const id = href.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note) {
            return note.innerHTML;
          } else {
            return "";
          }
        });
      }
      const findCites = (el) => {
        const parentEl = el.parentElement;
        if (parentEl) {
          const cites = parentEl.dataset.cites;
          if (cites) {
            return {
              el,
              cites: cites.split(' ')
            };
          } else {
            return findCites(el.parentElement)
          }
        } else {
          return undefined;
        }
      };
      var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
      for (var i=0; i<bibliorefs.length; i++) {
        const ref = bibliorefs[i];
        const citeInfo = findCites(ref);
        if (citeInfo) {
          tippyHover(citeInfo.el, function() {
            var popup = window.document.createElement('div');
            citeInfo.cites.forEach(function(cite) {
              var citeDiv = window.document.createElement('div');
              citeDiv.classList.add('hanging-indent');
              citeDiv.classList.add('csl-entry');
              var biblioDiv = window.document.getElementById('ref-' + cite);
              if (biblioDiv) {
                citeDiv.innerHTML = biblioDiv.innerHTML;
              }
              popup.appendChild(citeDiv);
            });
            return popup.innerHTML;
          });
        }
      }
    });
    </script>
    

</body></html>